
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<meta content="Genomics Aotearoa" name="author"/>
<link href="https://genomicsaotearoa.github.io/Pangenome-Graphs-Workshop/simulate_and_compare/" rel="canonical"/>
<link href="../preparing_data_files/" rel="prev"/>
<link href="../vc_position_based_comparison/" rel="next"/>
<link href="../theme_figures/nesi_ga.png" rel="icon"/>
<meta content="mkdocs-1.4.3, mkdocs-material-9.1.9" name="generator"/>
<title>Simulate and compare - Pangenome Graphs</title>
<link href="../assets/stylesheets/main.85bb2934.min.css" rel="stylesheet"/>
<link href="../assets/stylesheets/palette.a6bdf11c.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
<link href="../stylesheets/extra.css" rel="stylesheet"/>
<script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
<link href="../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            
                .gscrollbar-fixer { padding-right: 15px; }
                .gdesc-inner { font-size: 0.75rem; }
                body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
                body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
                body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
                </style><script src="../assets/javascripts/glightbox.min.js"></script></head>
<body data-md-color-accent="indigo" data-md-color-primary="indigo" data-md-color-scheme="default" dir="ltr">
<script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#a-script-to-simulate-and-compare-variant-calls-generated-using-linear-method-bwa-mem-and-graph-method-vg-giraffe">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--shadow" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Pangenome Graphs" class="md-header__button md-logo" data-md-component="logo" href=".." title="Pangenome Graphs">
<img alt="logo" src="../theme_figures/nesi_ga.png"/>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Pangenome Graphs
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              Simulate and compare
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_3" hidden="" title="Switch to light mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"></path></svg>
</label>
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_2" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
<input aria-label="Switch to system preference" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="indigo" data-md-color-scheme="slate" id="__palette_3" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to system preference">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
</form>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/Pangenome-Graphs-Workshop" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/Pangenome-Graphs-Workshop
  </div>
</a>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Pangenome Graphs" class="md-nav__button md-logo" data-md-component="logo" href=".." title="Pangenome Graphs">
<img alt="logo" src="../theme_figures/nesi_ga.png"/>
</a>
    Pangenome Graphs
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/Pangenome-Graphs-Workshop" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/Pangenome-Graphs-Workshop
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="..">
        Pangenome Graphs
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../Graph_based_variant_calling_for_NGS/">
        NGS data analysis used graph as reference on Nesi
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../Intro_1/">
        Pangenome Graphs Workshop
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../Intro_2/">
        Intro 2
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../mapping_reads/">
        Mapping Reads using `bwa mem` (Linear Method)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../pangenome_graph_construction/">
        pangenome graph construction with PGGB on Nesi
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../preparing_data_files/">
        Preparing required data files with simuG tool
      </a>
</li>
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" id="__toc" type="checkbox"/>
<label class="md-nav__link md-nav__link--active" for="__toc">
          Simulate and compare
          <span class="md-nav__icon md-icon"></span>
</label>
<a class="md-nav__link md-nav__link--active" href="./">
        Simulate and compare
      </a>
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#a-script-to-simulate-and-compare-variant-calls-generated-using-linear-method-bwa-mem-and-graph-method-vg-giraffe">
    A script to simulate and compare variant calls generated using linear method (bwa mem) and graph method (vg giraffe)
  </a>
<nav aria-label="A script to simulate and compare variant calls generated using linear method (bwa mem) and graph method (vg giraffe)" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#definitions">
    Definitions
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#references">
    References
  </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../vc_position_based_comparison/">
        How to compare 2 VCF files based on the exact variant position?
      </a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#a-script-to-simulate-and-compare-variant-calls-generated-using-linear-method-bwa-mem-and-graph-method-vg-giraffe">
    A script to simulate and compare variant calls generated using linear method (bwa mem) and graph method (vg giraffe)
  </a>
<nav aria-label="A script to simulate and compare variant calls generated using linear method (bwa mem) and graph method (vg giraffe)" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#definitions">
    Definitions
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#references">
    References
  </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<h1>Simulate and compare</h1>
<h3 id="a-script-to-simulate-and-compare-variant-calls-generated-using-linear-method-bwa-mem-and-graph-method-vg-giraffe">A script to simulate and compare variant calls generated using linear method (<code>bwa mem</code>) and graph method (<code>vg giraffe</code>)<a class="headerlink" href="#a-script-to-simulate-and-compare-variant-calls-generated-using-linear-method-bwa-mem-and-graph-method-vg-giraffe" title="Permanent link">¶</a></h3>
<p>This script <a href="https://github.com/nuzla/Pangenome-Graphs-Workshop/blob/main/Scripts/sim_vc_compare.sh">sim_vc_compare.sh</a> peforms the below tasks; (<em>Nesi folder : /nesi/nobackup/nesi02659/pg_workshop/vc_compare_script/</em>)
1. Simulate new sequence with predefined SNPs and INDELs count usnig <a href="https://github.com/yjx1217/simuG">simuG</a> and create ground truth VCF file (call it as <code>ground_truth.vcf</code>)
2. Simulate reads from the new sequence with with specific coverage depth and read length using <code>wgsim</code>
3. Map the reads with the reference using <code>bwa mem</code> and generate VCF file (call it as <code>simulated.vcf</code>)
4. Compare <code>ground_truth.vcf</code> and <code>simulated.vcf</code> using <code>bcftools isec</code> and generate a report
5. Repeat steps 3 and 4 using <code>vg giraffe</code>
6. Generate a comparision stats report</p>
<p>The script accepts following options. </p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-0-1" id="__codelineno-0-1" name="__codelineno-0-1"></a>$ ./sim_vc_compare.sh --help
<a href="#__codelineno-0-2" id="__codelineno-0-2" name="__codelineno-0-2"></a>Program : sim_vc_compare
<a href="#__codelineno-0-3" id="__codelineno-0-3" name="__codelineno-0-3"></a>Version : 1.0
<a href="#__codelineno-0-4" id="__codelineno-0-4" name="__codelineno-0-4"></a>Contact : fathima.nuzla.ismail@gmail.com
<a href="#__codelineno-0-5" id="__codelineno-0-5" name="__codelineno-0-5"></a>Usage   : sim_vc_compare.sh [options]
<a href="#__codelineno-0-6" id="__codelineno-0-6" name="__codelineno-0-6"></a>Options :
<a href="#__codelineno-0-7" id="__codelineno-0-7" name="__codelineno-0-7"></a>-r | --ref STR reference sequence file
<a href="#__codelineno-0-8" id="__codelineno-0-8" name="__codelineno-0-8"></a>-s | --snp INT Number of SNPs to simulate (Default 0)
<a href="#__codelineno-0-9" id="__codelineno-0-9" name="__codelineno-0-9"></a>-i | --indel INT Number of INDELs to simulate (Default 0)
<a href="#__codelineno-0-10" id="__codelineno-0-10" name="__codelineno-0-10"></a>-d | --depth INT Cover depth of the reads (Default 30)
<a href="#__codelineno-0-11" id="__codelineno-0-11" name="__codelineno-0-11"></a>-l | --length INT of a read (Default 100)
<a href="#__codelineno-0-12" id="__codelineno-0-12" name="__codelineno-0-12"></a>-o | --output STR Output folder name (Default 'output')
<a href="#__codelineno-0-13" id="__codelineno-0-13" name="__codelineno-0-13"></a>-h | --help Display this help message
</code></pre></div>
<p>For an example if we try the below 
<div class="highlight"><pre><span></span><code><a href="#__codelineno-1-1" id="__codelineno-1-1" name="__codelineno-1-1"></a>./sim_vc_compare.sh --ref GCF_000191525.1_ASM19152v1_genomic.fna -snp 5000 --indel 1000
</code></pre></div>
Script will produce the below report for <code>bwa mem</code>.</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-2-1" id="__codelineno-2-1" name="__codelineno-2-1"></a>+------------------------------------------------+
<a href="#__codelineno-2-2" id="__codelineno-2-2" name="__codelineno-2-2"></a>|  REPORT (BWA MEM)                              |
<a href="#__codelineno-2-3" id="__codelineno-2-3" name="__codelineno-2-3"></a>+------------------------------------------------+
<a href="#__codelineno-2-4" id="__codelineno-2-4" name="__codelineno-2-4"></a>|  Ground Truth SNPs                =      5,000 |
<a href="#__codelineno-2-5" id="__codelineno-2-5" name="__codelineno-2-5"></a>|  Ground Truth INDELs              =      1,000 |
<a href="#__codelineno-2-6" id="__codelineno-2-6" name="__codelineno-2-6"></a>|  Identified SNPs in Simulation    =      6,646 |
<a href="#__codelineno-2-7" id="__codelineno-2-7" name="__codelineno-2-7"></a>|  Identified INDELs in Simulation  =      1,666 |
<a href="#__codelineno-2-8" id="__codelineno-2-8" name="__codelineno-2-8"></a>|  SNPs Private to Simulation       =      1,831 |
<a href="#__codelineno-2-9" id="__codelineno-2-9" name="__codelineno-2-9"></a>|  INDELs Private to Simulation     =      1,153 |
<a href="#__codelineno-2-10" id="__codelineno-2-10" name="__codelineno-2-10"></a>|  Exact Matched SNPs               =      4,815 |
<a href="#__codelineno-2-11" id="__codelineno-2-11" name="__codelineno-2-11"></a>|  Exact Matched INDELs             =        513 |
<a href="#__codelineno-2-12" id="__codelineno-2-12" name="__codelineno-2-12"></a>|  True Positive (TP)               =      5,328 |
<a href="#__codelineno-2-13" id="__codelineno-2-13" name="__codelineno-2-13"></a>|  False Positive (FP)              =      2,984 |
<a href="#__codelineno-2-14" id="__codelineno-2-14" name="__codelineno-2-14"></a>|  True Negative (TN)               =  2,239,982 |
<a href="#__codelineno-2-15" id="__codelineno-2-15" name="__codelineno-2-15"></a>|  False Negative (FN)              =        672 |
<a href="#__codelineno-2-16" id="__codelineno-2-16" name="__codelineno-2-16"></a>+------------------------------------------------+
<a href="#__codelineno-2-17" id="__codelineno-2-17" name="__codelineno-2-17"></a>|  Sensitivity                      =   88.8000% |
<a href="#__codelineno-2-18" id="__codelineno-2-18" name="__codelineno-2-18"></a>|  Specificity                      =   99.8669% |
<a href="#__codelineno-2-19" id="__codelineno-2-19" name="__codelineno-2-19"></a>|  F1 Score                         =   74.4550% |
<a href="#__codelineno-2-20" id="__codelineno-2-20" name="__codelineno-2-20"></a>+------------------------------------------------+
</code></pre></div>
<h4 id="definitions">Definitions<a class="headerlink" href="#definitions" title="Permanent link">¶</a></h4>
<ol>
<li>True Positive (TP) = SNPs+INDELs which are exactly matched in <code>groud_truth.vcf</code> and <code>simulated.vcf</code> (TP=4,815+513=5,328)</li>
<li>False Positive (FP) = SNPs+INDELs private to <code>simulated.vcf</code> and not found in <code>groud_truth.vcf</code> (FP=1,831+1,153=2,984)</li>
<li>True Negative (TN) = Length of Reference the Sequence - Ground Truth SNPs - Ground Truth INDELs - False Positive. (TN=2,248,966-5,000-1,000-2,984=2,239,982)</li>
<li>False Negative (FN) = Ground Truth SNPs+Ground Truth INDELs - True Positive. (FN=5,000+1,000-5,328=672)</li>
<li>Sensitivity, Specificity, and F1 Score will be, </li>
</ol>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-3-1" id="__codelineno-3-1" name="__codelineno-3-1"></a>\begin{aligned}
<a href="#__codelineno-3-2" id="__codelineno-3-2" name="__codelineno-3-2"></a>Sensitivity  &amp; = \frac{TP}{TP+FN} \\
<a href="#__codelineno-3-3" id="__codelineno-3-3" name="__codelineno-3-3"></a>              &amp;  = \frac{5,328}{5,328+672} \\
<a href="#__codelineno-3-4" id="__codelineno-3-4" name="__codelineno-3-4"></a>              &amp; = 88.8000\% \\ \\
<a href="#__codelineno-3-5" id="__codelineno-3-5" name="__codelineno-3-5"></a>Specificity &amp; = \frac{TN}{TN+FP} \\
<a href="#__codelineno-3-6" id="__codelineno-3-6" name="__codelineno-3-6"></a>            &amp;  = \frac{2,239,982}{2,239,982+2,984} \\
<a href="#__codelineno-3-7" id="__codelineno-3-7" name="__codelineno-3-7"></a>            &amp; = 99.8669\% \\
<a href="#__codelineno-3-8" id="__codelineno-3-8" name="__codelineno-3-8"></a>F1\:Score &amp; = \frac{TP}{TP+\frac{1}{2}(FP+FN)} \\
<a href="#__codelineno-3-9" id="__codelineno-3-9" name="__codelineno-3-9"></a>            &amp;  = \frac{5,328}{5,328+0.5\times(2,984+672)} \\ \\
<a href="#__codelineno-3-10" id="__codelineno-3-10" name="__codelineno-3-10"></a>            &amp; = 74.4550\% \\
<a href="#__codelineno-3-11" id="__codelineno-3-11" name="__codelineno-3-11"></a>\end{aligned}
</code></pre></div>
<p>Script will also produce the below report for <code>vg giraffe</code> and the final comparison report.</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-4-1" id="__codelineno-4-1" name="__codelineno-4-1"></a>+------------------------------------------------+
<a href="#__codelineno-4-2" id="__codelineno-4-2" name="__codelineno-4-2"></a>|  REPORT (VG GIRAFFE)                           |
<a href="#__codelineno-4-3" id="__codelineno-4-3" name="__codelineno-4-3"></a>+------------------------------------------------+
<a href="#__codelineno-4-4" id="__codelineno-4-4" name="__codelineno-4-4"></a>|  Ground Truth SNPs                =      5,000 |
<a href="#__codelineno-4-5" id="__codelineno-4-5" name="__codelineno-4-5"></a>|  Ground Truth INDELs              =      1,000 |
<a href="#__codelineno-4-6" id="__codelineno-4-6" name="__codelineno-4-6"></a>|  Identified SNPs in Simulation    =      6,832 |
<a href="#__codelineno-4-7" id="__codelineno-4-7" name="__codelineno-4-7"></a>|  Identified INDELs in Simulation  =      1,486 |
<a href="#__codelineno-4-8" id="__codelineno-4-8" name="__codelineno-4-8"></a>|  SNPs Private to Simulation       =      1,839 |
<a href="#__codelineno-4-9" id="__codelineno-4-9" name="__codelineno-4-9"></a>|  INDELs Private to Simulation     =        753 |
<a href="#__codelineno-4-10" id="__codelineno-4-10" name="__codelineno-4-10"></a>|  Exact Matched SNPs               =      4,993 |
<a href="#__codelineno-4-11" id="__codelineno-4-11" name="__codelineno-4-11"></a>|  Exact Matched INDELs             =        733 |
<a href="#__codelineno-4-12" id="__codelineno-4-12" name="__codelineno-4-12"></a>|  True Positive (TP)               =      5,726 |
<a href="#__codelineno-4-13" id="__codelineno-4-13" name="__codelineno-4-13"></a>|  False Positive (FP)              =      2,592 |
<a href="#__codelineno-4-14" id="__codelineno-4-14" name="__codelineno-4-14"></a>|  True Negative (TN)               =  2,240,374 |
<a href="#__codelineno-4-15" id="__codelineno-4-15" name="__codelineno-4-15"></a>|  False Negative (FN)              =        274 |
<a href="#__codelineno-4-16" id="__codelineno-4-16" name="__codelineno-4-16"></a>+------------------------------------------------+
<a href="#__codelineno-4-17" id="__codelineno-4-17" name="__codelineno-4-17"></a>|  Sensitivity                      =   95.4333% |
<a href="#__codelineno-4-18" id="__codelineno-4-18" name="__codelineno-4-18"></a>|  Specificity                      =   99.8844% |
<a href="#__codelineno-4-19" id="__codelineno-4-19" name="__codelineno-4-19"></a>|  F1 Score                         =   79.9832% |
<a href="#__codelineno-4-20" id="__codelineno-4-20" name="__codelineno-4-20"></a>+------------------------------------------------+
</code></pre></div>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-5-1" id="__codelineno-5-1" name="__codelineno-5-1"></a>+-------------------------------------------------------------------------------------------------------------------------+
<a href="#__codelineno-5-2" id="__codelineno-5-2" name="__codelineno-5-2"></a>|  Method        |     TP       |     TN       |     FP       |     FN       |  Sensitivity |  Specificity |   F1 Score   |
<a href="#__codelineno-5-3" id="__codelineno-5-3" name="__codelineno-5-3"></a>+-------------------------------------------------------------------------------------------------------------------------+
<a href="#__codelineno-5-4" id="__codelineno-5-4" name="__codelineno-5-4"></a>|  bwa mem       |       5,328  |   2,239,982  |       2,984  |         672  |    88.8000%  |    99.8669%  |    74.4550%  |
<a href="#__codelineno-5-5" id="__codelineno-5-5" name="__codelineno-5-5"></a>+-------------------------------------------------------------------------------------------------------------------------+
<a href="#__codelineno-5-6" id="__codelineno-5-6" name="__codelineno-5-6"></a>|  vg giraffe    |       5,726  |   2,240,374  |       2,592  |         274  |    95.4333%  |    99.8844%  |    79.9832%  |
<a href="#__codelineno-5-7" id="__codelineno-5-7" name="__codelineno-5-7"></a>+-------------------------------------------------------------------------------------------------------------------------+
</code></pre></div>
<h4 id="references"><em>References</em><a class="headerlink" href="#references" title="Permanent link">¶</a></h4>
<ol>
<li><em>Alignment of high-throughput sequencing data using BWA. UC Davis Bioinformatics Core 2017 Variant Analysis Workshop. (n.d.). <a href="https://ucdavis-bioinformatics-training.github.io/2017-August-Variant-Analysis-Workshop/wednesday/alignment.html">https://ucdavis-bioinformatics-training.github.io/2017-August-Variant-Analysis-Workshop/wednesday/alignment.html</a></em> </li>
<li><em>Vgteam. (n.d.). Mapping short reads with giraffe. GitHub. <a href="https://github.com/vgteam/vg/wiki/Mapping-short-reads-with-Giraffe">https://github.com/vgteam/vg/wiki/Mapping-short-reads-with-Giraffe</a></em> </li>
<li><em>Paired-End vs. Single-Read Sequencing Technology. (n.d.). <a href="https://www.illumina.com/science/technology/next-generation-sequencing/plan-experiments/paired-end-vs-single-read.html">https://www.illumina.com/science/technology/next-generation-sequencing/plan-experiments/paired-end-vs-single-read.html</a></em></li>
<li><em>Buffalo, V. (2015). Bioinformatics Data Skills: Reproducible and Robust Research with Open Source Tools. “O’Reilly Media, Inc.”</em></li>
<li><em>Dudley, J. T., &amp; Karczewski, K. J. (2013). Exploring Personal Genomics. OUP Oxford.</em></li>
<li><em>Samtools - Documentation. (n.d.). <a href="https://www.htslib.org/doc/">https://www.htslib.org/doc/</a></em></li>
<li><em>Bash Reference Manual. (n.d.). <a href="https://www.gnu.org/software/bash/manual/bash.html">https://www.gnu.org/software/bash/manual/bash.html</a></em></li>
<li><em>Yjx. (n.d.). GitHub - yjx1217/simuG: simuG: a general-purpose genome simulator. GitHub. <a href="https://github.com/yjx1217/simuG">https://github.com/yjx1217/simuG</a></em></li>
<li><em>The Sequencing Center. (2022, September 26). What is de novo assembly? - The Sequencing Center. <a href="https://thesequencingcenter.com/knowledge-base/de-novo-assembly/">https://thesequencingcenter.com/knowledge-base/de-novo-assembly/</a></em></li>
<li><em>Lh. (n.d.). GitHub - lh3/wgsim: Reads simulator. GitHub. <a href="https://github.com/lh3/wgsim">https://github.com/lh3/wgsim</a></em></li>
<li><em>Wikipedia contributors. (2023). Sensitivity and specificity. Wikipedia. <a href="https://en.wikipedia.org/wiki/Sensitivity_and_specificity">https://en.wikipedia.org/wiki/Sensitivity_and_specificity</a></em></li>
</ol>
</article>
</div>
</div>
<button class="md-top md-icon" data-md-component="top" hidden="" type="button">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg>
            Back to top
          </button>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top"], "search": "../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
<script src="../assets/javascripts/bundle.fac441b0.min.js"></script>
<script src="../javascripts/mathjax.js"></script>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>