
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<meta content="Genomics Aotearoa" name="author"/>
<link href="https://genomicsaotearoa.github.io/Pangenome-Graphs-Workshop/mapping_reads/" rel="canonical"/>
<link href="../Intro_2/" rel="prev"/>
<link href="../pangenome_graph_construction/" rel="next"/>
<link href="../theme_figures/nesi_ga.png" rel="icon"/>
<meta content="mkdocs-1.4.3, mkdocs-material-9.1.9" name="generator"/>
<title>Mapping Reads using `bwa mem` (Linear Method) - Pangenome Graphs</title>
<link href="../assets/stylesheets/main.85bb2934.min.css" rel="stylesheet"/>
<link href="../assets/stylesheets/palette.a6bdf11c.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
<link href="../stylesheets/extra.css" rel="stylesheet"/>
<script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
<link href="../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            
                .gscrollbar-fixer { padding-right: 15px; }
                .gdesc-inner { font-size: 0.75rem; }
                body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
                body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
                body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
                </style><script src="../assets/javascripts/glightbox.min.js"></script></head>
<body data-md-color-accent="indigo" data-md-color-primary="indigo" data-md-color-scheme="default" dir="ltr">
<script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#mapping-reads-using-bwa-mem-linear-method">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--shadow" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Pangenome Graphs" class="md-header__button md-logo" data-md-component="logo" href=".." title="Pangenome Graphs">
<img alt="logo" src="../theme_figures/nesi_ga.png"/>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Pangenome Graphs
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              Mapping Reads using `bwa mem` (Linear Method)
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_3" hidden="" title="Switch to light mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"></path></svg>
</label>
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_2" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
<input aria-label="Switch to system preference" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="indigo" data-md-color-scheme="slate" id="__palette_3" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to system preference">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
</form>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/Pangenome-Graphs-Workshop" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/Pangenome-Graphs-Workshop
  </div>
</a>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Pangenome Graphs" class="md-nav__button md-logo" data-md-component="logo" href=".." title="Pangenome Graphs">
<img alt="logo" src="../theme_figures/nesi_ga.png"/>
</a>
    Pangenome Graphs
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/Pangenome-Graphs-Workshop" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/Pangenome-Graphs-Workshop
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="..">
        Pangenome Graphs
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../Graph_based_variant_calling_for_NGS/">
        NGS data analysis used graph as reference on Nesi
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../Intro_1/">
        Pangenome Graphs Workshop
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../Intro_2/">
        Intro 2
      </a>
</li>
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" id="__toc" type="checkbox"/>
<a class="md-nav__link md-nav__link--active" href="./">
        Mapping Reads using `bwa mem` (Linear Method)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../pangenome_graph_construction/">
        pangenome graph construction with PGGB on Nesi
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../preparing_data_files/">
        Preparing required data files with simuG tool
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../simulate_and_compare/">
        Simulate and compare
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../vc_position_based_comparison/">
        How to compare 2 VCF files based on the exact variant position?
      </a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<h1 id="mapping-reads-using-bwa-mem-linear-method">Mapping Reads using <code>bwa mem</code> (Linear Method)<a class="headerlink" href="#mapping-reads-using-bwa-mem-linear-method" title="Permanent link">¶</a></h1>
<p><em>Note: folder : /nesi/nobackup/nesi02659/pg_workshop/vc_exact_compare/</em>
We can use below script to map the reads to the reference sequnce using linear method <code>bwa mem</code>. 
<div class="highlight"><pre><span></span><code><a href="#__codelineno-0-1" id="__codelineno-0-1" name="__codelineno-0-1"></a><span class="c1">#Load required modules with specific versions</span>
<a href="#__codelineno-0-2" id="__codelineno-0-2" name="__codelineno-0-2"></a>module<span class="w"> </span>purge
<a href="#__codelineno-0-3" id="__codelineno-0-3" name="__codelineno-0-3"></a>module<span class="w"> </span>load<span class="w"> </span>BCFtools/1.9-GCC-7.4.0
<a href="#__codelineno-0-4" id="__codelineno-0-4" name="__codelineno-0-4"></a>module<span class="w"> </span>load<span class="w"> </span>SAMtools/1.9-GCC-7.4.0
<a href="#__codelineno-0-5" id="__codelineno-0-5" name="__codelineno-0-5"></a>module<span class="w"> </span>load<span class="w"> </span>BWA/0.7.17-GCC-9.2.0
<a href="#__codelineno-0-6" id="__codelineno-0-6" name="__codelineno-0-6"></a>module<span class="w"> </span>load<span class="w"> </span>wgsim/20111017-GCC-11.3.0
<a href="#__codelineno-0-7" id="__codelineno-0-7" name="__codelineno-0-7"></a>
<a href="#__codelineno-0-8" id="__codelineno-0-8" name="__codelineno-0-8"></a>mkdir<span class="w"> </span>vc_exact_compare
<a href="#__codelineno-0-9" id="__codelineno-0-9" name="__codelineno-0-9"></a><span class="nb">cd</span><span class="w"> </span>vc_exact_compare
<a href="#__codelineno-0-10" id="__codelineno-0-10" name="__codelineno-0-10"></a>
<a href="#__codelineno-0-11" id="__codelineno-0-11" name="__codelineno-0-11"></a><span class="c1">#Copy the below files into the folder</span>
<a href="#__codelineno-0-12" id="__codelineno-0-12" name="__codelineno-0-12"></a>$<span class="w"> </span>ls<span class="w"> </span>-1trhs
<a href="#__codelineno-0-13" id="__codelineno-0-13" name="__codelineno-0-13"></a>total<span class="w"> </span>14M
<a href="#__codelineno-0-14" id="__codelineno-0-14" name="__codelineno-0-14"></a><span class="m">2</span>.3M<span class="w"> </span>GCF_000191525.1_ASM19152v1_genomic.fna
<a href="#__codelineno-0-15" id="__codelineno-0-15" name="__codelineno-0-15"></a><span class="m">2</span>.3M<span class="w"> </span>Simulation_INDEL_5000.simseq.genome.fa
<a href="#__codelineno-0-16" id="__codelineno-0-16" name="__codelineno-0-16"></a><span class="m">2</span>.5M<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_CNV_4.simseq.genome.fa
<a href="#__codelineno-0-17" id="__codelineno-0-17" name="__codelineno-0-17"></a><span class="m">2</span>.3M<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_INV_4.simseq.genome.fa
<a href="#__codelineno-0-18" id="__codelineno-0-18" name="__codelineno-0-18"></a><span class="m">2</span>.3M<span class="w"> </span>Simulation_SNP_4000_INDEL_4000.simseq.genome.fa
<a href="#__codelineno-0-19" id="__codelineno-0-19" name="__codelineno-0-19"></a><span class="m">2</span>.3M<span class="w"> </span>Simulation_SNP_5000.simseq.genome.fa
<a href="#__codelineno-0-20" id="__codelineno-0-20" name="__codelineno-0-20"></a>
<a href="#__codelineno-0-21" id="__codelineno-0-21" name="__codelineno-0-21"></a><span class="c1">#indexing the reference sample</span>
<a href="#__codelineno-0-22" id="__codelineno-0-22" name="__codelineno-0-22"></a>bwa<span class="w"> </span>index<span class="w"> </span>GCF_000191525.1_ASM19152v1_genomic.fna<span class="w"> </span>
</code></pre></div>
In order to simulate a real sequensing experiment, we'll simulate the short reads too from the simulated full sequence using <code>wgsim</code> and map those reads to the reference sequnce using <code>bwa</code>. Since the length of the each sequnce is around 2.3 million, 0.7 millions of 100pb reads will give 30x read depth. (700000x100/2300000 ~ 30)</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-1-1" id="__codelineno-1-1" name="__codelineno-1-1"></a><span class="c1">#creating VCF files for each sample file considering GCF_000191525.1_ASM19152v1_genomic.fna as reference and silulating 30x 100bp reads. </span>
<a href="#__codelineno-1-2" id="__codelineno-1-2" name="__codelineno-1-2"></a><span class="c1">#export OMP_NUM_THREADS=1</span>
<a href="#__codelineno-1-3" id="__codelineno-1-3" name="__codelineno-1-3"></a>wgsim<span class="w"> </span>-N675000<span class="w"> </span>-1100<span class="w"> </span>-2100<span class="w"> </span>Simulation_SNP_5000.simseq.genome.fa<span class="w"> </span>Simulation_SNP_5000.read1.fq<span class="w"> </span>Simulation_SNP_5000.read2.fq<span class="w"> </span>
<a href="#__codelineno-1-4" id="__codelineno-1-4" name="__codelineno-1-4"></a>bwa<span class="w"> </span>mem<span class="w"> </span>-R<span class="w"> </span><span class="s2">"@RG\tID:Simulation_SNP_5000\tSM:Simulation_SNP_5000\tLB:L1"</span><span class="w"> </span>GCF_000191525.1_ASM19152v1_genomic.fna<span class="w"> </span>Simulation_SNP_5000.read1.fq<span class="w"> </span>Simulation_SNP_5000.read2.fq<span class="w"> </span>&gt;<span class="w"> </span>Simulation_SNP_5000.sam
<a href="#__codelineno-1-5" id="__codelineno-1-5" name="__codelineno-1-5"></a>samtools<span class="w"> </span>view<span class="w"> </span>-bS<span class="w"> </span>Simulation_SNP_5000.sam<span class="w"> </span><span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>sort<span class="w"> </span>-<span class="w"> </span>&gt;<span class="w"> </span>Simulation_SNP_5000.bam
<a href="#__codelineno-1-6" id="__codelineno-1-6" name="__codelineno-1-6"></a>bcftools<span class="w"> </span>mpileup<span class="w"> </span>-Ou<span class="w"> </span>-f<span class="w"> </span>GCF_000191525.1_ASM19152v1_genomic.fna<span class="w"> </span>Simulation_SNP_5000.bam<span class="w"> </span><span class="p">|</span><span class="w"> </span>bcftools<span class="w"> </span>call<span class="w"> </span>-vmO<span class="w"> </span>z<span class="w"> </span>-o<span class="w"> </span>Simulation_SNP_5000.bwa.30x.100R.vcf.gz
<a href="#__codelineno-1-7" id="__codelineno-1-7" name="__codelineno-1-7"></a>bcftools<span class="w"> </span>index<span class="w"> </span>Simulation_SNP_5000.bwa.30x.100R.vcf.gz<span class="w"> </span>
<a href="#__codelineno-1-8" id="__codelineno-1-8" name="__codelineno-1-8"></a>
<a href="#__codelineno-1-9" id="__codelineno-1-9" name="__codelineno-1-9"></a>wgsim<span class="w"> </span>-N675000<span class="w"> </span>-1100<span class="w"> </span>-2100<span class="w"> </span>Simulation_INDEL_5000.simseq.genome.fa<span class="w"> </span>Simulation_INDEL_5000.read1.fq<span class="w"> </span>Simulation_INDEL_5000.read2.fq<span class="w"> </span>
<a href="#__codelineno-1-10" id="__codelineno-1-10" name="__codelineno-1-10"></a>bwa<span class="w"> </span>mem<span class="w"> </span>-R<span class="w"> </span><span class="s2">"@RG\tID:Simulation_INDEL_5000\tSM:Simulation_INDEL_5000\tLB:L1"</span><span class="w"> </span>GCF_000191525.1_ASM19152v1_genomic.fna<span class="w"> </span>Simulation_INDEL_5000.read1.fq<span class="w"> </span>Simulation_INDEL_5000.read2.fq<span class="w"> </span>&gt;<span class="w"> </span>Simulation_INDEL_5000.sam
<a href="#__codelineno-1-11" id="__codelineno-1-11" name="__codelineno-1-11"></a>samtools<span class="w"> </span>view<span class="w"> </span>-bS<span class="w"> </span>Simulation_INDEL_5000.sam<span class="w"> </span><span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>sort<span class="w"> </span>-<span class="w"> </span>&gt;<span class="w"> </span>Simulation_INDEL_5000.bam
<a href="#__codelineno-1-12" id="__codelineno-1-12" name="__codelineno-1-12"></a>bcftools<span class="w"> </span>mpileup<span class="w"> </span>-Ou<span class="w"> </span>-f<span class="w"> </span>GCF_000191525.1_ASM19152v1_genomic.fna<span class="w"> </span>Simulation_INDEL_5000.bam<span class="w"> </span><span class="p">|</span><span class="w"> </span>bcftools<span class="w"> </span>call<span class="w"> </span>-vmO<span class="w"> </span>z<span class="w"> </span>-o<span class="w"> </span>Simulation_INDEL_5000.bwa.30x.100R.vcf.gz
<a href="#__codelineno-1-13" id="__codelineno-1-13" name="__codelineno-1-13"></a>bcftools<span class="w"> </span>index<span class="w"> </span>Simulation_INDEL_5000.bwa.30x.100R.vcf.gz<span class="w"> </span>
<a href="#__codelineno-1-14" id="__codelineno-1-14" name="__codelineno-1-14"></a>
<a href="#__codelineno-1-15" id="__codelineno-1-15" name="__codelineno-1-15"></a>wgsim<span class="w"> </span>-N675000<span class="w"> </span>-1100<span class="w"> </span>-2100<span class="w"> </span>Simulation_SNP_4000_INDEL_4000.simseq.genome.fa<span class="w"> </span>Simulation_SNP_4000_INDEL_4000.read1.fq<span class="w"> </span>Simulation_SNP_4000_INDEL_4000.read2.fq<span class="w"> </span>
<a href="#__codelineno-1-16" id="__codelineno-1-16" name="__codelineno-1-16"></a>bwa<span class="w"> </span>mem<span class="w"> </span>-R<span class="w"> </span><span class="s2">"@RG\tID:Simulation_SNP_4000_INDEL_4000\tSM:Simulation_SNP_4000_INDEL_4000\tLB:L1"</span><span class="w"> </span>GCF_000191525.1_ASM19152v1_genomic.fna<span class="w"> </span>Simulation_SNP_4000_INDEL_4000.read1.fq<span class="w">  </span>Simulation_SNP_4000_INDEL_4000.read2.fq<span class="w"> </span>&gt;<span class="w"> </span>Simulation_SNP_4000_INDEL_4000.sam
<a href="#__codelineno-1-17" id="__codelineno-1-17" name="__codelineno-1-17"></a>samtools<span class="w"> </span>view<span class="w"> </span>-bS<span class="w"> </span>Simulation_SNP_4000_INDEL_4000.sam<span class="w"> </span><span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>sort<span class="w"> </span>-<span class="w"> </span>&gt;<span class="w"> </span>Simulation_SNP_4000_INDEL_4000.bam
<a href="#__codelineno-1-18" id="__codelineno-1-18" name="__codelineno-1-18"></a>bcftools<span class="w"> </span>mpileup<span class="w"> </span>-Ou<span class="w"> </span>-f<span class="w"> </span>GCF_000191525.1_ASM19152v1_genomic.fna<span class="w"> </span>Simulation_SNP_4000_INDEL_4000.bam<span class="w"> </span><span class="p">|</span><span class="w"> </span>bcftools<span class="w"> </span>call<span class="w"> </span>-vmO<span class="w"> </span>z<span class="w"> </span>-o<span class="w"> </span>Simulation_SNP_4000_INDEL_4000.bwa.30x.100R.vcf.gz
<a href="#__codelineno-1-19" id="__codelineno-1-19" name="__codelineno-1-19"></a>bcftools<span class="w"> </span>index<span class="w"> </span>Simulation_SNP_4000_INDEL_4000.bwa.30x.100R.vcf.gz
<a href="#__codelineno-1-20" id="__codelineno-1-20" name="__codelineno-1-20"></a>
<a href="#__codelineno-1-21" id="__codelineno-1-21" name="__codelineno-1-21"></a>wgsim<span class="w"> </span>-N675000<span class="w"> </span>-1100<span class="w"> </span>-2100<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_INV_4.simseq.genome.fa<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_INV_4.read1.fq<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_INV_4.read2.fq
<a href="#__codelineno-1-22" id="__codelineno-1-22" name="__codelineno-1-22"></a>bwa<span class="w"> </span>mem<span class="w"> </span>-R<span class="w"> </span><span class="s2">"@RG\tID:Simulation_SNP_4000_INDEL_4000_INV_4\tSM:Simulation_SNP_4000_INDEL_4000_INV_4\tLB:L1"</span><span class="w"> </span>GCF_000191525.1_ASM19152v1_genomic.fna<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_INV_4.read1.fq<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_INV_4.read2.fq<span class="w"> </span>&gt;<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_INV_4.sam
<a href="#__codelineno-1-23" id="__codelineno-1-23" name="__codelineno-1-23"></a>samtools<span class="w"> </span>view<span class="w"> </span>-bS<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_INV_4.sam<span class="w"> </span><span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>sort<span class="w"> </span>-<span class="w"> </span>&gt;<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_INV_4.bam
<a href="#__codelineno-1-24" id="__codelineno-1-24" name="__codelineno-1-24"></a>bcftools<span class="w"> </span>mpileup<span class="w"> </span>-Ou<span class="w"> </span>-f<span class="w"> </span>GCF_000191525.1_ASM19152v1_genomic.fna<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_INV_4.bam<span class="w"> </span><span class="p">|</span><span class="w"> </span>bcftools<span class="w"> </span>call<span class="w"> </span>-vmO<span class="w"> </span>z<span class="w"> </span>-o<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_INV_4.bwa.30x.100R.vcf.gz
<a href="#__codelineno-1-25" id="__codelineno-1-25" name="__codelineno-1-25"></a>bcftools<span class="w"> </span>index<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_INV_4.bwa.30x.100R.vcf.gz
<a href="#__codelineno-1-26" id="__codelineno-1-26" name="__codelineno-1-26"></a>
<a href="#__codelineno-1-27" id="__codelineno-1-27" name="__codelineno-1-27"></a>wgsim<span class="w"> </span>-N675000<span class="w"> </span>-1100<span class="w"> </span>-2100<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_CNV_4.simseq.genome.fa<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_CNV_4.read1.fq<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_CNV_4.read2.fq
<a href="#__codelineno-1-28" id="__codelineno-1-28" name="__codelineno-1-28"></a>bwa<span class="w"> </span>mem<span class="w"> </span>-R<span class="w"> </span><span class="s2">"@RG\tID:Simulation_SNP_4000_INDEL_4000_CNV_4\tSM:Simulation_SNP_4000_INDEL_4000_CNV_4\tLB:L1"</span><span class="w"> </span>GCF_000191525.1_ASM19152v1_genomic.fna<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_CNV_4.read1.fq<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_CNV_4.read2.fq<span class="w"> </span>&gt;<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_CNV_4.sam
<a href="#__codelineno-1-29" id="__codelineno-1-29" name="__codelineno-1-29"></a>samtools<span class="w"> </span>view<span class="w"> </span>-bS<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_CNV_4.sam<span class="w"> </span><span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>sort<span class="w"> </span>-<span class="w"> </span>&gt;<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_CNV_4.bam
<a href="#__codelineno-1-30" id="__codelineno-1-30" name="__codelineno-1-30"></a>bcftools<span class="w"> </span>mpileup<span class="w"> </span>-Ou<span class="w"> </span>-f<span class="w"> </span>GCF_000191525.1_ASM19152v1_genomic.fna<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_CNV_4.bam<span class="w"> </span><span class="p">|</span><span class="w"> </span>bcftools<span class="w"> </span>call<span class="w"> </span>-vmO<span class="w"> </span>z<span class="w"> </span>-o<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_CNV_4.bwa.30x.100R.vcf.gz
<a href="#__codelineno-1-31" id="__codelineno-1-31" name="__codelineno-1-31"></a>bcftools<span class="w"> </span>index<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_CNV_4.bwa.30x.100R.vcf.gz
</code></pre></div>
<p>(You can find above script as a SLURM job script here <a href="https://github.com/nuzla/Pangenome-Graphs-Workshop/blob/main/Scripts/vc_bwa_compare.sh">vc_bwa_compare.sh</a>)</p>
<p>Now we can find the variant call stats using <code>bcftools stats</code>. </p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-2-1" id="__codelineno-2-1" name="__codelineno-2-1"></a>$ bcftools stats Simulation_SNP_5000.bwa.30x.100R.vcf.gz | head -30
<a href="#__codelineno-2-2" id="__codelineno-2-2" name="__codelineno-2-2"></a># This file was produced by bcftools stats (1.9+htslib-1.9) and can be plotted using plot-vcfstats.
<a href="#__codelineno-2-3" id="__codelineno-2-3" name="__codelineno-2-3"></a># The command line was: bcftools stats  Simulation_SNP_5000.bwa.30x.100R.vcf.gz
<a href="#__codelineno-2-4" id="__codelineno-2-4" name="__codelineno-2-4"></a>#
<a href="#__codelineno-2-5" id="__codelineno-2-5" name="__codelineno-2-5"></a># Definition of sets:
<a href="#__codelineno-2-6" id="__codelineno-2-6" name="__codelineno-2-6"></a># ID    [2]id   [3]tab-separated file names
<a href="#__codelineno-2-7" id="__codelineno-2-7" name="__codelineno-2-7"></a>ID      0       Simulation_SNP_5000.bwa.30x.100R.vcf.gz
<a href="#__codelineno-2-8" id="__codelineno-2-8" name="__codelineno-2-8"></a># SN, Summary numbers:
<a href="#__codelineno-2-9" id="__codelineno-2-9" name="__codelineno-2-9"></a>#   number of records   .. number of data rows in the VCF
<a href="#__codelineno-2-10" id="__codelineno-2-10" name="__codelineno-2-10"></a>#   number of no-ALTs   .. reference-only sites, ALT is either "." or identical to REF
<a href="#__codelineno-2-11" id="__codelineno-2-11" name="__codelineno-2-11"></a>#   number of SNPs      .. number of rows with a SNP
<a href="#__codelineno-2-12" id="__codelineno-2-12" name="__codelineno-2-12"></a>#   number of MNPs      .. number of rows with a MNP, such as CC&gt;TT
<a href="#__codelineno-2-13" id="__codelineno-2-13" name="__codelineno-2-13"></a>#   number of indels    .. number of rows with an indel
<a href="#__codelineno-2-14" id="__codelineno-2-14" name="__codelineno-2-14"></a>#   number of others    .. number of rows with other type, for example a symbolic allele or
<a href="#__codelineno-2-15" id="__codelineno-2-15" name="__codelineno-2-15"></a>#                          a complex substitution, such as ACT&gt;TCGA
<a href="#__codelineno-2-16" id="__codelineno-2-16" name="__codelineno-2-16"></a>#   number of multiallelic sites     .. number of rows with multiple alternate alleles
<a href="#__codelineno-2-17" id="__codelineno-2-17" name="__codelineno-2-17"></a>#   number of multiallelic SNP sites .. number of rows with multiple alternate alleles, all SNPs
<a href="#__codelineno-2-18" id="__codelineno-2-18" name="__codelineno-2-18"></a># 
<a href="#__codelineno-2-19" id="__codelineno-2-19" name="__codelineno-2-19"></a>#   Note that rows containing multiple types will be counted multiple times, in each
<a href="#__codelineno-2-20" id="__codelineno-2-20" name="__codelineno-2-20"></a>#   counter. For example, a row with a SNP and an indel increments both the SNP and
<a href="#__codelineno-2-21" id="__codelineno-2-21" name="__codelineno-2-21"></a>#   the indel counter.
<a href="#__codelineno-2-22" id="__codelineno-2-22" name="__codelineno-2-22"></a># 
<a href="#__codelineno-2-23" id="__codelineno-2-23" name="__codelineno-2-23"></a># SN    [2]id   [3]key  [4]value
<a href="#__codelineno-2-24" id="__codelineno-2-24" name="__codelineno-2-24"></a>SN      0       number of samples:      1
<a href="#__codelineno-2-25" id="__codelineno-2-25" name="__codelineno-2-25"></a>SN      0       number of records:      6740
<a href="#__codelineno-2-26" id="__codelineno-2-26" name="__codelineno-2-26"></a>SN      0       number of no-ALTs:      0
<a href="#__codelineno-2-27" id="__codelineno-2-27" name="__codelineno-2-27"></a>SN      0       number of SNPs: 6395
<a href="#__codelineno-2-28" id="__codelineno-2-28" name="__codelineno-2-28"></a>SN      0       number of MNPs: 0
<a href="#__codelineno-2-29" id="__codelineno-2-29" name="__codelineno-2-29"></a>SN      0       number of indels:       345
<a href="#__codelineno-2-30" id="__codelineno-2-30" name="__codelineno-2-30"></a>SN      0       number of others:       0
<a href="#__codelineno-2-31" id="__codelineno-2-31" name="__codelineno-2-31"></a>SN      0       number of multiallelic sites:   1
</code></pre></div>
<h1 id="mapping-reads-using-vg-giraffe-graph-method">Mapping Reads using <code>vg giraffe</code> (Graph Method)<a class="headerlink" href="#mapping-reads-using-vg-giraffe-graph-method" title="Permanent link">¶</a></h1>
<p>Here we map reads to a pangenome graph instead of single linear reference sequence. For example we'll consider the first case <code>Simulation_INDEL_5000.simseq.genome.fa</code>. We can build a graph considering the reference sequence <code>GCF_000191525.1_ASM19152v1_genomic.fna</code> and the ground truth VCF file </p>
<p><div class="highlight"><pre><span></span><code><a href="#__codelineno-3-1" id="__codelineno-3-1" name="__codelineno-3-1"></a><span class="c1">#Load  additional model need for vg</span>
<a href="#__codelineno-3-2" id="__codelineno-3-2" name="__codelineno-3-2"></a>module<span class="w"> </span>load<span class="w"> </span>vg/1.46.0
<a href="#__codelineno-3-3" id="__codelineno-3-3" name="__codelineno-3-3"></a>
<a href="#__codelineno-3-4" id="__codelineno-3-4" name="__codelineno-3-4"></a><span class="c1">#Copy the vcf files into the folder</span>
<a href="#__codelineno-3-5" id="__codelineno-3-5" name="__codelineno-3-5"></a>ls<span class="w"> </span>-1trhs<span class="w"> </span>Simulation_*.vcf
<a href="#__codelineno-3-6" id="__codelineno-3-6" name="__codelineno-3-6"></a>768K<span class="w"> </span>Simulation_SNP_5000.refseq2simseq.SNP.vcf
<a href="#__codelineno-3-7" id="__codelineno-3-7" name="__codelineno-3-7"></a>768K<span class="w"> </span>Simulation_INDEL_5000.refseq2simseq.INDEL.vcf
<a href="#__codelineno-3-8" id="__codelineno-3-8" name="__codelineno-3-8"></a>768K<span class="w"> </span>Simulation_SNP_4000_INDEL_4000.refseq2simseq.SNP.vcf
<a href="#__codelineno-3-9" id="__codelineno-3-9" name="__codelineno-3-9"></a>768K<span class="w"> </span>Simulation_SNP_4000_INDEL_4000.refseq2simseq.INDEL.vcf
<a href="#__codelineno-3-10" id="__codelineno-3-10" name="__codelineno-3-10"></a><span class="w"> </span><span class="m">512</span><span class="w"> </span>Simulation_SNP_4000_INDEL_4000_INV_4.refseq2simseq.inversion.vcf
<a href="#__codelineno-3-11" id="__codelineno-3-11" name="__codelineno-3-11"></a>256K<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_CNV_4.refseq2simseq.CNV.vcf
<a href="#__codelineno-3-12" id="__codelineno-3-12" name="__codelineno-3-12"></a>
<a href="#__codelineno-3-13" id="__codelineno-3-13" name="__codelineno-3-13"></a><span class="c1">#create tabix index</span>
<a href="#__codelineno-3-14" id="__codelineno-3-14" name="__codelineno-3-14"></a>bgzip<span class="w"> </span>Simulation_SNP_5000.refseq2simseq.SNP.vcf
<a href="#__codelineno-3-15" id="__codelineno-3-15" name="__codelineno-3-15"></a>tabix<span class="w"> </span>Simulation_SNP_5000.refseq2simseq.SNP.vcf.gz
<a href="#__codelineno-3-16" id="__codelineno-3-16" name="__codelineno-3-16"></a>
<a href="#__codelineno-3-17" id="__codelineno-3-17" name="__codelineno-3-17"></a><span class="c1">#create the graph and index (-p is prefix for filenames)</span>
<a href="#__codelineno-3-18" id="__codelineno-3-18" name="__codelineno-3-18"></a>vg<span class="w"> </span>autoindex<span class="w"> </span>--workflow<span class="w"> </span>giraffe<span class="w"> </span>-r<span class="w"> </span>GCF_000191525.1_ASM19152v1_genomic.fna<span class="w"> </span>-v<span class="w"> </span>Simulation_SNP_5000.refseq2simseq.SNP.vcf.gz<span class="w"> </span>-p<span class="w"> </span>VG_SNP_5000
<a href="#__codelineno-3-19" id="__codelineno-3-19" name="__codelineno-3-19"></a>
<a href="#__codelineno-3-20" id="__codelineno-3-20" name="__codelineno-3-20"></a><span class="c1">#Map reads to the graph</span>
<a href="#__codelineno-3-21" id="__codelineno-3-21" name="__codelineno-3-21"></a>vg<span class="w"> </span>giraffe<span class="w"> </span>-Z<span class="w"> </span>VG_SNP_5000.giraffe.gbz<span class="w"> </span>-f<span class="w"> </span>Simulation_INDEL_5000.read1.fq<span class="w"> </span>-f<span class="w"> </span>Simulation_INDEL_5000.read2.fq<span class="w"> </span>-o<span class="w"> </span>SAM<span class="w"> </span>&gt;<span class="w"> </span>Simulation_VG_SNP_5000.sam
<a href="#__codelineno-3-22" id="__codelineno-3-22" name="__codelineno-3-22"></a>
<a href="#__codelineno-3-23" id="__codelineno-3-23" name="__codelineno-3-23"></a><span class="c1">#Follow same procedure for VCF file creation</span>
<a href="#__codelineno-3-24" id="__codelineno-3-24" name="__codelineno-3-24"></a>samtools<span class="w"> </span>view<span class="w"> </span>-bS<span class="w"> </span>Simulation_VG_SNP_5000.sam<span class="w"> </span><span class="p">|</span><span class="w"> </span>samtools<span class="w"> </span>sort<span class="w"> </span>-<span class="w"> </span>&gt;<span class="w"> </span>Simulation_VG_SNP_5000.bam
<a href="#__codelineno-3-25" id="__codelineno-3-25" name="__codelineno-3-25"></a>bcftools<span class="w"> </span>mpileup<span class="w"> </span>-Ou<span class="w"> </span>-f<span class="w"> </span>GCF_000191525.1_ASM19152v1_genomic.fna<span class="w"> </span>Simulation_VG_SNP_5000.bam<span class="w"> </span><span class="p">|</span><span class="w"> </span>bcftools<span class="w"> </span>call<span class="w"> </span>-vmO<span class="w"> </span>z<span class="w"> </span>-o<span class="w"> </span>Simulation_VG_SNP_5000.giraffe.30x.100R.vcf.gz
<a href="#__codelineno-3-26" id="__codelineno-3-26" name="__codelineno-3-26"></a>bcftools<span class="w"> </span>index<span class="w"> </span>Simulation_VG_SNP_5000.giraffe.30x.100R.vcf.gz<span class="w"> </span>
</code></pre></div>
We can follow the same procedure for the rest of the samples and generate VCF files. </p>
</article>
</div>
</div>
<button class="md-top md-icon" data-md-component="top" hidden="" type="button">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg>
            Back to top
          </button>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top"], "search": "../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
<script src="../assets/javascripts/bundle.fac441b0.min.js"></script>
<script src="../javascripts/mathjax.js"></script>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>