
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<meta content="Genomics Aotearoa" name="author"/>
<link href="https://genomicsaotearoa.github.io/Pangenome-Graphs-Workshop/vc_position_based_comparison/" rel="canonical"/>
<link href="../simulate_and_compare/" rel="prev"/>
<link href="../theme_figures/nesi_ga.png" rel="icon"/>
<meta content="mkdocs-1.4.3, mkdocs-material-9.1.9" name="generator"/>
<title>How to compare 2 VCF files based on the exact variant position? - Pangenome Graphs</title>
<link href="../assets/stylesheets/main.85bb2934.min.css" rel="stylesheet"/>
<link href="../assets/stylesheets/palette.a6bdf11c.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
<link href="../stylesheets/extra.css" rel="stylesheet"/>
<script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
<link href="../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            
                .gscrollbar-fixer { padding-right: 15px; }
                .gdesc-inner { font-size: 0.75rem; }
                body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
                body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
                body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
                </style><script src="../assets/javascripts/glightbox.min.js"></script></head>
<body data-md-color-accent="indigo" data-md-color-primary="indigo" data-md-color-scheme="default" dir="ltr">
<script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#how-to-compare-2-vcf-files-based-on-the-exact-variant-position">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--shadow" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Pangenome Graphs" class="md-header__button md-logo" data-md-component="logo" href=".." title="Pangenome Graphs">
<img alt="logo" src="../theme_figures/nesi_ga.png"/>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Pangenome Graphs
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              How to compare 2 VCF files based on the exact variant position?
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_3" hidden="" title="Switch to light mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"></path></svg>
</label>
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_2" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
<input aria-label="Switch to system preference" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="indigo" data-md-color-scheme="slate" id="__palette_3" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to system preference">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
</form>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/Pangenome-Graphs-Workshop" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/Pangenome-Graphs-Workshop
  </div>
</a>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Pangenome Graphs" class="md-nav__button md-logo" data-md-component="logo" href=".." title="Pangenome Graphs">
<img alt="logo" src="../theme_figures/nesi_ga.png"/>
</a>
    Pangenome Graphs
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/Pangenome-Graphs-Workshop" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/Pangenome-Graphs-Workshop
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="..">
        Pangenome Graphs
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../Graph_based_variant_calling_for_NGS/">
        NGS data analysis used graph as reference on Nesi
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../Intro_1/">
        Pangenome Graphs Workshop
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../Intro_2/">
        Intro 2
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../mapping_reads/">
        Mapping Reads using `bwa mem` (Linear Method)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../pangenome_graph_construction/">
        pangenome graph construction with PGGB on Nesi
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../preparing_data_files/">
        Preparing required data files with simuG tool
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../simulate_and_compare/">
        Simulate and compare
      </a>
</li>
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" id="__toc" type="checkbox"/>
<label class="md-nav__link md-nav__link--active" for="__toc">
          How to compare 2 VCF files based on the exact variant position?
          <span class="md-nav__icon md-icon"></span>
</label>
<a class="md-nav__link md-nav__link--active" href="./">
        How to compare 2 VCF files based on the exact variant position?
      </a>
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#ground-truth-vs-linear-based-variant-comparision">
    Ground Truth vs Linear based variant comparision
  </a>
</li>
</ul>
</nav>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#ground-truth-vs-linear-based-variant-comparision">
    Ground Truth vs Linear based variant comparision
  </a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<h1 id="how-to-compare-2-vcf-files-based-on-the-exact-variant-position">How to compare 2 VCF files based on the exact variant position?<a class="headerlink" href="#how-to-compare-2-vcf-files-based-on-the-exact-variant-position" title="Permanent link">¶</a></h1>
<p>When simulating the new sequences in the previous procedure <a href="https://github.com/nuzla/Pangenome-Graphs-Workshop/blob/main/preparing_data_files.md">Preparing required data files with simuG tool</a> related VCF files were also generated (e.g. Simulation_SNP_5000.refseq2simseq.SNP.vcf). We can compare that VCF file with linear based VCF file and the graph based files to find variant differences considering the exact bp positions. <code>bcftools isec</code> command can be used for this. </p>
<p>If the ground truch simulated VCF is <code>Simulation_SNP_5000.refseq2simseq.SNP.vcf</code> (generated by <a href="https://github.com/nuzla/Pangenome-Graphs-Workshop/blob/main/preparing_data_files.md">https://github.com/nuzla/Pangenome-Graphs-Workshop/blob/main/preparing_data_files.md</a>) and the linear method based vcf file (generated by <a href="https://github.com/nuzla/Pangenome-Graphs-Workshop/blob/main/linear_reference_vc.md">https://github.com/nuzla/Pangenome-Graphs-Workshop/blob/main/linear_reference_vc.md</a>) is <code>Simulation_SNP_5000.bwa.30x.100R.vcf</code>; first we need to <code>bgzip</code> the files and make indexes. </p>
<h3 id="ground-truth-vs-linear-based-variant-comparision">Ground Truth vs Linear based variant comparision<a class="headerlink" href="#ground-truth-vs-linear-based-variant-comparision" title="Permanent link">¶</a></h3>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-0-1" id="__codelineno-0-1" name="__codelineno-0-1"></a>bgzip<span class="w"> </span>Simulation_SNP_5000.refseq2simseq.SNP.vcf
<a href="#__codelineno-0-2" id="__codelineno-0-2" name="__codelineno-0-2"></a>bgzip<span class="w"> </span>Simulation_SNP_5000.bwa.30x.100R.vcf
<a href="#__codelineno-0-3" id="__codelineno-0-3" name="__codelineno-0-3"></a>bcftools<span class="w"> </span>index<span class="w"> </span>Simulation_SNP_5000.bwa.30x.100R.vcf.gz
<a href="#__codelineno-0-4" id="__codelineno-0-4" name="__codelineno-0-4"></a>bcftools<span class="w"> </span>index<span class="w"> </span>Simulation_SNP_5000.refseq2simseq.SNP.vcf.gz
</code></pre></div>
<p>Then use the below commands to compare. Here the option <code>-c none</code> will do the exact matching <code>-p</code> for specifying the output folder. </p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-1-1" id="__codelineno-1-1" name="__codelineno-1-1"></a>$<span class="w"> </span>bcftools<span class="w"> </span>isec<span class="w"> </span>-c<span class="w"> </span>none<span class="w"> </span>-p<span class="w"> </span>output_SNP_5000_bwa.30x.100R<span class="w"> </span>Simulation_SNP_5000.refseq2simseq.SNP.vcf.gz<span class="w"> </span>Simulation_SNP_5000.bwa.30x.100R.vcf.gz
<a href="#__codelineno-1-2" id="__codelineno-1-2" name="__codelineno-1-2"></a>ls<span class="w"> </span>-1sh<span class="w"> </span>output_SNP_5000_bwa.30x.100R
<a href="#__codelineno-1-3" id="__codelineno-1-3" name="__codelineno-1-3"></a>total<span class="w"> </span><span class="m">2</span>.3M
<a href="#__codelineno-1-4" id="__codelineno-1-4" name="__codelineno-1-4"></a>256K<span class="w"> </span><span class="m">0000</span>.vcf
<a href="#__codelineno-1-5" id="__codelineno-1-5" name="__codelineno-1-5"></a>512K<span class="w"> </span><span class="m">0001</span>.vcf
<a href="#__codelineno-1-6" id="__codelineno-1-6" name="__codelineno-1-6"></a>768K<span class="w"> </span><span class="m">0002</span>.vcf
<a href="#__codelineno-1-7" id="__codelineno-1-7" name="__codelineno-1-7"></a>768K<span class="w"> </span><span class="m">0003</span>.vcf
<a href="#__codelineno-1-8" id="__codelineno-1-8" name="__codelineno-1-8"></a><span class="w"> </span><span class="m">512</span><span class="w"> </span>README.txt
</code></pre></div>
<p>The README.txt file has the details about what each vcf file has. </p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-2-1" id="__codelineno-2-1" name="__codelineno-2-1"></a>$ cat output_SNP_5000_bwa.30x.100R/README.txt 
<a href="#__codelineno-2-2" id="__codelineno-2-2" name="__codelineno-2-2"></a>This file was produced by vcfisec.
<a href="#__codelineno-2-3" id="__codelineno-2-3" name="__codelineno-2-3"></a>The command line was:   bcftools isec  -c none -p output_SNP_5000_bwa.30x.100R Simulation_SNP_5000.refseq2simseq.SNP.vcf.gz Simulation_SNP_5000.bwa.30x.100R.vcf.gz
<a href="#__codelineno-2-4" id="__codelineno-2-4" name="__codelineno-2-4"></a>
<a href="#__codelineno-2-5" id="__codelineno-2-5" name="__codelineno-2-5"></a>Using the following file names:
<a href="#__codelineno-2-6" id="__codelineno-2-6" name="__codelineno-2-6"></a>output_SNP_5000_bwa.30x.100R/0000.vcf   for records private to  Simulation_SNP_5000.refseq2simseq.SNP.vcf.gz
<a href="#__codelineno-2-7" id="__codelineno-2-7" name="__codelineno-2-7"></a>output_SNP_5000_bwa.30x.100R/0001.vcf   for records private to  Simulation_SNP_5000.bwa.30x.100R.vcf.gz
<a href="#__codelineno-2-8" id="__codelineno-2-8" name="__codelineno-2-8"></a>output_SNP_5000_bwa.30x.100R/0002.vcf   for records from Simulation_SNP_5000.refseq2simseq.SNP.vcf.gz shared by both    Simulation_SNP_5000.refseq2simseq.SNP.vcf.gz Simulation_SNP_5000.bwa.30x.100R.vcf.gz
<a href="#__codelineno-2-9" id="__codelineno-2-9" name="__codelineno-2-9"></a>output_SNP_5000_bwa.30x.100R/0003.vcf   for records from Simulation_SNP_5000.bwa.30x.100R.vcf.gz shared by both Simulation_SNP_5000.refseq2simseq.SNP.vcf.gz Simulation_SNP_5000.bwa.30x.100R.vcf.gz
</code></pre></div>
<p>As per this explnation, 0002.vcf file should have the True Positive (TP) stats. </p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-3-1" id="__codelineno-3-1" name="__codelineno-3-1"></a>$ bcftools stats output_SNP_5000_bwa.30x.100R/0002.vcf | head -40
<a href="#__codelineno-3-2" id="__codelineno-3-2" name="__codelineno-3-2"></a># This file was produced by bcftools stats (1.9+htslib-1.9) and can be plotted using plot-vcfstats.
<a href="#__codelineno-3-3" id="__codelineno-3-3" name="__codelineno-3-3"></a># The command line was: bcftools stats  output_SNP_5000_bwa.30x.100R/0002.vcf
<a href="#__codelineno-3-4" id="__codelineno-3-4" name="__codelineno-3-4"></a>#
<a href="#__codelineno-3-5" id="__codelineno-3-5" name="__codelineno-3-5"></a># Definition of sets:
<a href="#__codelineno-3-6" id="__codelineno-3-6" name="__codelineno-3-6"></a># ID    [2]id   [3]tab-separated file names
<a href="#__codelineno-3-7" id="__codelineno-3-7" name="__codelineno-3-7"></a>ID      0       output_SNP_5000_bwa.30x.100R/0002.vcf
<a href="#__codelineno-3-8" id="__codelineno-3-8" name="__codelineno-3-8"></a># SN, Summary numbers:
<a href="#__codelineno-3-9" id="__codelineno-3-9" name="__codelineno-3-9"></a>#   number of records   .. number of data rows in the VCF
<a href="#__codelineno-3-10" id="__codelineno-3-10" name="__codelineno-3-10"></a>#   number of no-ALTs   .. reference-only sites, ALT is either "." or identical to REF
<a href="#__codelineno-3-11" id="__codelineno-3-11" name="__codelineno-3-11"></a>#   number of SNPs      .. number of rows with a SNP
<a href="#__codelineno-3-12" id="__codelineno-3-12" name="__codelineno-3-12"></a>#   number of MNPs      .. number of rows with a MNP, such as CC&gt;TT
<a href="#__codelineno-3-13" id="__codelineno-3-13" name="__codelineno-3-13"></a>#   number of indels    .. number of rows with an indel
<a href="#__codelineno-3-14" id="__codelineno-3-14" name="__codelineno-3-14"></a>#   number of others    .. number of rows with other type, for example a symbolic allele or
<a href="#__codelineno-3-15" id="__codelineno-3-15" name="__codelineno-3-15"></a>#                          a complex substitution, such as ACT&gt;TCGA
<a href="#__codelineno-3-16" id="__codelineno-3-16" name="__codelineno-3-16"></a>#   number of multiallelic sites     .. number of rows with multiple alternate alleles
<a href="#__codelineno-3-17" id="__codelineno-3-17" name="__codelineno-3-17"></a>#   number of multiallelic SNP sites .. number of rows with multiple alternate alleles, all SNPs
<a href="#__codelineno-3-18" id="__codelineno-3-18" name="__codelineno-3-18"></a># 
<a href="#__codelineno-3-19" id="__codelineno-3-19" name="__codelineno-3-19"></a>#   Note that rows containing multiple types will be counted multiple times, in each
<a href="#__codelineno-3-20" id="__codelineno-3-20" name="__codelineno-3-20"></a>#   counter. For example, a row with a SNP and an indel increments both the SNP and
<a href="#__codelineno-3-21" id="__codelineno-3-21" name="__codelineno-3-21"></a>#   the indel counter.
<a href="#__codelineno-3-22" id="__codelineno-3-22" name="__codelineno-3-22"></a># 
<a href="#__codelineno-3-23" id="__codelineno-3-23" name="__codelineno-3-23"></a># SN    [2]id   [3]key  [4]value
<a href="#__codelineno-3-24" id="__codelineno-3-24" name="__codelineno-3-24"></a>SN      0       number of samples:      0
<a href="#__codelineno-3-25" id="__codelineno-3-25" name="__codelineno-3-25"></a>SN      0       number of records:      4667
<a href="#__codelineno-3-26" id="__codelineno-3-26" name="__codelineno-3-26"></a>SN      0       number of no-ALTs:      0
<a href="#__codelineno-3-27" id="__codelineno-3-27" name="__codelineno-3-27"></a>SN      0       number of SNPs: 4667
<a href="#__codelineno-3-28" id="__codelineno-3-28" name="__codelineno-3-28"></a>SN      0       number of MNPs: 0
<a href="#__codelineno-3-29" id="__codelineno-3-29" name="__codelineno-3-29"></a>SN      0       number of indels:       0
<a href="#__codelineno-3-30" id="__codelineno-3-30" name="__codelineno-3-30"></a>SN      0       number of others:       0
<a href="#__codelineno-3-31" id="__codelineno-3-31" name="__codelineno-3-31"></a>SN      0       number of multiallelic sites:   0
<a href="#__codelineno-3-32" id="__codelineno-3-32" name="__codelineno-3-32"></a>SN      0       number of multiallelic SNP sites:       0
<a href="#__codelineno-3-33" id="__codelineno-3-33" name="__codelineno-3-33"></a># TSTV, transitions/transversions:
<a href="#__codelineno-3-34" id="__codelineno-3-34" name="__codelineno-3-34"></a># TSTV  [2]id   [3]ts   [4]tv   [5]ts/tv        [6]ts (1st ALT) [7]tv (1st ALT) [8]ts/tv (1st ALT)
<a href="#__codelineno-3-35" id="__codelineno-3-35" name="__codelineno-3-35"></a>TSTV    0       1550    3117    0.50    1550    3117    0.50
<a href="#__codelineno-3-36" id="__codelineno-3-36" name="__codelineno-3-36"></a># SiS, Singleton stats:
<a href="#__codelineno-3-37" id="__codelineno-3-37" name="__codelineno-3-37"></a># SiS   [2]id   [3]allele count [4]number of SNPs       [5]number of transitions        [6]number of transversions      [7]number of indels     [8]repeat-consistent    [9]repeat-inconsistent        [10]not applicable
<a href="#__codelineno-3-38" id="__codelineno-3-38" name="__codelineno-3-38"></a>SiS     0       1       4667    1550    3117    0       0       0       0
<a href="#__codelineno-3-39" id="__codelineno-3-39" name="__codelineno-3-39"></a># AF, Stats by non-reference allele frequency:
<a href="#__codelineno-3-40" id="__codelineno-3-40" name="__codelineno-3-40"></a># AF    [2]id   [3]allele frequency     [4]number of SNPs       [5]number of transitions        [6]number of transversions      [7]number of indels     [8]repeat-consistent [9]repeat-inconsistent   [10]not applicable
<a href="#__codelineno-3-41" id="__codelineno-3-41" name="__codelineno-3-41"></a>AF      0       0.000000        4667    1550    3117    0       0       0       0
</code></pre></div>
<p>When simulating multiple types of variants using simuG tool it creates multiple VCF files as well. We need to merge them and create one VCF file before comapring. For an example for the simulated sample NC_017518.1_SNP_4000_INDEL_4000, </p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-4-1" id="__codelineno-4-1" name="__codelineno-4-1"></a>$ ls -1hs Simulation_SNP_4000_INDEL_4000.*
<a href="#__codelineno-4-2" id="__codelineno-4-2" name="__codelineno-4-2"></a>768K Simulation_SNP_4000_INDEL_4000.refseq2simseq.INDEL.vcf
<a href="#__codelineno-4-3" id="__codelineno-4-3" name="__codelineno-4-3"></a>768K Simulation_SNP_4000_INDEL_4000.refseq2simseq.map.txt
<a href="#__codelineno-4-4" id="__codelineno-4-4" name="__codelineno-4-4"></a>768K Simulation_SNP_4000_INDEL_4000.refseq2simseq.SNP.vcf
<a href="#__codelineno-4-5" id="__codelineno-4-5" name="__codelineno-4-5"></a>2.3M Simulation_SNP_4000_INDEL_4000.simseq.genome.fa
</code></pre></div>
<p>zip the files and making indexes. </p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-5-1" id="__codelineno-5-1" name="__codelineno-5-1"></a>$ bgzip Simulation_SNP_4000_INDEL_4000.refseq2simseq.INDEL.vcf
<a href="#__codelineno-5-2" id="__codelineno-5-2" name="__codelineno-5-2"></a>$ bgzip Simulation_SNP_4000_INDEL_4000.refseq2simseq.SNP.vcf
<a href="#__codelineno-5-3" id="__codelineno-5-3" name="__codelineno-5-3"></a>$ bcftools index Simulation_SNP_4000_INDEL_4000.refseq2simseq.SNP.vcf.gz
<a href="#__codelineno-5-4" id="__codelineno-5-4" name="__codelineno-5-4"></a>$ bcftools index Simulation_SNP_4000_INDEL_4000.refseq2simseq.INDEL.vcf.gz
</code></pre></div>
<p>Merge 2 vcf files usnimng <code>bcftools</code></p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-6-1" id="__codelineno-6-1" name="__codelineno-6-1"></a>bcftools merge Simulation_SNP_4000_INDEL_4000.refseq2simseq.SNP.vcf.gz Simulation_SNP_4000_INDEL_4000.refseq2simseq.INDEL.vcf.gz -O z -o Simulation_SNP_4000_INDEL_4000.vcf.gz
</code></pre></div>
<p>Stats of the merged vcf file should show 4000 SNPs and 4000 INDELs.</p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-7-1" id="__codelineno-7-1" name="__codelineno-7-1"></a>$ bcftools stats Simulation_SNP_4000_INDEL_4000.vcf.gz | less
<a href="#__codelineno-7-2" id="__codelineno-7-2" name="__codelineno-7-2"></a># This file was produced by bcftools stats (1.9+htslib-1.9) and can be plotted using plot-vcfstats.
<a href="#__codelineno-7-3" id="__codelineno-7-3" name="__codelineno-7-3"></a># The command line was: bcftools stats  Simulation_SNP_4000_INDEL_4000.vcf.gz
<a href="#__codelineno-7-4" id="__codelineno-7-4" name="__codelineno-7-4"></a>#
<a href="#__codelineno-7-5" id="__codelineno-7-5" name="__codelineno-7-5"></a># Definition of sets:
<a href="#__codelineno-7-6" id="__codelineno-7-6" name="__codelineno-7-6"></a># ID    [2]id   [3]tab-separated file names
<a href="#__codelineno-7-7" id="__codelineno-7-7" name="__codelineno-7-7"></a>ID      0       Simulation_SNP_4000_INDEL_4000.vcf.gz
<a href="#__codelineno-7-8" id="__codelineno-7-8" name="__codelineno-7-8"></a># SN, Summary numbers:
<a href="#__codelineno-7-9" id="__codelineno-7-9" name="__codelineno-7-9"></a>#   number of records   .. number of data rows in the VCF
<a href="#__codelineno-7-10" id="__codelineno-7-10" name="__codelineno-7-10"></a>#   number of no-ALTs   .. reference-only sites, ALT is either "." or identical to REF
<a href="#__codelineno-7-11" id="__codelineno-7-11" name="__codelineno-7-11"></a>#   number of SNPs      .. number of rows with a SNP
<a href="#__codelineno-7-12" id="__codelineno-7-12" name="__codelineno-7-12"></a>#   number of MNPs      .. number of rows with a MNP, such as CC&gt;TT
<a href="#__codelineno-7-13" id="__codelineno-7-13" name="__codelineno-7-13"></a>#   number of indels    .. number of rows with an indel
<a href="#__codelineno-7-14" id="__codelineno-7-14" name="__codelineno-7-14"></a>#   number of others    .. number of rows with other type, for example a symbolic allele or
<a href="#__codelineno-7-15" id="__codelineno-7-15" name="__codelineno-7-15"></a>#                          a complex substitution, such as ACT&gt;TCGA
<a href="#__codelineno-7-16" id="__codelineno-7-16" name="__codelineno-7-16"></a>#   number of multiallelic sites     .. number of rows with multiple alternate alleles
<a href="#__codelineno-7-17" id="__codelineno-7-17" name="__codelineno-7-17"></a>#   number of multiallelic SNP sites .. number of rows with multiple alternate alleles, all SNPs
<a href="#__codelineno-7-18" id="__codelineno-7-18" name="__codelineno-7-18"></a># 
<a href="#__codelineno-7-19" id="__codelineno-7-19" name="__codelineno-7-19"></a>#   Note that rows containing multiple types will be counted multiple times, in each
<a href="#__codelineno-7-20" id="__codelineno-7-20" name="__codelineno-7-20"></a>#   counter. For example, a row with a SNP and an indel increments both the SNP and
<a href="#__codelineno-7-21" id="__codelineno-7-21" name="__codelineno-7-21"></a>#   the indel counter.
<a href="#__codelineno-7-22" id="__codelineno-7-22" name="__codelineno-7-22"></a># 
<a href="#__codelineno-7-23" id="__codelineno-7-23" name="__codelineno-7-23"></a># SN    [2]id   [3]key  [4]value
<a href="#__codelineno-7-24" id="__codelineno-7-24" name="__codelineno-7-24"></a>SN      0       number of samples:      0
<a href="#__codelineno-7-25" id="__codelineno-7-25" name="__codelineno-7-25"></a>SN      0       number of records:      8000
<a href="#__codelineno-7-26" id="__codelineno-7-26" name="__codelineno-7-26"></a>SN      0       number of no-ALTs:      0
<a href="#__codelineno-7-27" id="__codelineno-7-27" name="__codelineno-7-27"></a>SN      0       number of SNPs: 4000
<a href="#__codelineno-7-28" id="__codelineno-7-28" name="__codelineno-7-28"></a>SN      0       number of MNPs: 0
<a href="#__codelineno-7-29" id="__codelineno-7-29" name="__codelineno-7-29"></a>SN      0       number of indels:       4000
<a href="#__codelineno-7-30" id="__codelineno-7-30" name="__codelineno-7-30"></a>SN      0       number of others:       0
<a href="#__codelineno-7-31" id="__codelineno-7-31" name="__codelineno-7-31"></a>SN      0       number of multiallelic sites:   0
<a href="#__codelineno-7-32" id="__codelineno-7-32" name="__codelineno-7-32"></a>SN      0       number of multiallelic SNP sites:       0
<a href="#__codelineno-7-33" id="__codelineno-7-33" name="__codelineno-7-33"></a># TSTV, transitions/transversions:
<a href="#__codelineno-7-34" id="__codelineno-7-34" name="__codelineno-7-34"></a># TSTV  [2]id   [3]ts   [4]tv   [5]ts/tv        [6]ts (1st ALT) [7]tv (1st ALT) [8]ts/tv (1st ALT)
<a href="#__codelineno-7-35" id="__codelineno-7-35" name="__codelineno-7-35"></a>TSTV    0       1372    2628    0.52    1372    2628    0.52
<a href="#__codelineno-7-36" id="__codelineno-7-36" name="__codelineno-7-36"></a># SiS, Singleton stats:
<a href="#__codelineno-7-37" id="__codelineno-7-37" name="__codelineno-7-37"></a># SiS   [2]id   [3]allele count [4]number of SNPs       [5]number of transitions        [6]number of transversions      [7]number of indels     [8]repeat-consistent    [9]repeat-inconsistent  [10]not applicable
<a href="#__codelineno-7-38" id="__codelineno-7-38" name="__codelineno-7-38"></a>SiS     0       1       4000    1372    2628    4000    0       0       4000
</code></pre></div>
<p>Now  we can apply the same procedure for comparision using <code>bcftools isec</code>.</p>
<!---

### Ground Truth vs Graph based variant comparision 
As explain in the main page we have applied some filtering to exatract the PGGB VCF and then compare to the Ground Truth VCF using bcftool isec command.

<div class="highlight"><pre><span></span><code><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a><span class="c1">#for 95</span>
<a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a>bcftools<span class="w"> </span>view<span class="w"> </span>-s<span class="w"> </span>NC_017518.1_SNP_5000<span class="w"> </span>--min-ac<span class="o">=</span><span class="m">1</span><span class="w"> </span>pggb_1k95_NC_017518.1.vcf<span class="w">  </span>&gt;<span class="w">  </span>Simulation_SNP_5000.graph95.vcf<span class="w"> </span>
<a id="__codelineno-8-3" name="__codelineno-8-3" href="#__codelineno-8-3"></a>bgzip<span class="w"> </span>Simulation_SNP_5000.graph95.vcf
<a id="__codelineno-8-4" name="__codelineno-8-4" href="#__codelineno-8-4"></a>bcftools<span class="w"> </span>index<span class="w"> </span>Simulation_SNP_5000.graph95.vcf.gz<span class="w"> </span>
<a id="__codelineno-8-5" name="__codelineno-8-5" href="#__codelineno-8-5"></a>bcftools<span class="w"> </span>isec<span class="w"> </span>-c<span class="w"> </span>none<span class="w"> </span>-p<span class="w"> </span>output_SNP_5000_graph95<span class="w"> </span>Simulation_SNP_5000.refseq2simseq.SNP.vcf.gz<span class="w"> </span>Simulation_SNP_5000.graph95.vcf.gz
<a id="__codelineno-8-6" name="__codelineno-8-6" href="#__codelineno-8-6"></a>
<a id="__codelineno-8-7" name="__codelineno-8-7" href="#__codelineno-8-7"></a>$<span class="w"> </span><span class="nb">cd</span><span class="w"> </span>output_SNP_5000_graph95
<a id="__codelineno-8-8" name="__codelineno-8-8" href="#__codelineno-8-8"></a>
<a id="__codelineno-8-9" name="__codelineno-8-9" href="#__codelineno-8-9"></a>$<span class="w"> </span>ls<span class="w"> </span>-1hs<span class="w"> </span>
<a id="__codelineno-8-10" name="__codelineno-8-10" href="#__codelineno-8-10"></a>total<span class="w"> </span><span class="m">5</span>.1M
<a id="__codelineno-8-11" name="__codelineno-8-11" href="#__codelineno-8-11"></a>256K<span class="w"> </span><span class="m">0000</span>.vcf
<a id="__codelineno-8-12" name="__codelineno-8-12" href="#__codelineno-8-12"></a><span class="m">3</span>.3M<span class="w"> </span><span class="m">0001</span>.vcf
<a id="__codelineno-8-13" name="__codelineno-8-13" href="#__codelineno-8-13"></a>768K<span class="w"> </span><span class="m">0002</span>.vcf
<a id="__codelineno-8-14" name="__codelineno-8-14" href="#__codelineno-8-14"></a>768K<span class="w"> </span><span class="m">0003</span>.vcf
<a id="__codelineno-8-15" name="__codelineno-8-15" href="#__codelineno-8-15"></a><span class="w"> </span><span class="m">512</span><span class="w"> </span>README.txt
<a id="__codelineno-8-16" name="__codelineno-8-16" href="#__codelineno-8-16"></a>
<a id="__codelineno-8-17" name="__codelineno-8-17" href="#__codelineno-8-17"></a>$<span class="w"> </span>cat<span class="w"> </span>README.txt<span class="w"> </span>
<a id="__codelineno-8-18" name="__codelineno-8-18" href="#__codelineno-8-18"></a>This<span class="w"> </span>file<span class="w"> </span>was<span class="w"> </span>produced<span class="w"> </span>by<span class="w"> </span>vcfisec.
<a id="__codelineno-8-19" name="__codelineno-8-19" href="#__codelineno-8-19"></a>The<span class="w"> </span><span class="nb">command</span><span class="w"> </span>line<span class="w"> </span>was:<span class="w">   </span>bcftools<span class="w"> </span>isec<span class="w">  </span>-c<span class="w"> </span>none<span class="w"> </span>-p<span class="w"> </span>output_SNP_5000_graph95<span class="w"> </span>Simulation_SNP_5000.refseq2simseq.SNP.vcf.gz<span class="w"> </span>Simulation_SNP_5000.graph95.vcf.gz
<a id="__codelineno-8-20" name="__codelineno-8-20" href="#__codelineno-8-20"></a>
<a id="__codelineno-8-21" name="__codelineno-8-21" href="#__codelineno-8-21"></a>Using<span class="w"> </span>the<span class="w"> </span>following<span class="w"> </span>file<span class="w"> </span>names:
<a id="__codelineno-8-22" name="__codelineno-8-22" href="#__codelineno-8-22"></a>output_SNP_5000_graph95/0000.vcf<span class="w">        </span><span class="k">for</span><span class="w"> </span>records<span class="w"> </span>private<span class="w"> </span>to<span class="w">  </span>Simulation_SNP_5000.refseq2simseq.SNP.vcf.gz
<a id="__codelineno-8-23" name="__codelineno-8-23" href="#__codelineno-8-23"></a>output_SNP_5000_graph95/0001.vcf<span class="w">        </span><span class="k">for</span><span class="w"> </span>records<span class="w"> </span>private<span class="w"> </span>to<span class="w">  </span>Simulation_SNP_5000.graph95.vcf.gz
<a id="__codelineno-8-24" name="__codelineno-8-24" href="#__codelineno-8-24"></a>output_SNP_5000_graph95/0002.vcf<span class="w">        </span><span class="k">for</span><span class="w"> </span>records<span class="w"> </span>from<span class="w"> </span>Simulation_SNP_5000.refseq2simseq.SNP.vcf.gz<span class="w"> </span>shared<span class="w"> </span>by<span class="w"> </span>both<span class="w">    </span>Simulation_SNP_5000.refseq2simseq.SNP.vcf.gz<span class="w"> </span>Simulation_SNP_5000.graph95.vcf.gz
<a id="__codelineno-8-25" name="__codelineno-8-25" href="#__codelineno-8-25"></a>output_SNP_5000_graph95/0003.vcf<span class="w">        </span><span class="k">for</span><span class="w"> </span>records<span class="w"> </span>from<span class="w"> </span>Simulation_SNP_5000.graph95.vcf.gz<span class="w"> </span>shared<span class="w"> </span>by<span class="w"> </span>both<span class="w">      </span>Simulation_SNP_5000.refseq2simseq.SNP.vcf.gz<span class="w"> </span>Simulation_SNP_5000.graph95.vcf.gz
<a id="__codelineno-8-26" name="__codelineno-8-26" href="#__codelineno-8-26"></a>
<a id="__codelineno-8-27" name="__codelineno-8-27" href="#__codelineno-8-27"></a>$<span class="w"> </span>bcftools<span class="w"> </span>stats<span class="w"> </span><span class="m">0002</span>.vcf<span class="w"> </span>
<a id="__codelineno-8-28" name="__codelineno-8-28" href="#__codelineno-8-28"></a><span class="c1"># This file was produced by bcftools stats (1.9+htslib-1.9) and can be plotted using plot-vcfstats.</span>
<a id="__codelineno-8-29" name="__codelineno-8-29" href="#__codelineno-8-29"></a><span class="c1"># The command line was: bcftools stats  0002.vcf</span>
<a id="__codelineno-8-30" name="__codelineno-8-30" href="#__codelineno-8-30"></a><span class="c1">#</span>
<a id="__codelineno-8-31" name="__codelineno-8-31" href="#__codelineno-8-31"></a><span class="c1"># Definition of sets:</span>
<a id="__codelineno-8-32" name="__codelineno-8-32" href="#__codelineno-8-32"></a><span class="c1"># ID    [2]id   [3]tab-separated file names</span>
<a id="__codelineno-8-33" name="__codelineno-8-33" href="#__codelineno-8-33"></a>ID<span class="w">      </span><span class="m">0</span><span class="w">       </span><span class="m">0002</span>.vcf
<a id="__codelineno-8-34" name="__codelineno-8-34" href="#__codelineno-8-34"></a><span class="c1"># SN, Summary numbers:</span>
<a id="__codelineno-8-35" name="__codelineno-8-35" href="#__codelineno-8-35"></a><span class="c1">#   number of records   .. number of data rows in the VCF</span>
<a id="__codelineno-8-36" name="__codelineno-8-36" href="#__codelineno-8-36"></a><span class="c1">#   number of no-ALTs   .. reference-only sites, ALT is either &quot;.&quot; or identical to REF</span>
<a id="__codelineno-8-37" name="__codelineno-8-37" href="#__codelineno-8-37"></a><span class="c1">#   number of SNPs      .. number of rows with a SNP</span>
<a id="__codelineno-8-38" name="__codelineno-8-38" href="#__codelineno-8-38"></a><span class="c1">#   number of MNPs      .. number of rows with a MNP, such as CC&gt;TT</span>
<a id="__codelineno-8-39" name="__codelineno-8-39" href="#__codelineno-8-39"></a><span class="c1">#   number of indels    .. number of rows with an indel</span>
<a id="__codelineno-8-40" name="__codelineno-8-40" href="#__codelineno-8-40"></a><span class="c1">#   number of others    .. number of rows with other type, for example a symbolic allele or</span>
<a id="__codelineno-8-41" name="__codelineno-8-41" href="#__codelineno-8-41"></a><span class="c1">#                          a complex substitution, such as ACT&gt;TCGA</span>
<a id="__codelineno-8-42" name="__codelineno-8-42" href="#__codelineno-8-42"></a><span class="c1">#   number of multiallelic sites     .. number of rows with multiple alternate alleles</span>
<a id="__codelineno-8-43" name="__codelineno-8-43" href="#__codelineno-8-43"></a><span class="c1">#   number of multiallelic SNP sites .. number of rows with multiple alternate alleles, all SNPs</span>
<a id="__codelineno-8-44" name="__codelineno-8-44" href="#__codelineno-8-44"></a><span class="c1"># </span>
<a id="__codelineno-8-45" name="__codelineno-8-45" href="#__codelineno-8-45"></a><span class="c1">#   Note that rows containing multiple types will be counted multiple times, in each</span>
<a id="__codelineno-8-46" name="__codelineno-8-46" href="#__codelineno-8-46"></a><span class="c1">#   counter. For example, a row with a SNP and an indel increments both the SNP and</span>
<a id="__codelineno-8-47" name="__codelineno-8-47" href="#__codelineno-8-47"></a><span class="c1">#   the indel counter.</span>
<a id="__codelineno-8-48" name="__codelineno-8-48" href="#__codelineno-8-48"></a><span class="c1"># </span>
<a id="__codelineno-8-49" name="__codelineno-8-49" href="#__codelineno-8-49"></a><span class="c1"># SN    [2]id   [3]key  [4]value</span>
<a id="__codelineno-8-50" name="__codelineno-8-50" href="#__codelineno-8-50"></a>SN<span class="w">      </span><span class="m">0</span><span class="w">       </span>number<span class="w"> </span>of<span class="w"> </span>samples:<span class="w">      </span><span class="m">0</span>
<a id="__codelineno-8-51" name="__codelineno-8-51" href="#__codelineno-8-51"></a>SN<span class="w">      </span><span class="m">0</span><span class="w">       </span>number<span class="w"> </span>of<span class="w"> </span>records:<span class="w">      </span><span class="m">4838</span>
<a id="__codelineno-8-52" name="__codelineno-8-52" href="#__codelineno-8-52"></a>SN<span class="w">      </span><span class="m">0</span><span class="w">       </span>number<span class="w"> </span>of<span class="w"> </span>no-ALTs:<span class="w">      </span><span class="m">0</span>
<a id="__codelineno-8-53" name="__codelineno-8-53" href="#__codelineno-8-53"></a>SN<span class="w">      </span><span class="m">0</span><span class="w">       </span>number<span class="w"> </span>of<span class="w"> </span>SNPs:<span class="w"> </span><span class="m">4838</span>
<a id="__codelineno-8-54" name="__codelineno-8-54" href="#__codelineno-8-54"></a>SN<span class="w">      </span><span class="m">0</span><span class="w">       </span>number<span class="w"> </span>of<span class="w"> </span>MNPs:<span class="w"> </span><span class="m">0</span>
<a id="__codelineno-8-55" name="__codelineno-8-55" href="#__codelineno-8-55"></a>SN<span class="w">      </span><span class="m">0</span><span class="w">       </span>number<span class="w"> </span>of<span class="w"> </span>indels:<span class="w">       </span><span class="m">0</span>
<a id="__codelineno-8-56" name="__codelineno-8-56" href="#__codelineno-8-56"></a>SN<span class="w">      </span><span class="m">0</span><span class="w">       </span>number<span class="w"> </span>of<span class="w"> </span>others:<span class="w">       </span><span class="m">0</span>
<a id="__codelineno-8-57" name="__codelineno-8-57" href="#__codelineno-8-57"></a>SN<span class="w">      </span><span class="m">0</span><span class="w">       </span>number<span class="w"> </span>of<span class="w"> </span>multiallelic<span class="w"> </span>sites:<span class="w">   </span><span class="m">0</span>
<a id="__codelineno-8-58" name="__codelineno-8-58" href="#__codelineno-8-58"></a>SN<span class="w">      </span><span class="m">0</span><span class="w">       </span>number<span class="w"> </span>of<span class="w"> </span>multiallelic<span class="w"> </span>SNP<span class="w"> </span>sites:<span class="w">       </span><span class="m">0</span>
<a id="__codelineno-8-59" name="__codelineno-8-59" href="#__codelineno-8-59"></a><span class="c1"># TSTV, transitions/transversions:</span>
<a id="__codelineno-8-60" name="__codelineno-8-60" href="#__codelineno-8-60"></a><span class="c1"># TSTV  [2]id   [3]ts   [4]tv   [5]ts/tv        [6]ts (1st ALT) [7]tv (1st ALT) [8]ts/tv (1st ALT)</span>
<a id="__codelineno-8-61" name="__codelineno-8-61" href="#__codelineno-8-61"></a>TSTV<span class="w">    </span><span class="m">0</span><span class="w">       </span><span class="m">1601</span><span class="w">    </span><span class="m">3237</span><span class="w">    </span><span class="m">0</span>.49<span class="w">    </span><span class="m">1601</span><span class="w">    </span><span class="m">3237</span><span class="w">    </span><span class="m">0</span>.49
<a id="__codelineno-8-62" name="__codelineno-8-62" href="#__codelineno-8-62"></a><span class="c1"># SiS, Singleton stats:</span>
<a id="__codelineno-8-63" name="__codelineno-8-63" href="#__codelineno-8-63"></a><span class="c1"># SiS   [2]id   [3]allele count [4]number of SNPs       [5]number of transitions        [6]number of transversions      [7]number of indels     [8]repeat-consistent    [9]repeat-inconsistent [10]not applicable</span>
<a id="__codelineno-8-64" name="__codelineno-8-64" href="#__codelineno-8-64"></a>SiS<span class="w">     </span><span class="m">0</span><span class="w">       </span><span class="m">1</span><span class="w">       </span><span class="m">4838</span><span class="w">    </span><span class="m">1601</span><span class="w">    </span><span class="m">3237</span><span class="w">    </span><span class="m">0</span><span class="w">       </span><span class="m">0</span><span class="w">       </span><span class="m">0</span><span class="w">       </span><span class="m">0</span>
<a id="__codelineno-8-65" name="__codelineno-8-65" href="#__codelineno-8-65"></a><span class="c1"># AF, Stats by non-reference allele frequency:</span>
<a id="__codelineno-8-66" name="__codelineno-8-66" href="#__codelineno-8-66"></a><span class="c1"># AF    [2]id   [3]allele frequency     [4]number of SNPs       [5]number of transitions        [6]number of transversions      [7]number of indels     [8]repeat-consistent    [9]repeat-inconsistent [10]not applicable</span>
<a id="__codelineno-8-67" name="__codelineno-8-67" href="#__codelineno-8-67"></a>AF<span class="w">      </span><span class="m">0</span><span class="w">       </span><span class="m">0</span>.000000<span class="w">        </span><span class="m">4838</span><span class="w">    </span><span class="m">1601</span><span class="w">    </span><span class="m">3237</span><span class="w">    </span><span class="m">0</span><span class="w">       </span><span class="m">0</span><span class="w">       </span><span class="m">0</span><span class="w">       </span><span class="m">0</span>
<a id="__codelineno-8-68" name="__codelineno-8-68" href="#__codelineno-8-68"></a><span class="c1"># QUAL, Stats by quality:</span>
<a id="__codelineno-8-69" name="__codelineno-8-69" href="#__codelineno-8-69"></a><span class="c1"># QUAL  [2]id   [3]Quality      [4]number of SNPs       [5]number of transitions (1st ALT)      [6]number of transversions (1st ALT)    [7]number of indels</span>
<a id="__codelineno-8-70" name="__codelineno-8-70" href="#__codelineno-8-70"></a>QUAL<span class="w">    </span><span class="m">0</span><span class="w">       </span><span class="m">998</span><span class="w">     </span><span class="m">4838</span><span class="w">    </span><span class="m">1601</span><span class="w">    </span><span class="m">3237</span><span class="w">    </span><span class="m">0</span>
<a id="__codelineno-8-71" name="__codelineno-8-71" href="#__codelineno-8-71"></a><span class="c1"># IDD, InDel distribution:</span>
<a id="__codelineno-8-72" name="__codelineno-8-72" href="#__codelineno-8-72"></a><span class="c1"># IDD   [2]id   [3]length (deletions negative)  [4]count</span>
<a id="__codelineno-8-73" name="__codelineno-8-73" href="#__codelineno-8-73"></a><span class="c1"># ST, Substitution types:</span>
<a id="__codelineno-8-74" name="__codelineno-8-74" href="#__codelineno-8-74"></a><span class="c1"># ST    [2]id   [3]type [4]count</span>
<a id="__codelineno-8-75" name="__codelineno-8-75" href="#__codelineno-8-75"></a>ST<span class="w">      </span><span class="m">0</span><span class="w">       </span>A&gt;C<span class="w">     </span><span class="m">391</span>
<a id="__codelineno-8-76" name="__codelineno-8-76" href="#__codelineno-8-76"></a>ST<span class="w">      </span><span class="m">0</span><span class="w">       </span>A&gt;G<span class="w">     </span><span class="m">406</span>
<a id="__codelineno-8-77" name="__codelineno-8-77" href="#__codelineno-8-77"></a>ST<span class="w">      </span><span class="m">0</span><span class="w">       </span>A&gt;T<span class="w">     </span><span class="m">420</span>
<a id="__codelineno-8-78" name="__codelineno-8-78" href="#__codelineno-8-78"></a>ST<span class="w">      </span><span class="m">0</span><span class="w">       </span>C&gt;A<span class="w">     </span><span class="m">410</span>
<a id="__codelineno-8-79" name="__codelineno-8-79" href="#__codelineno-8-79"></a>ST<span class="w">      </span><span class="m">0</span><span class="w">       </span>C&gt;G<span class="w">     </span><span class="m">429</span>
<a id="__codelineno-8-80" name="__codelineno-8-80" href="#__codelineno-8-80"></a>ST<span class="w">      </span><span class="m">0</span><span class="w">       </span>C&gt;T<span class="w">     </span><span class="m">396</span>
<a id="__codelineno-8-81" name="__codelineno-8-81" href="#__codelineno-8-81"></a>ST<span class="w">      </span><span class="m">0</span><span class="w">       </span>G&gt;A<span class="w">     </span><span class="m">380</span>
<a id="__codelineno-8-82" name="__codelineno-8-82" href="#__codelineno-8-82"></a>ST<span class="w">      </span><span class="m">0</span><span class="w">       </span>G&gt;C<span class="w">     </span><span class="m">409</span>
<a id="__codelineno-8-83" name="__codelineno-8-83" href="#__codelineno-8-83"></a>ST<span class="w">      </span><span class="m">0</span><span class="w">       </span>G&gt;T<span class="w">     </span><span class="m">391</span>
<a id="__codelineno-8-84" name="__codelineno-8-84" href="#__codelineno-8-84"></a>ST<span class="w">      </span><span class="m">0</span><span class="w">       </span>T&gt;A<span class="w">     </span><span class="m">392</span>
<a id="__codelineno-8-85" name="__codelineno-8-85" href="#__codelineno-8-85"></a>ST<span class="w">      </span><span class="m">0</span><span class="w">       </span>T&gt;C<span class="w">     </span><span class="m">419</span>
<a id="__codelineno-8-86" name="__codelineno-8-86" href="#__codelineno-8-86"></a>ST<span class="w">      </span><span class="m">0</span><span class="w">       </span>T&gt;G<span class="w">     </span><span class="m">395</span>
<a id="__codelineno-8-87" name="__codelineno-8-87" href="#__codelineno-8-87"></a><span class="c1"># DP, Depth distribution</span>
<a id="__codelineno-8-88" name="__codelineno-8-88" href="#__codelineno-8-88"></a><span class="c1"># DP    [2]id   [3]bin  [4]number of genotypes  [5]fraction of genotypes (%)    [6]number of sites      [7]fraction of sites (%)</span>
</code></pre></div>
Using the same procedure to follow PGGB 98 files.
<div class="highlight"><pre><span></span><code><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a>#for 98
<a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a>bcftools view -s NC_017518.1_SNP_5000 --min-ac=1 pggb_1k98_NC_017518.1.vcf  &gt;  Simulation_SNP_5000.graph98.vcf
<a id="__codelineno-9-3" name="__codelineno-9-3" href="#__codelineno-9-3"></a>bgzip Simulation_SNP_5000.graph98.vcf
<a id="__codelineno-9-4" name="__codelineno-9-4" href="#__codelineno-9-4"></a>bcftools index Simulation_SNP_5000.graph98.vcf.gz 
<a id="__codelineno-9-5" name="__codelineno-9-5" href="#__codelineno-9-5"></a>bcftools isec -c none -p output_SNP_5000_graph98 Simulation_SNP_5000.refseq2simseq.SNP.vcf.gz Simulation_SNP_5000.graph98.vcf.gz
<a id="__codelineno-9-6" name="__codelineno-9-6" href="#__codelineno-9-6"></a>
<a id="__codelineno-9-7" name="__codelineno-9-7" href="#__codelineno-9-7"></a>$ cd output_SNP_5000_graph98
<a id="__codelineno-9-8" name="__codelineno-9-8" href="#__codelineno-9-8"></a>$ ls -1hs
<a id="__codelineno-9-9" name="__codelineno-9-9" href="#__codelineno-9-9"></a>total 5.1M
<a id="__codelineno-9-10" name="__codelineno-9-10" href="#__codelineno-9-10"></a>256K 0000.vcf
<a id="__codelineno-9-11" name="__codelineno-9-11" href="#__codelineno-9-11"></a>3.3M 0001.vcf
<a id="__codelineno-9-12" name="__codelineno-9-12" href="#__codelineno-9-12"></a>768K 0002.vcf
<a id="__codelineno-9-13" name="__codelineno-9-13" href="#__codelineno-9-13"></a>768K 0003.vcf
<a id="__codelineno-9-14" name="__codelineno-9-14" href="#__codelineno-9-14"></a> 512 README.txt
<a id="__codelineno-9-15" name="__codelineno-9-15" href="#__codelineno-9-15"></a>
<a id="__codelineno-9-16" name="__codelineno-9-16" href="#__codelineno-9-16"></a>$ cat README.txt 
<a id="__codelineno-9-17" name="__codelineno-9-17" href="#__codelineno-9-17"></a>This file was produced by vcfisec.
<a id="__codelineno-9-18" name="__codelineno-9-18" href="#__codelineno-9-18"></a>The command line was:   bcftools isec  -c none -p output_SNP_5000_graph98 Simulation_SNP_5000.refseq2simseq.SNP.vcf.gz Simulation_SNP_5000.graph98.vcf.gz
<a id="__codelineno-9-19" name="__codelineno-9-19" href="#__codelineno-9-19"></a>
<a id="__codelineno-9-20" name="__codelineno-9-20" href="#__codelineno-9-20"></a>Using the following file names:
<a id="__codelineno-9-21" name="__codelineno-9-21" href="#__codelineno-9-21"></a>output_SNP_5000_graph98/0000.vcf        for records private to  Simulation_SNP_5000.refseq2simseq.SNP.vcf.gz
<a id="__codelineno-9-22" name="__codelineno-9-22" href="#__codelineno-9-22"></a>output_SNP_5000_graph98/0001.vcf        for records private to  Simulation_SNP_5000.graph98.vcf.gz
<a id="__codelineno-9-23" name="__codelineno-9-23" href="#__codelineno-9-23"></a>output_SNP_5000_graph98/0002.vcf        for records from Simulation_SNP_5000.refseq2simseq.SNP.vcf.gz shared by both    Simulation_SNP_5000.refseq2simseq.SNP.vcf.gz Simulation_SNP_5000.graph98.vcf.gz
<a id="__codelineno-9-24" name="__codelineno-9-24" href="#__codelineno-9-24"></a>output_SNP_5000_graph98/0003.vcf        for records from Simulation_SNP_5000.graph98.vcf.gz shared by both      Simulation_SNP_5000.refseq2simseq.SNP.vcf.gz Simulation_SNP_5000.graph98.vcf.gz
<a id="__codelineno-9-25" name="__codelineno-9-25" href="#__codelineno-9-25"></a>
<a id="__codelineno-9-26" name="__codelineno-9-26" href="#__codelineno-9-26"></a>$ bcftools stats 0002.vcf 
<a id="__codelineno-9-27" name="__codelineno-9-27" href="#__codelineno-9-27"></a># This file was produced by bcftools stats (1.9+htslib-1.9) and can be plotted using plot-vcfstats.
<a id="__codelineno-9-28" name="__codelineno-9-28" href="#__codelineno-9-28"></a># The command line was: bcftools stats  0002.vcf
<a id="__codelineno-9-29" name="__codelineno-9-29" href="#__codelineno-9-29"></a>#
<a id="__codelineno-9-30" name="__codelineno-9-30" href="#__codelineno-9-30"></a># Definition of sets:
<a id="__codelineno-9-31" name="__codelineno-9-31" href="#__codelineno-9-31"></a># ID    [2]id   [3]tab-separated file names
<a id="__codelineno-9-32" name="__codelineno-9-32" href="#__codelineno-9-32"></a>ID      0       0002.vcf
<a id="__codelineno-9-33" name="__codelineno-9-33" href="#__codelineno-9-33"></a># SN, Summary numbers:
<a id="__codelineno-9-34" name="__codelineno-9-34" href="#__codelineno-9-34"></a>#   number of records   .. number of data rows in the VCF
<a id="__codelineno-9-35" name="__codelineno-9-35" href="#__codelineno-9-35"></a>#   number of no-ALTs   .. reference-only sites, ALT is either &quot;.&quot; or identical to REF
<a id="__codelineno-9-36" name="__codelineno-9-36" href="#__codelineno-9-36"></a>#   number of SNPs      .. number of rows with a SNP
<a id="__codelineno-9-37" name="__codelineno-9-37" href="#__codelineno-9-37"></a>#   number of MNPs      .. number of rows with a MNP, such as CC&gt;TT
<a id="__codelineno-9-38" name="__codelineno-9-38" href="#__codelineno-9-38"></a>#   number of indels    .. number of rows with an indel
<a id="__codelineno-9-39" name="__codelineno-9-39" href="#__codelineno-9-39"></a>#   number of others    .. number of rows with other type, for example a symbolic allele or
<a id="__codelineno-9-40" name="__codelineno-9-40" href="#__codelineno-9-40"></a>#                          a complex substitution, such as ACT&gt;TCGA
<a id="__codelineno-9-41" name="__codelineno-9-41" href="#__codelineno-9-41"></a>#   number of multiallelic sites     .. number of rows with multiple alternate alleles
<a id="__codelineno-9-42" name="__codelineno-9-42" href="#__codelineno-9-42"></a>#   number of multiallelic SNP sites .. number of rows with multiple alternate alleles, all SNPs
<a id="__codelineno-9-43" name="__codelineno-9-43" href="#__codelineno-9-43"></a># 
<a id="__codelineno-9-44" name="__codelineno-9-44" href="#__codelineno-9-44"></a>#   Note that rows containing multiple types will be counted multiple times, in each
<a id="__codelineno-9-45" name="__codelineno-9-45" href="#__codelineno-9-45"></a>#   counter. For example, a row with a SNP and an indel increments both the SNP and
<a id="__codelineno-9-46" name="__codelineno-9-46" href="#__codelineno-9-46"></a>#   the indel counter.
<a id="__codelineno-9-47" name="__codelineno-9-47" href="#__codelineno-9-47"></a># 
<a id="__codelineno-9-48" name="__codelineno-9-48" href="#__codelineno-9-48"></a># SN    [2]id   [3]key  [4]value
<a id="__codelineno-9-49" name="__codelineno-9-49" href="#__codelineno-9-49"></a>SN      0       number of samples:      0
<a id="__codelineno-9-50" name="__codelineno-9-50" href="#__codelineno-9-50"></a>SN      0       number of records:      4843
<a id="__codelineno-9-51" name="__codelineno-9-51" href="#__codelineno-9-51"></a>SN      0       number of no-ALTs:      0
<a id="__codelineno-9-52" name="__codelineno-9-52" href="#__codelineno-9-52"></a>SN      0       number of SNPs: 4843
<a id="__codelineno-9-53" name="__codelineno-9-53" href="#__codelineno-9-53"></a>SN      0       number of MNPs: 0
<a id="__codelineno-9-54" name="__codelineno-9-54" href="#__codelineno-9-54"></a>SN      0       number of indels:       0
<a id="__codelineno-9-55" name="__codelineno-9-55" href="#__codelineno-9-55"></a>SN      0       number of others:       0
<a id="__codelineno-9-56" name="__codelineno-9-56" href="#__codelineno-9-56"></a>SN      0       number of multiallelic sites:   0
<a id="__codelineno-9-57" name="__codelineno-9-57" href="#__codelineno-9-57"></a>SN      0       number of multiallelic SNP sites:       0
<a id="__codelineno-9-58" name="__codelineno-9-58" href="#__codelineno-9-58"></a># TSTV, transitions/transversions:
<a id="__codelineno-9-59" name="__codelineno-9-59" href="#__codelineno-9-59"></a># TSTV  [2]id   [3]ts   [4]tv   [5]ts/tv        [6]ts (1st ALT) [7]tv (1st ALT) [8]ts/tv (1st ALT)
<a id="__codelineno-9-60" name="__codelineno-9-60" href="#__codelineno-9-60"></a>TSTV    0       1602    3241    0.49    1602    3241    0.49
<a id="__codelineno-9-61" name="__codelineno-9-61" href="#__codelineno-9-61"></a># SiS, Singleton stats:
<a id="__codelineno-9-62" name="__codelineno-9-62" href="#__codelineno-9-62"></a># SiS   [2]id   [3]allele count [4]number of SNPs       [5]number of transitions        [6]number of transversions      [7]number of indels     [8]repeat-consistent    [9]repeat-inconsistent  [10]not applicable
<a id="__codelineno-9-63" name="__codelineno-9-63" href="#__codelineno-9-63"></a>SiS     0       1       4843    1602    3241    0       0       0       0
<a id="__codelineno-9-64" name="__codelineno-9-64" href="#__codelineno-9-64"></a># AF, Stats by non-reference allele frequency:
<a id="__codelineno-9-65" name="__codelineno-9-65" href="#__codelineno-9-65"></a># AF    [2]id   [3]allele frequency     [4]number of SNPs       [5]number of transitions        [6]number of transversions      [7]number of indels     [8]repeat-consistent    [9]repeat-inconsistent  [10]not applicable
<a id="__codelineno-9-66" name="__codelineno-9-66" href="#__codelineno-9-66"></a>AF      0       0.000000        4843    1602    3241    0       0       0       0
<a id="__codelineno-9-67" name="__codelineno-9-67" href="#__codelineno-9-67"></a># QUAL, Stats by quality:
<a id="__codelineno-9-68" name="__codelineno-9-68" href="#__codelineno-9-68"></a># QUAL  [2]id   [3]Quality      [4]number of SNPs       [5]number of transitions (1st ALT)      [6]number of transversions (1st ALT)    [7]number of indels
<a id="__codelineno-9-69" name="__codelineno-9-69" href="#__codelineno-9-69"></a>QUAL    0       998     4843    1602    3241    0
<a id="__codelineno-9-70" name="__codelineno-9-70" href="#__codelineno-9-70"></a># IDD, InDel distribution:
<a id="__codelineno-9-71" name="__codelineno-9-71" href="#__codelineno-9-71"></a># IDD   [2]id   [3]length (deletions negative)  [4]count
<a id="__codelineno-9-72" name="__codelineno-9-72" href="#__codelineno-9-72"></a># ST, Substitution types:
<a id="__codelineno-9-73" name="__codelineno-9-73" href="#__codelineno-9-73"></a># ST    [2]id   [3]type [4]count
<a id="__codelineno-9-74" name="__codelineno-9-74" href="#__codelineno-9-74"></a>ST      0       A&gt;C     392
<a id="__codelineno-9-75" name="__codelineno-9-75" href="#__codelineno-9-75"></a>ST      0       A&gt;G     406
<a id="__codelineno-9-76" name="__codelineno-9-76" href="#__codelineno-9-76"></a>ST      0       A&gt;T     419
<a id="__codelineno-9-77" name="__codelineno-9-77" href="#__codelineno-9-77"></a>ST      0       C&gt;A     412
<a id="__codelineno-9-78" name="__codelineno-9-78" href="#__codelineno-9-78"></a>ST      0       C&gt;G     429
<a id="__codelineno-9-79" name="__codelineno-9-79" href="#__codelineno-9-79"></a>ST      0       C&gt;T     398
<a id="__codelineno-9-80" name="__codelineno-9-80" href="#__codelineno-9-80"></a>ST      0       G&gt;A     381
<a id="__codelineno-9-81" name="__codelineno-9-81" href="#__codelineno-9-81"></a>ST      0       G&gt;C     411
<a id="__codelineno-9-82" name="__codelineno-9-82" href="#__codelineno-9-82"></a>ST      0       G&gt;T     391
<a id="__codelineno-9-83" name="__codelineno-9-83" href="#__codelineno-9-83"></a>ST      0       T&gt;A     392
<a id="__codelineno-9-84" name="__codelineno-9-84" href="#__codelineno-9-84"></a>ST      0       T&gt;C     417
<a id="__codelineno-9-85" name="__codelineno-9-85" href="#__codelineno-9-85"></a>ST      0       T&gt;G     395
<a id="__codelineno-9-86" name="__codelineno-9-86" href="#__codelineno-9-86"></a># DP, Depth distribution
<a id="__codelineno-9-87" name="__codelineno-9-87" href="#__codelineno-9-87"></a># DP    [2]id   [3]bin  [4]number of genotypes  [5]fraction of genotypes (%)    [6]number of sites      [7]fraction of sites (%)
</code></pre></div>
--->
</article>
</div>
</div>
<button class="md-top md-icon" data-md-component="top" hidden="" type="button">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg>
            Back to top
          </button>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top"], "search": "../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
<script src="../assets/javascripts/bundle.fac441b0.min.js"></script>
<script src="../javascripts/mathjax.js"></script>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>