
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Genomics Aotearoa">
      
      
        <link rel="canonical" href="https://genomicsaotearoa.github.io/Pangenome-Graphs-Workshop/3_Introduction_to_pggb/">
      
      
        <link rel="prev" href="../2_Tools_and_setup/">
      
      
        <link rel="next" href="../4_Running_pggb/">
      
      <link rel="icon" href="../theme_figures/nesi_ga.png">
      <meta name="generator" content="mkdocs-1.4.3, mkdocs-material-9.1.19">
    
    
      
        <title>3. Introduction to PGGB - Pangenome Graphs</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.eebd395e.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.ecc896b0.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
   <link href="../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
        html.glightbox-open { overflow: initial; height: 100%; }
        .gslide-title { margin-top: 0px; user-select: text; }
        .gslide-desc { color: #666; user-select: text; }
        .gslide-image img { background: white; }
        
            .gscrollbar-fixer { padding-right: 15px; }
            .gdesc-inner { font-size: 0.75rem; }
            body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
            body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
            body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
            </style> <script src="../assets/javascripts/glightbox.min.js"></script></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="custom" data-md-color-accent="indigo">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#3-introduction-to-pggb" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
<!-- Custom Banner -->
<div class="banner">
    <p style="color:rgb(250, 246, 246); display:inline;font-size: 160%;">This repository is in the early stages of development (Beta Version)</p>
</div>

<!-- Original Header -->


  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Pangenome Graphs" class="md-header__button md-logo" aria-label="Pangenome Graphs" data-md-component="logo">
      
  <img src="../theme_figures/nesi_ga.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Pangenome Graphs
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              3. Introduction to PGGB
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
          
            
            
            
            <input class="md-option" data-md-color-media="(prefers-color-scheme)" data-md-color-scheme="default" data-md-color-primary="custom" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
            
              <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_3" hidden>
                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"/></svg>
              </label>
            
          
            
            
            
            <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="custom" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_2">
            
              <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
              </label>
            
          
            
            
            
            <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="custom" data-md-color-accent="indigo"  aria-label="Switch to system preference"  type="radio" name="__palette" id="__palette_3">
            
              <label class="md-header__button md-icon" title="Switch to system preference" for="__palette_2" hidden>
                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
              </label>
            
          
        </form>
      
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/GenomicsAotearoa/Pangenome-Graphs-Workshop" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/Pangenome-Graphs-Workshop
  </div>
</a>
      </div>
    
  </nav>
  
</header>

    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Pangenome Graphs" class="md-nav__button md-logo" aria-label="Pangenome Graphs" data-md-component="logo">
      
  <img src="../theme_figures/nesi_ga.png" alt="logo">

    </a>
    Pangenome Graphs
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/GenomicsAotearoa/Pangenome-Graphs-Workshop" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/Pangenome-Graphs-Workshop
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../1_Introduction_to_pangenome_graphs/" class="md-nav__link">
        1. Introduction to pangenome graphs
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../2_Tools_and_setup/" class="md-nav__link">
        2. Tools & setup
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          3. Introduction to PGGB
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        3. Introduction to PGGB
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#variation-graph-and-pggb" class="md-nav__link">
    Variation graph and PGGB
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#how-does-pggb-work" class="md-nav__link">
    How does PGGB work?
  </a>
  
    <nav class="md-nav" aria-label="How does PGGB work?">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#input-fasta-file" class="md-nav__link">
    Input fasta file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-to-all-alignment" class="md-nav__link">
    All-to-all alignment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inducing-the-graph" class="md-nav__link">
    Inducing the graph
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#normalizing-the-graph-with-smoothxg-and-gfafix" class="md-nav__link">
    Normalizing the graph with smoothxg and gfafix
  </a>
  
    <nav class="md-nav" aria-label="Normalizing the graph with smoothxg and gfafix">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#smoothxg" class="md-nav__link">
    smoothxg
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gfafix" class="md-nav__link">
    gfafix
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#multiqc-and-graph-statistics" class="md-nav__link">
    MultiQC and graph statistics
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pggb-workflow" class="md-nav__link">
    PGGB workflow
  </a>
  
    <nav class="md-nav" aria-label="PGGB workflow">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#key-parameters" class="md-nav__link">
    Key parameters
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#other-parameters-for-executing-pggb" class="md-nav__link">
    Other parameters for executing PGGB
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#examples-of-key-parameters-for-executing-pggb" class="md-nav__link">
    Examples of key parameters for executing PGGB
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../4_Running_pggb/" class="md-nav__link">
        4. Running PGGB
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../5_Evaluating_pggb_output/" class="md-nav__link">
        5. Evaluating output
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../6_Variant_calling/" class="md-nav__link">
        6. Variant calling
      </a>
    </li>
  

    
      
      
      

  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
        
      
      <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_8" >
      
      
      
        <label class="md-nav__link" for="__nav_8" id="__nav_8_label" tabindex="0">
          Supplementary
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_8_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_8">
          <span class="md-nav__icon md-icon"></span>
          Supplementary
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../Supplementary/supplementary_1/" class="md-nav__link">
        NeSI HPC Auth.Factors Setup & Jupyter Login
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../Supplementary/supplementary_2/" class="md-nav__link">
        Jupyter Virtual Desktop
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../Supplementary/supplementary_3_Short_reads/" class="md-nav__link">
        Short reads
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#variation-graph-and-pggb" class="md-nav__link">
    Variation graph and PGGB
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#how-does-pggb-work" class="md-nav__link">
    How does PGGB work?
  </a>
  
    <nav class="md-nav" aria-label="How does PGGB work?">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#input-fasta-file" class="md-nav__link">
    Input fasta file
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#all-to-all-alignment" class="md-nav__link">
    All-to-all alignment
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#inducing-the-graph" class="md-nav__link">
    Inducing the graph
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#normalizing-the-graph-with-smoothxg-and-gfafix" class="md-nav__link">
    Normalizing the graph with smoothxg and gfafix
  </a>
  
    <nav class="md-nav" aria-label="Normalizing the graph with smoothxg and gfafix">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#smoothxg" class="md-nav__link">
    smoothxg
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gfafix" class="md-nav__link">
    gfafix
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#multiqc-and-graph-statistics" class="md-nav__link">
    MultiQC and graph statistics
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#pggb-workflow" class="md-nav__link">
    PGGB workflow
  </a>
  
    <nav class="md-nav" aria-label="PGGB workflow">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#key-parameters" class="md-nav__link">
    Key parameters
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#other-parameters-for-executing-pggb" class="md-nav__link">
    Other parameters for executing PGGB
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#examples-of-key-parameters-for-executing-pggb" class="md-nav__link">
    Examples of key parameters for executing PGGB
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="3-introduction-to-pggb">3. Introduction to PGGB<a class="headerlink" href="#3-introduction-to-pggb" title="Permanent link">&para;</a></h1>
<h2 id="variation-graph-and-pggb">Variation graph and PGGB<a class="headerlink" href="#variation-graph-and-pggb" title="Permanent link">&para;</a></h2>
<div class="admonition info"></div>
<p><a class="glightbox" href="../theme_figures/example_of_Graph_reference_small.png" data-type="image" data-width="100%" data-height="auto" data-desc-position="bottom"><img alt="Variation graphs" src="../theme_figures/example_of_Graph_reference_small.png" /></a>
<a href="https://www.bioworld.com/articles/696850-pangenome-gives-more-panoramic-view-of-human-diversity?v=preview">https://www.bioworld.com/articles/696850-pangenome-gives-more-panoramic-view-of-human-diversity?v=preview</a></p>
<ul>
<li>
<p>Variation graphs provide a compact representation of genetic variation across a population in the form of bidirected DNA sequence graphs, and they can encompass large-scale structural variants (SVs) such as inversions and duplications.</p>
</li>
<li>
<p>PGGB, the the PanGenome Graph Builder, is a reference-free pangenome graph constuction method. PGGB builds pangenome graphs from a set of input sequences.</p>
</li>
<li>
<p>The main novelty of PGGB  is not just that it doesn't rely on a reference, but more importantly, it can accurately and fully capture every part of the input genomes.</p>
</li>
</ul>
<h2 id="how-does-pggb-work">How does PGGB work?<a class="headerlink" href="#how-does-pggb-work" title="Permanent link">&para;</a></h2>
<div class="admonition info"></div>
<ul>
<li>
<p><strong>PGGB</strong> generates graphs using an all-to-all alignment of input sequences (<code>wfmash</code>), graph induction (<code>seqwish</code>), and progressive normalization (<code>smoothxg</code>, <code>gfaffix</code>).</p>
</li>
<li>
<p><strong>Scalable  Graph Construction</strong>: The PGGB algorithm is designed to handle large-scale genomes and can efficiently construct genome graphs containing extensive genetic variations.</p>
</li>
<li>
<p>After construction, PGGB generates diagnostic visualizations of the graph (<code>odgi</code>).</p>
</li>
<li>
<p>PGGB writes its output in <strong>GFAv1(.gfa)</strong> format, which can be used as input by numerous "genome graph" and pangenome tools, such as the vg and odgi toolkits. <strong>.gfa file is the new data structure for graph references</strong></p>
</li>
<li>
<p>A variant call report (in Variant Call Format - VCF) representing both small and large variants can be generated <strong>based on any reference genome</strong> included in the graph.</p>
</li>
</ul>
<h3 id="input-fasta-file">Input fasta file<a class="headerlink" href="#input-fasta-file" title="Permanent link">&para;</a></h3>
<div class="admonition info"></div>
<p><center>
<a class="glightbox" href="../theme_figures/PGGB_workflow_1_small.png" data-type="image" data-width="100%" data-height="auto" data-desc-position="bottom"><img alt="bacterial-pangenome" src="../theme_figures/PGGB_workflow_1_small.png" /></a>
</center></p>
<h3 id="all-to-all-alignment">All-to-all alignment<a class="headerlink" href="#all-to-all-alignment" title="Permanent link">&para;</a></h3>
<div class="admonition info"></div>
<p><strong>All-to-all alignment</strong> refers to the process of aligning all sequences in a given set against each other, rather than aligning them to a single reference sequence.
PGGB begins with an alignment using <code>wfmash</code> to align the input genomes. This compares all sequences to each other and finds the best <code>N</code> mappings for each. It produces base-level alignments.</p>
<p><center>
<a class="glightbox" href="../theme_figures/PGGB_workflow_2_small.png" data-type="image" data-width="100%" data-height="auto" data-desc-position="bottom"><img alt="bacterial-pangenome" src="../theme_figures/PGGB_workflow_2_small.png" /></a>
</center></p>
<h3 id="inducing-the-graph">Inducing the graph<a class="headerlink" href="#inducing-the-graph" title="Permanent link">&para;</a></h3>
<div class="admonition info"></div>
<p><strong>Graph induction</strong> refers to the process of constructing the genome graph by progressively integrating genetic variants into a reference genome.
These base-level alignments are converted into a graph with <code>seqwish</code>. A filter is applied to remove short matches, which anchors the graph on confident longer exact matches.</p>
<p><center>
<a class="glightbox" href="../theme_figures/PGGB_workflow_3_small.png" data-type="image" data-width="100%" data-height="auto" data-desc-position="bottom"><img alt="bacterial-pangenome" src="../theme_figures/PGGB_workflow_3_small.png" /></a>
</center></p>
<h3 id="normalizing-the-graph-with-smoothxg-and-gfafix">Normalizing the graph with smoothxg and gfafix<a class="headerlink" href="#normalizing-the-graph-with-smoothxg-and-gfafix" title="Permanent link">&para;</a></h3>
<div class="admonition info"></div>
<p>This process aims to optimize the structure and representation of the genome graph by resolving redundant or overlapping elements. This step is typically performed after the initial construction of the graph.</p>
<h4 id="smoothxg">smoothxg<a class="headerlink" href="#smoothxg" title="Permanent link">&para;</a></h4>
<div class="admonition info"></div>
<p>Through a series of passes over the pangenome, <code>smoothxg</code> reshapes the graph to reduce local complexity and underalignment. This resolves key problems encountered in earlier attempts to implement all-vs-all alignment based graph construction, which typically resulted in very complex, looping, graph motifs at small scales, and redundancy caused by match filtering.
The normalization process in PGGB involves several steps, which may vary depending on the specific implementation or version of the tool. Here are some common steps involved in normalizing the graph:<br />
1.  <strong>Removal of Redundant Nodes</strong>: During the construction of the genome graph, it is possible that some nodes become redundant due to overlapping or repetitive sequences. Normalization involves identifying and removing these redundant nodes, streamlining the graph structure.<br />
2.  <strong>Edge Optimization</strong>: Edges represent connections between nodes. During normalization, the edges are optimized to minimize redundancy and improve the efficiency of the graph. This can include merging or repositioning edges to create a more streamlined and accurate representation of the genome.<br />
<a class="glightbox" href="../theme_figures/PGGB_workflow_4_small.png" data-type="image" data-width="100%" data-height="auto" data-desc-position="bottom"><img alt="bacterial-pangenome" src="../theme_figures/PGGB_workflow_4_small.png" /></a></p>
<h4 id="gfafix">gfafix<a class="headerlink" href="#gfafix" title="Permanent link">&para;</a></h4>
<div class="admonition info"></div>
<ol>
<li><strong>Compact Representation</strong>: Normalization aims to reduce the overall size of the graph by compacting the representation. This can involve compressing repetitive regions or simplifying complex structures while preserving the essential information and variant representation.</li>
<li><strong>Graph Refinement</strong>: The normalization process also involves refining the graph structure by resolving inconsistencies, correcting errors, and improving the overall quality of the graph representation. This may include resolving conflicts between nodes and edges, addressing mismatches, and ensuring the graph accurately reflects the underlying genetic variations.
<a class="glightbox" href="../theme_figures/PGGB_workflow_5_small.png" data-type="image" data-width="100%" data-height="auto" data-desc-position="bottom"><img alt="bacterial-pangenome" src="../theme_figures/PGGB_workflow_5_small.png" /></a></li>
</ol>
<h3 id="multiqc-and-graph-statistics">MultiQC and graph statistics<a class="headerlink" href="#multiqc-and-graph-statistics" title="Permanent link">&para;</a></h3>
<div class="admonition info"></div>
<p><a class="glightbox" href="../theme_figures/PGGB_workflow_6_small.png" data-type="image" data-width="100%" data-height="auto" data-desc-position="bottom"><img alt="bacterial-pangenome" src="../theme_figures/PGGB_workflow_6_small.png" /></a></p>
<h2 id="pggb-workflow">PGGB workflow<a class="headerlink" href="#pggb-workflow" title="Permanent link">&para;</a></h2>
<div class="admonition info"></div>
<p><a class="glightbox" href="../theme_figures/PGGB_workflow_7_small.png" data-type="image" data-width="100%" data-height="auto" data-desc-position="bottom"><img alt="bacterial-pangenome" src="../theme_figures/PGGB_workflow_7_small.png" /></a></p>
<h3 id="key-parameters">Key parameters<a class="headerlink" href="#key-parameters" title="Permanent link">&para;</a></h3>
<div class="admonition info"></div>
<p>The overall structure of PGGB's output graph is defined by three parameters: genome number (<code>-n</code>), segment length (<code>-s</code>), and pairwise identity (<code>-p</code>). </p>
<ul>
<li><code>-n</code> : Genome number.</li>
<li><code>-s</code> : Segment length defines the seed length used by the "MashMap3" homology mapper in wfmash.</li>
<li>
<p><code>-p</code> : The pairwise identity (-p) is the minimum allowed pairwise identity between seeds. <strong>default, 90</strong></p>
</li>
<li>
<p><code>-k</code> : An additional parameter, can also greatly affect graph structure by pruning matches shorter than a given threshold from the initial graph model. In effect, <strong>-k N removes any match shorter than Nbp</strong> from the initial alignment. This filter removes potentially ambiguous pairwise alignments from consideration in establishing the initial scaffold of the graph. <strong>default 19</strong></p>
</li>
</ul>
<h3 id="other-parameters-for-executing-pggb">Other parameters for executing PGGB<a class="headerlink" href="#other-parameters-for-executing-pggb" title="Permanent link">&para;</a></h3>
<div class="admonition info"></div>
<ul>
<li>
<p><code>-S</code> : Generate statistics of the seqwish and smoothxg graph</p>
</li>
<li>
<p><code>-m</code> : Generates MultiQC report of graphs' statistics and visualizations, automatically runs odgi stats</p>
</li>
<li>
<p><code>-V</code> : Specify a path to generate a VCF from the graph. </p>
</li>
<li>
<p><code>-o</code>, <code>--output-dir PATH</code> : Output directory</p>
</li>
</ul>
<h3 id="examples-of-key-parameters-for-executing-pggb">Examples of key parameters for executing PGGB<a class="headerlink" href="#examples-of-key-parameters-for-executing-pggb" title="Permanent link">&para;</a></h3>
<div class="admonition info"></div>
<ul>
<li>Human, whole genome, 90 haplotypes: <code>pggb -p 98 -s 50k -n 90 -k 79 ...</code></li>
<li>15 helicobacter genomes, 5% divergence: <code>pggb -p 90 -n 15 -k 79 ....</code></li>
<li>Yeast genomes, 5% divergence: PGGB's defaults should work well, just set -n.</li>
<li>Aligning 9 Major Histocompatibility Complex (MHC) class II assemblies from vertebrate genomes (5-10% divergence): <code>pggb -p 90 -n 9 -k 29 ...</code></li>
<li>A few thousand bacterial genomes <code>pggb -x auto -n 2146 ...</code>. In general mapping sparsification <strong>(<code>-x auto</code>)</strong> is a good idea when you have many hundreds to thousands of genomes.</li>
</ul>





                
              </article>
            </div>
          
          
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </button>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../2_Tools_and_setup/" class="md-footer__link md-footer__link--prev" aria-label="Previous: 2. Tools &amp; setup" rel="prev">
            <div class="md-footer__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                2. Tools & setup
              </div>
            </div>
          </a>
        
        
          
          <a href="../4_Running_pggb/" class="md-footer__link md-footer__link--next" aria-label="Next: 4. Running PGGB" rel="next">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                4. Running PGGB
              </div>
            </div>
            <div class="md-footer__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Pangenome Graphs workshop material is licensed under a <a rel="noopener" target="_blank" href="https://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html">GNU General Public License v2.0</a>
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.footer", "content.code.copy", "content.code.annotate"], "search": "../assets/javascripts/workers/search.74e28a9f.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.220ee61c.min.js"></script>
      
        
          <script src="../javascripts/mathjax.js"></script>
        
      
        
          <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
        
      
        
          <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
        
      
    
  <script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>