
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Genomics Aotearoa">
      
      
        <link rel="canonical" href="https://genomicsaotearoa.github.io/Pangenome-Graphs-Workshop/2_Tools_and_setup/">
      
      
        <link rel="prev" href="../1_Introduction_to_pangenome_graphs/">
      
      
        <link rel="next" href="../3_Introduction_to_pggb/">
      
      <link rel="icon" href="../theme_figures/nesi_ga.png">
      <meta name="generator" content="mkdocs-1.4.3, mkdocs-material-9.1.19">
    
    
      
        <title>2. Tools & setup - Pangenome Graphs</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.eebd395e.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.ecc896b0.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../stylesheets/extra.css">
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
   <link href="../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
        html.glightbox-open { overflow: initial; height: 100%; }
        .gslide-title { margin-top: 0px; user-select: text; }
        .gslide-desc { color: #666; user-select: text; }
        .gslide-image img { background: white; }
        
            .gscrollbar-fixer { padding-right: 15px; }
            .gdesc-inner { font-size: 0.75rem; }
            body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
            body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
            body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
            </style> <script src="../assets/javascripts/glightbox.min.js"></script></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="custom" data-md-color-accent="indigo">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#2-tools-setup" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
<!-- Custom Banner -->
<div class="banner">
    <p style="color:rgb(250, 246, 246); display:inline;font-size: 160%;">This repository is in the early stages of development (Beta Version)</p>
</div>

<!-- Original Header -->


  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="Pangenome Graphs" class="md-header__button md-logo" aria-label="Pangenome Graphs" data-md-component="logo">
      
  <img src="../theme_figures/nesi_ga.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Pangenome Graphs
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              2. Tools & setup
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
          
            
            
            
            <input class="md-option" data-md-color-media="(prefers-color-scheme)" data-md-color-scheme="default" data-md-color-primary="custom" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
            
              <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_3" hidden>
                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"/></svg>
              </label>
            
          
            
            
            
            <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="custom" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_2">
            
              <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
              </label>
            
          
            
            
            
            <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="custom" data-md-color-accent="indigo"  aria-label="Switch to system preference"  type="radio" name="__palette" id="__palette_3">
            
              <label class="md-header__button md-icon" title="Switch to system preference" for="__palette_2" hidden>
                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
              </label>
            
          
        </form>
      
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/GenomicsAotearoa/Pangenome-Graphs-Workshop" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/Pangenome-Graphs-Workshop
  </div>
</a>
      </div>
    
  </nav>
  
</header>

    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="Pangenome Graphs" class="md-nav__button md-logo" aria-label="Pangenome Graphs" data-md-component="logo">
      
  <img src="../theme_figures/nesi_ga.png" alt="logo">

    </a>
    Pangenome Graphs
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/GenomicsAotearoa/Pangenome-Graphs-Workshop" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/Pangenome-Graphs-Workshop
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../1_Introduction_to_pangenome_graphs/" class="md-nav__link">
        1. Introduction to pangenome graphs
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          2. Tools & setup
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        2. Tools & setup
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#obtaining-pggb" class="md-nav__link">
    Obtaining pggb
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#using-pggb-via-singularity" class="md-nav__link">
    Using pggb via Singularity
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#neisseria-meningitidis-data-set" class="md-nav__link">
    Neisseria meningitidis data set
  </a>
  
    <nav class="md-nav" aria-label="Neisseria meningitidis data set">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#circlator" class="md-nav__link">
    Circlator
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#setting-up-your-project-directory-and-downloading-the-dataset" class="md-nav__link">
    Setting up your project directory and downloading the dataset
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../3_Introduction_to_pggb/" class="md-nav__link">
        3. Introduction to PGGB
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../4_Running_pggb/" class="md-nav__link">
        4. Running PGGB
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../5_Evaluating_pggb_output/" class="md-nav__link">
        5. Evaluating output
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../6_Variant_calling/" class="md-nav__link">
        6. Variant calling
      </a>
    </li>
  

    
      
      
      

  
  
  
    
      
    
    <li class="md-nav__item md-nav__item--section md-nav__item--nested">
      
      
      
      
        
      
      <input class="md-nav__toggle md-toggle md-toggle--indeterminate" type="checkbox" id="__nav_8" >
      
      
      
        <label class="md-nav__link" for="__nav_8" id="__nav_8_label" tabindex="0">
          Supplementary
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_8_label" aria-expanded="false">
        <label class="md-nav__title" for="__nav_8">
          <span class="md-nav__icon md-icon"></span>
          Supplementary
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../Supplementary/supplementary_1/" class="md-nav__link">
        NeSI HPC Auth.Factors Setup & Jupyter Login
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../Supplementary/supplementary_2/" class="md-nav__link">
        Jupyter Virtual Desktop
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../Supplementary/supplementary_3_Short_reads/" class="md-nav__link">
        Short reads
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#obtaining-pggb" class="md-nav__link">
    Obtaining pggb
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#using-pggb-via-singularity" class="md-nav__link">
    Using pggb via Singularity
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#neisseria-meningitidis-data-set" class="md-nav__link">
    Neisseria meningitidis data set
  </a>
  
    <nav class="md-nav" aria-label="Neisseria meningitidis data set">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#circlator" class="md-nav__link">
    Circlator
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#setting-up-your-project-directory-and-downloading-the-dataset" class="md-nav__link">
    Setting up your project directory and downloading the dataset
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="2-tools-setup">2. Tools &amp; setup<a class="headerlink" href="#2-tools-setup" title="Permanent link">&para;</a></h1>
<div class="admonition screwdriver-wrench">
<p class="admonition-title">Tools used for the pangenome graph pipeline</p>
<ul>
<li>The PanGenome Graph Builder (PGGB) is used for graph construction (<a href="https://github.com/pangenome/pggb">https://github.com/pangenome/pggb</a>).</li>
<li>The Optimized Dynamic Genome/Graph Implementation (ODGI) is employed for graph manipulation (<a href="https://github.com/pangenome/odgi">https://github.com/pangenome/odgi</a>).</li>
<li>The VG Toolkit is utilized for variant calling with NGS against a pangenome graph reference (<a href="https://github.com/vgteam/vg">https://github.com/vgteam/vg</a>).</li>
<li>Mash is used to estimate pairwise distances among genomes (<a href="https://github.com/marbl/Mash">https://github.com/marbl/Mash</a>).</li>
<li>ProgressiveMauve is utilized to verify the alignment of genomes (<a href="https://darlinglab.org/mauve/user-guide/progressivemauve.html">https://darlinglab.org/mauve/user-guide/progressivemauve.html</a>)</li>
<li>Circlator is employed to fix the starting point of genomes (<a href="https://sanger-pathogens.github.io/circlator/">https://sanger-pathogens.github.io/circlator/</a>).</li>
<li>SAMtools is used to build an index of input FASTA files (<a href="https://github.com/samtools/samtools">https://github.com/samtools/samtools</a>).</li>
<li>BCFtools is used to check the statistics of VCF files (<a href="https://github.com/samtools/bcftools">https://github.com/samtools/bcftools</a>).</li>
</ul>
</div>
<details class="info">
<summary>Other Tools might be useful for pangenome graphs</summary>
<ul>
<li>The Minigraph-Cactus Pangenome Pipeline (<a href="https://github.com/ComparativeGenomicsToolkit/cactus/blob/master/doc/pangenome.md">https://github.com/ComparativeGenomicsToolkit/cactus/blob/master/doc/pangenome.md</a>).</li>
<li>gfaestus, which can be used for graph visualization (<a href="https://github.com/chfi/gfaestus">https://github.com/chfi/gfaestus</a>)</li>
<li>SequenceTubeMap for tube map visualization of genomic sequence graphs (<a href="https://github.com/vgteam/sequenceTubeMap">https://github.com/vgteam/sequenceTubeMap</a>)</li>
<li>......</li>
</ul>
</details>
<div class="admonition circle-info">
<p class="admonition-title">Software stack for pggb pipeline</p>
<p>All of the required software is pre-installed and configured as modules. We will load them as needed  </p>
<ul>
<li>pggb/0.5.3-Miniconda3</li>
<li>SAMtools</li>
<li>Mash/2.3-GCC-11.3.0</li>
<li>Circlator/1.5.5-gimkl-2022a-Python-3.10.5</li>
<li>vg/1.46.0</li>
</ul>
</div>
<details class="circle-info">
<summary>Running the pggb workflow locally</summary>
<p>The following code <strong>DOES NOT</strong> need to be run if you working on NeSI, but is 
provided here to show how to work through the workshop material on a non-NeSI 
system (e.g., if you are running the code on your own computer).</p>
<h3 id="obtaining-pggb">Obtaining pggb<a class="headerlink" href="#obtaining-pggb" title="Permanent link">&para;</a></h3>
<p>From <a href="https://github.com/pangenome/pggb">https://github.com/pangenome/pggb</a>, you can find the details about installing pggb with Docker, Singularity, bioconda, guix, or by manually building its dependencies.</p>
<h3 id="using-pggb-via-singularity">Using pggb via Singularity<a class="headerlink" href="#using-pggb-via-singularity" title="Permanent link">&para;</a></h3>
<p>Many managed HPCs utilize Singularity as a secure alternative to docker. Fortunately, docker images can be run through Singularity seamlessly.
First pull the docker file and create a Singularity SIF image from the dockerfile. This might take a few minutes.
<div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>singularity<span class="w"> </span>pull<span class="w"> </span>docker://ghcr.io/pangenome/pggb:latest
</code></pre></div>
Next clone the pggb repo and cd into it
<div class="highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>git<span class="w"> </span>clone<span class="w"> </span>--recursive<span class="w"> </span>https://github.com/pangenome/pggb.git
<a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a><span class="nb">cd</span><span class="w"> </span>pggb
</code></pre></div>
Finally, run pggb from the Singularity image. For Singularity to be able to read and write files to a directory on the host operating system, we need to 'bind' that directory using the -B option and pass the pggb command as an argument.
<div class="highlight"><pre><span></span><code><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>singularity<span class="w"> </span>run<span class="w"> </span>-B<span class="w"> </span><span class="si">${</span><span class="nv">PWD</span><span class="si">}</span>/data:/data<span class="w"> </span>../pggb_latest.sif<span class="w"> </span>pggb<span class="w"> </span>-i<span class="w"> </span>/data/HLA/DRB1-3123.fa.gz<span class="w"> </span>-p<span class="w"> </span><span class="m">70</span><span class="w"> </span>-s<span class="w"> </span><span class="m">3000</span><span class="w"> </span>-G<span class="w"> </span><span class="m">2000</span><span class="w"> </span>-n<span class="w"> </span><span class="m">10</span><span class="w"> </span>-t<span class="w"> </span><span class="m">16</span><span class="w"> </span>-v<span class="w"> </span>-V<span class="w"> </span><span class="s1">&#39;gi|568815561:#&#39;</span><span class="w"> </span>-o<span class="w"> </span>/data/out<span class="w"> </span>-M<span class="w"> </span>-m
</code></pre></div></p>
</details>
<div class="admonition info">
<h3 id="neisseria-meningitidis-data-set"><em>Neisseria meningitidis</em> data set<a class="headerlink" href="#neisseria-meningitidis-data-set" title="Permanent link">&para;</a></h3>
<p>In this workshop, we use the bacterium <em>Neisseria (N.) meningitidis</em> as our model organism.</p>
</div>
<ul>
<li>Bacteria called <em>N. meningitidis</em> cause meningococcal disease(IMD). About 1 in 10 people have these bacteria in the back of their nose and throat without being ill. This is called being ‘a carrier.’ Sometimes the bacteria invade the body and cause IMD.</li>
<li>Different capsular polysaccharides divide the meningococci into at least 12 serogroups, of which six groups (A, B, C, W, X, and Y) are responsible for most disease.</li>
<li>The genome of this bacterium spans approximately 2.1 to 2.4 Mb and possesses a GC content ranging from 51-52%.</li>
<li>A notable trait of <em>N. meningitidis</em> genomes is their high recombination rate, which contributes significantly to the substantial genetic diversity found in this species.</li>
<li>Throughout this workshop, we use five genome assemblies of <em>N. meningitidis</em> to evaluate the pangenome pipeline, from the construction of the pangenome graph to the calling of variants.</li>
</ul>
<table>
<thead>
<tr>
<th align="left">genomes</th>
<th align="right">ASM IDs</th>
<th align="right">GCF IDs</th>
<th align="right">SEROGROUP</th>
<th align="right">Sequence type</th>
<th align="right">Clonal Complex</th>
</tr>
</thead>
<tbody>
<tr>
<td align="left">NC_017518.1 Neisseria meningitidis NZ-05/33</td>
<td align="right">ASM19152v1</td>
<td align="right">GCF_000191525.1</td>
<td align="right">B</td>
<td align="right">42</td>
<td align="right">ST-41/44</td>
</tr>
<tr>
<td align="left">NC_003112.2 Neisseria meningitidis MC58</td>
<td align="right">ASM880v1</td>
<td align="right">GCF_000008805.1</td>
<td align="right">-</td>
<td align="right">74</td>
<td align="right">ST-32</td>
</tr>
<tr>
<td align="left">NZ_CP007668.1 Neisseria meningitidis M0579</td>
<td align="right">ASM102983v1</td>
<td align="right">GCF_001029835.1</td>
<td align="right">B</td>
<td align="right">-</td>
<td align="right">ST-41/44</td>
</tr>
<tr>
<td align="left">NZ_CP016880.1 Neisseria meningitidis strain M07165</td>
<td align="right">ASM170367v1</td>
<td align="right">GCF_001703675.1</td>
<td align="right">W</td>
<td align="right">11</td>
<td align="right">ST-11</td>
</tr>
<tr>
<td align="left">NZ_CP020423.2 Neisseria meningitidis strain FDAARGOS_212</td>
<td align="right">ASM207367v2</td>
<td align="right">GCF_002073675.2</td>
<td align="right">C</td>
<td align="right">-</td>
<td align="right">ST16521</td>
</tr>
</tbody>
</table>
<details class="circle-info">
<summary>How the <em>Neisseria meningitidis</em> genomes were formatted for this workshop</summary>
<p>The following code <strong>DOES NOT</strong> need to be run, but is provided here to show how the <em>Neisseria meningitidis</em> genomes
were downloaded and prepared for analysis.</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="c1"># Create a new directory called nm_genomes and change to that directory</span>
<a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a>mkdir<span class="w"> </span>nm_genomes
<a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a><span class="nb">cd</span><span class="w"> </span>nm_genomes
</code></pre></div>
<p>Download the genome assemblies from NCBI and uncompress.  In the Unix environment you can use the <code>curl</code> command.</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="c1">#NC_017518.1</span>
<a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a>curl<span class="w"> </span>-OJX<span class="w"> </span>GET<span class="w"> </span><span class="s2">&quot;https://api.ncbi.nlm.nih.gov/datasets/v2alpha/genome/accession/GCF_000191525.1/download?include_annotation_type=GENOME_FASTA,GENOME_GFF,RNA_FASTA,CDS_FASTA,PROT_FASTA,SEQUENCE_REPORT&amp;filename=GCF_000191525.1.zip&quot;</span><span class="w"> </span>-H<span class="w"> </span><span class="s2">&quot;Accept: application/zip&quot;</span>
<a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a>
<a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a><span class="c1"># ACC NUM?</span>
<a id="__codelineno-4-5" name="__codelineno-4-5" href="#__codelineno-4-5"></a>curl<span class="w"> </span>-OJX<span class="w"> </span>GET<span class="w"> </span><span class="s2">&quot;https://api.ncbi.nlm.nih.gov/datasets/v2alpha/genome/accession/GCF_001029835.1/download?include_annotation_type=GENOME_FASTA,GENOME_GFF,RNA_FASTA,CDS_FASTA,PROT_FASTA,SEQUENCE_REPORT&amp;filename=GCF_001029835.1.zip&quot;</span><span class="w"> </span>-H<span class="w"> </span><span class="s2">&quot;Accept: application/zip&quot;</span>
<a id="__codelineno-4-6" name="__codelineno-4-6" href="#__codelineno-4-6"></a>
<a id="__codelineno-4-7" name="__codelineno-4-7" href="#__codelineno-4-7"></a><span class="c1"># ACC NUM?</span>
<a id="__codelineno-4-8" name="__codelineno-4-8" href="#__codelineno-4-8"></a>curl<span class="w"> </span>-OJX<span class="w"> </span>GET<span class="w"> </span><span class="s2">&quot;https://api.ncbi.nlm.nih.gov/datasets/v2alpha/genome/accession/GCF_001703675.1/download?include_annotation_type=GENOME_FASTA,GENOME_GFF,RNA_FASTA,CDS_FASTA,PROT_FASTA,SEQUENCE_REPORT&amp;filename=GCF_001703675.1.zip&quot;</span><span class="w"> </span>-H<span class="w"> </span><span class="s2">&quot;Accept: application/zip&quot;</span>
<a id="__codelineno-4-9" name="__codelineno-4-9" href="#__codelineno-4-9"></a>
<a id="__codelineno-4-10" name="__codelineno-4-10" href="#__codelineno-4-10"></a><span class="c1"># ACC NUM?</span>
<a id="__codelineno-4-11" name="__codelineno-4-11" href="#__codelineno-4-11"></a>curl<span class="w"> </span>-OJX<span class="w"> </span>GET<span class="w"> </span><span class="s2">&quot;https://api.ncbi.nlm.nih.gov/datasets/v2alpha/genome/accession/GCF_002073675.2/download?include_annotation_type=GENOME_FASTA,GENOME_GFF,RNA_FASTA,CDS_FASTA,PROT_FASTA,SEQUENCE_REPORT&amp;filename=GCF_002073675.2.zip&quot;</span><span class="w"> </span>-H<span class="w"> </span><span class="s2">&quot;Accept: application/zip&quot;</span>
<a id="__codelineno-4-12" name="__codelineno-4-12" href="#__codelineno-4-12"></a>
<a id="__codelineno-4-13" name="__codelineno-4-13" href="#__codelineno-4-13"></a><span class="c1"># ACC NUM?</span>
<a id="__codelineno-4-14" name="__codelineno-4-14" href="#__codelineno-4-14"></a>curl<span class="w"> </span>-OJX<span class="w"> </span>GET<span class="w"> </span><span class="s2">&quot;https://api.ncbi.nlm.nih.gov/datasets/v2alpha/genome/accession/GCF_000008805.1/download?include_annotation_type=GENOME_FASTA,GENOME_GFF,RNA_FASTA,CDS_FASTA,PROT_FASTA,SEQUENCE_REPORT&amp;filename=GCF_000008805.1.zip&quot;</span><span class="w"> </span>-H<span class="w"> </span><span class="s2">&quot;Accept: application/zip&quot;</span>
</code></pre></div>
<p>On NeSI, a slurm job can be run to process the .fna genomes.  The contents of the slurm job file (<code>unzip_genomes.sl</code>) 
are as follows: </p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-5-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-5-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-5-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-5-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-5-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-5-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-5-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-5-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-5-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-5-10">10</a></span>
<span class="normal"><a href="#__codelineno-5-11">11</a></span>
<span class="normal"><a href="#__codelineno-5-12">12</a></span>
<span class="normal"><a href="#__codelineno-5-13">13</a></span>
<span class="normal"><a href="#__codelineno-5-14">14</a></span>
<span class="normal"><a href="#__codelineno-5-15">15</a></span>
<span class="normal"><a href="#__codelineno-5-16">16</a></span>
<span class="normal"><a href="#__codelineno-5-17">17</a></span>
<span class="normal"><a href="#__codelineno-5-18">18</a></span>
<span class="normal"><a href="#__codelineno-5-19">19</a></span>
<span class="normal"><a href="#__codelineno-5-20">20</a></span>
<span class="normal"><a href="#__codelineno-5-21">21</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-5-1" name="__codelineno-5-1"></a><span class="ch">#!/bin/bash -e</span>
<a id="__codelineno-5-2" name="__codelineno-5-2"></a>
<a id="__codelineno-5-3" name="__codelineno-5-3"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a id="__codelineno-5-4" name="__codelineno-5-4"></a><span class="c1">#SBATCH --job-name      extract_fna</span>
<a id="__codelineno-5-5" name="__codelineno-5-5"></a><span class="c1">#SBATCH --cpus-per-task 8</span>
<a id="__codelineno-5-6" name="__codelineno-5-6"></a><span class="c1">#SBATCH --mem           4G</span>
<a id="__codelineno-5-7" name="__codelineno-5-7"></a><span class="c1">#SBATCH --time          1:00:00</span>
<a id="__codelineno-5-8" name="__codelineno-5-8"></a>
<a id="__codelineno-5-9" name="__codelineno-5-9"></a><span class="nv">data</span><span class="o">=</span><span class="nv">$HOME</span>/nm_genomes/*.zip
<a id="__codelineno-5-10" name="__codelineno-5-10"></a>
<a id="__codelineno-5-11" name="__codelineno-5-11"></a><span class="k">for</span><span class="w"> </span>f<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="nv">$data</span>
<a id="__codelineno-5-12" name="__codelineno-5-12"></a><span class="k">do</span>
<a id="__codelineno-5-13" name="__codelineno-5-13"></a><span class="w">    </span><span class="nv">x</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="nv">$f</span><span class="w"> </span>.zip<span class="k">)</span>
<a id="__codelineno-5-14" name="__codelineno-5-14"></a><span class="w">    </span><span class="nb">echo</span><span class="w"> </span><span class="si">${</span><span class="nv">x</span><span class="si">}</span>
<a id="__codelineno-5-15" name="__codelineno-5-15"></a>
<a id="__codelineno-5-16" name="__codelineno-5-16"></a><span class="w">    </span>unzip<span class="w"> </span><span class="nv">$x</span>.zip
<a id="__codelineno-5-17" name="__codelineno-5-17"></a>
<a id="__codelineno-5-18" name="__codelineno-5-18"></a><span class="w">    </span>cp<span class="w"> </span><span class="nv">$HOME</span>/nm_genomes/ncbi_dataset/data/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>/*_genomic.fna<span class="w"> </span>/<span class="nv">$HOME</span>/nm_genomes/
<a id="__codelineno-5-19" name="__codelineno-5-19"></a>
<a id="__codelineno-5-20" name="__codelineno-5-20"></a><span class="w">    </span>rm<span class="w"> </span>-rf<span class="w"> </span>ncbi_dataset
<a id="__codelineno-5-21" name="__codelineno-5-21"></a><span class="k">done</span>
</code></pre></div></td></tr></table></div>
<p>The slurm job can be run via:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a>sbatch unzip_genomes.sl
</code></pre></div>
<p>Remove unneeded files:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a>rm<span class="w"> </span>cds_from_genomic.fna
<a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a>rm<span class="w"> </span>*.zip
<a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>rm<span class="w"> </span>README.md
<a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a>rm<span class="w"> </span>slurm*.out
</code></pre></div>
<p>Use the <code>cat</code> command to combine genomes into a single fasta file:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a>cat *_genomic.fna &gt; 5NM.fa
</code></pre></div>
</details>
<details class="screwdriver-wrench">
<summary>Fix the start point of the input genomes?</summary>
<h4 id="circlator">Circlator<a class="headerlink" href="#circlator" title="Permanent link">&para;</a></h4>
<p>Bacterial genomes are typically circular, so establishing a fixed starting point for each input genome during pangenome graph construction could reduce unnecessary complexity within the graph. Once the genomes are aligned with the same starting point, we can build their index using samtools faidx and proceed with constructing the pangenome graph. I encourage you to experiment with this approach after this workshop to see how effectively it functions. Please note that it may be necessary to try different regions as a starting point to ensure that all genomes are fixed with the same initial region.</p>
<p>let's fix the start for all genome using circlator, submit a slurm job. It takes less than one minute for each sample. </p>
<div class="highlight"><table class="highlighttable"><tr><td class="linenos"><div class="linenodiv"><pre><span></span><span class="normal"><a href="#__codelineno-9-1"> 1</a></span>
<span class="normal"><a href="#__codelineno-9-2"> 2</a></span>
<span class="normal"><a href="#__codelineno-9-3"> 3</a></span>
<span class="normal"><a href="#__codelineno-9-4"> 4</a></span>
<span class="normal"><a href="#__codelineno-9-5"> 5</a></span>
<span class="normal"><a href="#__codelineno-9-6"> 6</a></span>
<span class="normal"><a href="#__codelineno-9-7"> 7</a></span>
<span class="normal"><a href="#__codelineno-9-8"> 8</a></span>
<span class="normal"><a href="#__codelineno-9-9"> 9</a></span>
<span class="normal"><a href="#__codelineno-9-10">10</a></span>
<span class="normal"><a href="#__codelineno-9-11">11</a></span>
<span class="normal"><a href="#__codelineno-9-12">12</a></span>
<span class="normal"><a href="#__codelineno-9-13">13</a></span>
<span class="normal"><a href="#__codelineno-9-14">14</a></span>
<span class="normal"><a href="#__codelineno-9-15">15</a></span>
<span class="normal"><a href="#__codelineno-9-16">16</a></span>
<span class="normal"><a href="#__codelineno-9-17">17</a></span>
<span class="normal"><a href="#__codelineno-9-18">18</a></span>
<span class="normal"><a href="#__codelineno-9-19">19</a></span>
<span class="normal"><a href="#__codelineno-9-20">20</a></span>
<span class="normal"><a href="#__codelineno-9-21">21</a></span>
<span class="normal"><a href="#__codelineno-9-22">22</a></span>
<span class="normal"><a href="#__codelineno-9-23">23</a></span>
<span class="normal"><a href="#__codelineno-9-24">24</a></span>
<span class="normal"><a href="#__codelineno-9-25">25</a></span>
<span class="normal"><a href="#__codelineno-9-26">26</a></span></pre></div></td><td class="code"><div><pre><span></span><code><a id="__codelineno-9-1" name="__codelineno-9-1"></a><span class="ch">#!/bin/bash -e </span>
<a id="__codelineno-9-2" name="__codelineno-9-2"></a>
<a id="__codelineno-9-3" name="__codelineno-9-3"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a id="__codelineno-9-4" name="__codelineno-9-4"></a><span class="c1">#SBATCH --job-name      restart_fna</span>
<a id="__codelineno-9-5" name="__codelineno-9-5"></a><span class="c1">#SBATCH --cpus-per-task 8</span>
<a id="__codelineno-9-6" name="__codelineno-9-6"></a><span class="c1">#SBATCH --mem           4G</span>
<a id="__codelineno-9-7" name="__codelineno-9-7"></a><span class="c1">#SBATCH --time          1:00:00</span>
<a id="__codelineno-9-8" name="__codelineno-9-8"></a>
<a id="__codelineno-9-9" name="__codelineno-9-9"></a>module<span class="w"> </span>load<span class="w"> </span>Circlator/1.5.5-gimkl-2022a-Python-3.10.5
<a id="__codelineno-9-10" name="__codelineno-9-10"></a>
<a id="__codelineno-9-11" name="__codelineno-9-11"></a><span class="nb">cd</span><span class="w"> </span>/home/zyang/pg_test
<a id="__codelineno-9-12" name="__codelineno-9-12"></a><span class="nv">data</span><span class="o">=</span>/home/zyang/pg_test/*.fna
<a id="__codelineno-9-13" name="__codelineno-9-13"></a>
<a id="__codelineno-9-14" name="__codelineno-9-14"></a><span class="c1">#For some dataset, we may have to try different start point to make sure all genomes being fixed with the same region. </span>
<a id="__codelineno-9-15" name="__codelineno-9-15"></a><span class="nv">start_region</span><span class="o">=</span>start_region.fa
<a id="__codelineno-9-16" name="__codelineno-9-16"></a>
<a id="__codelineno-9-17" name="__codelineno-9-17"></a>
<a id="__codelineno-9-18" name="__codelineno-9-18"></a><span class="k">for</span><span class="w"> </span>f<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="nv">$data</span>
<a id="__codelineno-9-19" name="__codelineno-9-19"></a><span class="k">do</span>
<a id="__codelineno-9-20" name="__codelineno-9-20"></a>
<a id="__codelineno-9-21" name="__codelineno-9-21"></a><span class="nv">x</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="nv">$f</span><span class="w"> </span>.fna<span class="k">)</span>
<a id="__codelineno-9-22" name="__codelineno-9-22"></a><span class="nb">echo</span><span class="w"> </span><span class="si">${</span><span class="nv">x</span><span class="si">}</span>
<a id="__codelineno-9-23" name="__codelineno-9-23"></a>
<a id="__codelineno-9-24" name="__codelineno-9-24"></a>circlator<span class="w"> </span>fixstart<span class="w"> </span>--genes_fa<span class="w">  </span>start_region<span class="w">  </span><span class="si">${</span><span class="nv">x</span><span class="si">}</span>.fna<span class="w">  </span><span class="si">${</span><span class="nv">x</span><span class="si">}</span>.restart
<a id="__codelineno-9-25" name="__codelineno-9-25"></a>
<a id="__codelineno-9-26" name="__codelineno-9-26"></a><span class="k">done</span>
</code></pre></div></td></tr></table></div>
<p>Use the <code>cat</code> command to combine genomes into a single fasta file:</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a>cat *_genomic.restart.fasta &gt; 5NMfs.fa
</code></pre></div>
</details>
<details class="screwdriver-wrench">
<summary>Fix the start point of the 5NM genomes</summary>
<p><strong>One tricky aspect is finding a region that is present in all the genomes and is oriented in the same direction</strong>
The mauve alignment of the 5NM genomes 
<a class="glightbox" href="../theme_figures/5NM_marve.png" data-type="image" data-width="100%" data-height="auto" data-desc-position="bottom"><img alt="5NM_Mauve" src="../theme_figures/5NM_marve.png" /></a></p>
<p>The Mauve alignment after attempting to fix the start point using the first 50 lines of genome one. 
<a class="glightbox" href="../theme_figures/5NMre1_marve.png" data-type="image" data-width="100%" data-height="auto" data-desc-position="bottom"><img alt="5NM_Mauve" src="../theme_figures/5NMre1_marve.png" /></a></p>
<p>The Mauve alignment after fixing the start point using the first 50 lines of the second genome, which has been aligned with the first 50 lines of genome one.
<a class="glightbox" href="../theme_figures/5NMre2_marve.png" data-type="image" data-width="100%" data-height="auto" data-desc-position="bottom"><img alt="5NM_Mauve" src="../theme_figures/5NMre2_marve.png" /></a></p>
</details>
<h3 id="setting-up-your-project-directory-and-downloading-the-dataset">Setting up your project directory and downloading the dataset<a class="headerlink" href="#setting-up-your-project-directory-and-downloading-the-dataset" title="Permanent link">&para;</a></h3>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<ul>
<li>Create a new directory under your home folder and change to that directory.
<div class="highlight"><pre><span></span><code><a id="__codelineno-11-1" name="__codelineno-11-1" href="#__codelineno-11-1"></a>mkdir<span class="w"> </span>~/pg_workshop
</code></pre></div>
<div class="highlight"><pre><span></span><code><a id="__codelineno-12-1" name="__codelineno-12-1" href="#__codelineno-12-1"></a>cd ~/pg_workshop
</code></pre></div></li>
<li><code>~/pg_workshop</code> will be your working directory</li>
<li>Keep a note of the absolute path of your working directory
<div class="highlight"><pre><span></span><code><a id="__codelineno-13-1" name="__codelineno-13-1" href="#__codelineno-13-1"></a>pwd
</code></pre></div></li>
</ul>
<div class="admonition success">
<p class="admonition-title">Output</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-14-1" name="__codelineno-14-1" href="#__codelineno-14-1"></a>/home/&lt;YOUR_USER_ID&gt;/pg_workshop
</code></pre></div>
</div>
<ul>
<li>
<p>Please download the dataset. 
<div class="highlight"><pre><span></span><code><a id="__codelineno-15-1" name="__codelineno-15-1" href="#__codelineno-15-1"></a>git<span class="w"> </span>clone<span class="w"> </span>https://github.com/ZoeYang2020/dataset_for_pg_workshop
</code></pre></div></p>
</li>
<li>
<p>Copy the 5NM.fa and  dataset to your working directory (To keep the filenames simple, we have renamed the 5NMfs.fa to 5NM.fa for this workshop)
<div class="highlight"><pre><span></span><code><a id="__codelineno-16-1" name="__codelineno-16-1" href="#__codelineno-16-1"></a>cp ./dataset_for_pg_workshop/5NM.fa ./
</code></pre></div></p>
</li>
</ul>
</div>





                
              </article>
            </div>
          
          
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </button>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../1_Introduction_to_pangenome_graphs/" class="md-footer__link md-footer__link--prev" aria-label="Previous: 1. Introduction to pangenome graphs" rel="prev">
            <div class="md-footer__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                1. Introduction to pangenome graphs
              </div>
            </div>
          </a>
        
        
          
          <a href="../3_Introduction_to_pggb/" class="md-footer__link md-footer__link--next" aria-label="Next: 3. Introduction to PGGB" rel="next">
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Next
              </span>
              <div class="md-ellipsis">
                3. Introduction to PGGB
              </div>
            </div>
            <div class="md-footer__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M4 11v2h12l-5.5 5.5 1.42 1.42L19.84 12l-7.92-7.92L10.5 5.5 16 11H4Z"/></svg>
            </div>
          </a>
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Pangenome Graphs workshop material is licensed under a <a rel="noopener" target="_blank" href="https://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html">GNU General Public License v2.0</a>
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.footer", "content.code.copy", "content.code.annotate"], "search": "../assets/javascripts/workers/search.74e28a9f.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.220ee61c.min.js"></script>
      
        
          <script src="../javascripts/mathjax.js"></script>
        
      
        
          <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
        
      
        
          <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
        
      
    
  <script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>