
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<meta content="Genomics Aotearoa" name="author"/>
<link href="https://genomicsaotearoa.github.io/Pangenome-Graphs-Workshop/Graph_based_variant_calling_for_NGS/" rel="canonical"/>
<link href=".." rel="prev"/>
<link href="../Intro_1/" rel="next"/>
<link href="../theme_figures/nesi_ga.png" rel="icon"/>
<meta content="mkdocs-1.4.3, mkdocs-material-9.1.9" name="generator"/>
<title>NGS data analysis used graph as reference on Nesi - Pangenome Graphs</title>
<link href="../assets/stylesheets/main.85bb2934.min.css" rel="stylesheet"/>
<link href="../assets/stylesheets/palette.a6bdf11c.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
<link href="../stylesheets/extra.css" rel="stylesheet"/>
<script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
<link href="../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            
                .gscrollbar-fixer { padding-right: 15px; }
                .gdesc-inner { font-size: 0.75rem; }
                body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
                body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
                body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
                </style><script src="../assets/javascripts/glightbox.min.js"></script></head>
<body data-md-color-accent="indigo" data-md-color-primary="indigo" data-md-color-scheme="default" dir="ltr">
<script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#ngs-data-analysis-used-graph-as-reference-on-nesi">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--shadow" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Pangenome Graphs" class="md-header__button md-logo" data-md-component="logo" href=".." title="Pangenome Graphs">
<img alt="logo" src="../theme_figures/nesi_ga.png"/>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Pangenome Graphs
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              NGS data analysis used graph as reference on Nesi
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_3" hidden="" title="Switch to light mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"></path></svg>
</label>
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_2" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
<input aria-label="Switch to system preference" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="indigo" data-md-color-scheme="slate" id="__palette_3" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to system preference">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
</form>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/Pangenome-Graphs-Workshop" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/Pangenome-Graphs-Workshop
  </div>
</a>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Pangenome Graphs" class="md-nav__button md-logo" data-md-component="logo" href=".." title="Pangenome Graphs">
<img alt="logo" src="../theme_figures/nesi_ga.png"/>
</a>
    Pangenome Graphs
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/Pangenome-Graphs-Workshop" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/Pangenome-Graphs-Workshop
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="..">
        Pangenome Graphs
      </a>
</li>
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" id="__toc" type="checkbox"/>
<label class="md-nav__link md-nav__link--active" for="__toc">
          NGS data analysis used graph as reference on Nesi
          <span class="md-nav__icon md-icon"></span>
</label>
<a class="md-nav__link md-nav__link--active" href="./">
        NGS data analysis used graph as reference on Nesi
      </a>
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#vg-mapping-preliminaries">
    vg mapping preliminaries
  </a>
<nav aria-label="vg mapping preliminaries" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#learning-objectives">
    Learning objectives
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#use-wgsim-to-simulate-2x150-bp-ngs-data-error-rate-0005-we-choose-five-genomes-for-simulation">
    use wgsim to simulate 2X150 bp NGS data, error rate 0.005. We choose five genomes for simulation,
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#build-index-for-graph">
    build index for graph
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#vg-map-ngs-to-graph">
    vg map NGS to graph
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#genotying-known-variants">
    genotying known variants
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#noverl-variant-calling-using-graph-reference">
    noverl variant calling using graph reference
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../Intro_1/">
        Pangenome Graphs Workshop
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../Intro_2/">
        Intro 2
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../mapping_reads/">
        Mapping Reads using `bwa mem` (Linear Method)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../pangenome_graph_construction/">
        pangenome graph construction with PGGB on Nesi
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../preparing_data_files/">
        Preparing required data files with simuG tool
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../simulate_and_compare/">
        Simulate and compare
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../vc_position_based_comparison/">
        How to compare 2 VCF files based on the exact variant position?
      </a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#vg-mapping-preliminaries">
    vg mapping preliminaries
  </a>
<nav aria-label="vg mapping preliminaries" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#learning-objectives">
    Learning objectives
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#use-wgsim-to-simulate-2x150-bp-ngs-data-error-rate-0005-we-choose-five-genomes-for-simulation">
    use wgsim to simulate 2X150 bp NGS data, error rate 0.005. We choose five genomes for simulation,
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#build-index-for-graph">
    build index for graph
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#vg-map-ngs-to-graph">
    vg map NGS to graph
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#genotying-known-variants">
    genotying known variants
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#noverl-variant-calling-using-graph-reference">
    noverl variant calling using graph reference
  </a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<h1 id="ngs-data-analysis-used-graph-as-reference-on-nesi">NGS data analysis used graph as reference on Nesi<a class="headerlink" href="#ngs-data-analysis-used-graph-as-reference-on-nesi" title="Permanent link">¶</a></h1>
<p align="justify">
In this workshop, we employes the VG toolkit for NGS data analysis based on pangenome graph reference 
</p>
<h2 id="vg-mapping-preliminaries">vg mapping preliminaries<a class="headerlink" href="#vg-mapping-preliminaries" title="Permanent link">¶</a></h2>
<p>Although vg contains a number of tools for working with pangenome graphs, it is best-known for read mapping. This is ultimately what many of its users are interested in vg for. In fact, vg contains three mature short read mapping tools:</p>
<ul>
<li>vg map: the original, highly accurate mapping algorithm</li>
<li>vg giraffe: the much faster and still accurate haplotype-based mapping algorithm</li>
<li>vg mpmap: the splice-aware RNA-seq mapping algorithm</li>
</ul>
<p>more details of vg can be found <a href="https://github.com/vgteam/vg">https://github.com/vgteam/vg</a></p>
<p>we use vg map in this workshop </p>
<h3 id="learning-objectives">Learning objectives<a class="headerlink" href="#learning-objectives" title="Permanent link">¶</a></h3>
<ul>
<li>Map NGS data to graph using vg map</li>
<li>Variant calling for NGS data against genome graph </li>
</ul>
<h2 id="use-wgsim-to-simulate-2x150-bp-ngs-data-error-rate-0005-we-choose-five-genomes-for-simulation">use wgsim to simulate 2X150 bp NGS data, error rate 0.005. We choose five genomes for simulation,<a class="headerlink" href="#use-wgsim-to-simulate-2x150-bp-ngs-data-error-rate-0005-we-choose-five-genomes-for-simulation" title="Permanent link">¶</a></h2>
<p><div class="highlight"><pre><span></span><code><a href="#__codelineno-0-1" id="__codelineno-0-1" name="__codelineno-0-1"></a>mkdir<span class="w"> </span>graph_NGS
<a href="#__codelineno-0-2" id="__codelineno-0-2" name="__codelineno-0-2"></a><span class="nb">cd</span><span class="w"> </span>graph_NGS
<a href="#__codelineno-0-3" id="__codelineno-0-3" name="__codelineno-0-3"></a>mkdir<span class="w"> </span>simu_NGS_data
</code></pre></div>
The script for simulation NGS data
<div class="highlight"><pre><span></span><code><a href="#__codelineno-1-1" id="__codelineno-1-1" name="__codelineno-1-1"></a><span class="ch">#!/bin.bash</span>
<a href="#__codelineno-1-2" id="__codelineno-1-2" name="__codelineno-1-2"></a>
<a href="#__codelineno-1-3" id="__codelineno-1-3" name="__codelineno-1-3"></a><span class="c1">#SBATCH --account       ga03793</span>
<a href="#__codelineno-1-4" id="__codelineno-1-4" name="__codelineno-1-4"></a><span class="c1">#SBATCH --job-name      5NGS_simulate</span>
<a href="#__codelineno-1-5" id="__codelineno-1-5" name="__codelineno-1-5"></a><span class="c1">#SBATCH --cpus-per-task 8</span>
<a href="#__codelineno-1-6" id="__codelineno-1-6" name="__codelineno-1-6"></a><span class="c1">#SBATCH --mem           4G</span>
<a href="#__codelineno-1-7" id="__codelineno-1-7" name="__codelineno-1-7"></a><span class="c1">#SBATCH --time          1:00:00</span>
<a href="#__codelineno-1-8" id="__codelineno-1-8" name="__codelineno-1-8"></a>
<a href="#__codelineno-1-9" id="__codelineno-1-9" name="__codelineno-1-9"></a>module<span class="w"> </span>purge
<a href="#__codelineno-1-10" id="__codelineno-1-10" name="__codelineno-1-10"></a>module<span class="w"> </span>load<span class="w"> </span>SAMtools/1.16.1-GCC-11.3.0
<a href="#__codelineno-1-11" id="__codelineno-1-11" name="__codelineno-1-11"></a>
<a href="#__codelineno-1-12" id="__codelineno-1-12" name="__codelineno-1-12"></a><span class="nv">input_folder</span><span class="o">=</span>/home/zyang/pg_workshop/dataset_for_pg_workshop/12_genomes_for_NGS_simulation
<a href="#__codelineno-1-13" id="__codelineno-1-13" name="__codelineno-1-13"></a><span class="nv">output_folder</span><span class="o">=</span>/home/zyang/pg_workshop/graph_NGS/simu_NGS_data
<a href="#__codelineno-1-14" id="__codelineno-1-14" name="__codelineno-1-14"></a>
<a href="#__codelineno-1-15" id="__codelineno-1-15" name="__codelineno-1-15"></a><span class="k">for</span><span class="w"> </span>f<span class="w"> </span><span class="k">in</span><span class="w"> </span>NC_017518_6k.fa<span class="w"> </span>ST154_6k.fa<span class="w"> </span>ST154Sim_6k.fa<span class="w"> </span>ST41Sim_6k.fa<span class="w"> </span>ST42Sim_6k.fa
<a href="#__codelineno-1-16" id="__codelineno-1-16" name="__codelineno-1-16"></a>
<a href="#__codelineno-1-17" id="__codelineno-1-17" name="__codelineno-1-17"></a><span class="k">do</span>
<a href="#__codelineno-1-18" id="__codelineno-1-18" name="__codelineno-1-18"></a>
<a href="#__codelineno-1-19" id="__codelineno-1-19" name="__codelineno-1-19"></a><span class="nv">x</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="nv">$f</span><span class="w"> </span>.fa<span class="k">)</span>
<a href="#__codelineno-1-20" id="__codelineno-1-20" name="__codelineno-1-20"></a><span class="nb">echo</span><span class="w"> </span><span class="si">${</span><span class="nv">x</span><span class="si">}</span>
<a href="#__codelineno-1-21" id="__codelineno-1-21" name="__codelineno-1-21"></a>
<a href="#__codelineno-1-22" id="__codelineno-1-22" name="__codelineno-1-22"></a>wgsim<span class="w"> </span><span class="nv">$input_folder</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>.fa<span class="w"> </span>-N<span class="w"> </span><span class="m">1000000</span><span class="w"> </span>-1<span class="w"> </span><span class="m">150</span><span class="w"> </span>-2<span class="w"> </span><span class="m">150</span><span class="w">  </span>-e<span class="w"> </span><span class="m">0</span>.005<span class="w"> </span>-r<span class="w"> </span><span class="m">0</span><span class="w"> </span>-R<span class="w"> </span><span class="m">0</span><span class="w"> </span>-X<span class="w"> </span><span class="m">0</span><span class="w"> </span><span class="nv">$output_folder</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>.wgsim_er0.005.R1.fq<span class="w">  </span><span class="nv">$output_folder</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>.wgsim_er0.005.R2.fq
<a href="#__codelineno-1-23" id="__codelineno-1-23" name="__codelineno-1-23"></a>gzip<span class="w"> </span><span class="nv">$output_folder</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>.wgsim_er0.005.R1.fq
<a href="#__codelineno-1-24" id="__codelineno-1-24" name="__codelineno-1-24"></a>gzip<span class="w"> </span><span class="nv">$output_folder</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>.wgsim_er0.005.R2.fq
<a href="#__codelineno-1-25" id="__codelineno-1-25" name="__codelineno-1-25"></a>
<a href="#__codelineno-1-26" id="__codelineno-1-26" name="__codelineno-1-26"></a><span class="k">done</span>
</code></pre></div></p>
<h2 id="build-index-for-graph">build index for graph<a class="headerlink" href="#build-index-for-graph" title="Permanent link">¶</a></h2>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-2-1" id="__codelineno-2-1" name="__codelineno-2-1"></a>mkdir<span class="w"> </span>refs
<a href="#__codelineno-2-2" id="__codelineno-2-2" name="__codelineno-2-2"></a>
<a href="#__codelineno-2-3" id="__codelineno-2-3" name="__codelineno-2-3"></a><span class="c1"># copy graph to the refs work direvtory </span>
<a href="#__codelineno-2-4" id="__codelineno-2-4" name="__codelineno-2-4"></a>cp<span class="w"> </span>/home/zyang/pg_workshop/vg_deconstruct/4Sim_1K96.gfa<span class="w"> </span>/home/zyang/pg_workshop/graph_NGS/refs
<a href="#__codelineno-2-5" id="__codelineno-2-5" name="__codelineno-2-5"></a>
<a href="#__codelineno-2-6" id="__codelineno-2-6" name="__codelineno-2-6"></a><span class="c1">#make tem_dir</span>
<a href="#__codelineno-2-7" id="__codelineno-2-7" name="__codelineno-2-7"></a>mkdir<span class="w"> </span>/home/zyang/pg_workshop/graph_NGS/refs/temp_dir
</code></pre></div>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-3-1" id="__codelineno-3-1" name="__codelineno-3-1"></a><span class="ch">#!/bin.bash</span>
<a href="#__codelineno-3-2" id="__codelineno-3-2" name="__codelineno-3-2"></a>
<a href="#__codelineno-3-3" id="__codelineno-3-3" name="__codelineno-3-3"></a><span class="c1">#SBATCH --account       ga03793</span>
<a href="#__codelineno-3-4" id="__codelineno-3-4" name="__codelineno-3-4"></a><span class="c1">#SBATCH --job-name      build_index_for_4SimGraph</span>
<a href="#__codelineno-3-5" id="__codelineno-3-5" name="__codelineno-3-5"></a><span class="c1">#SBATCH --cpus-per-task 8</span>
<a href="#__codelineno-3-6" id="__codelineno-3-6" name="__codelineno-3-6"></a><span class="c1">#SBATCH --mem           4G</span>
<a href="#__codelineno-3-7" id="__codelineno-3-7" name="__codelineno-3-7"></a><span class="c1">#SBATCH --time          1:00:00</span>
<a href="#__codelineno-3-8" id="__codelineno-3-8" name="__codelineno-3-8"></a>
<a href="#__codelineno-3-9" id="__codelineno-3-9" name="__codelineno-3-9"></a>module<span class="w"> </span>purge
<a href="#__codelineno-3-10" id="__codelineno-3-10" name="__codelineno-3-10"></a>module<span class="w"> </span>load<span class="w"> </span>vg/1.46.0
<a href="#__codelineno-3-11" id="__codelineno-3-11" name="__codelineno-3-11"></a>
<a href="#__codelineno-3-12" id="__codelineno-3-12" name="__codelineno-3-12"></a><span class="nb">cd</span><span class="w"> </span>/home/zyang/pg_workshop/graph_NGS/refs
<a href="#__codelineno-3-13" id="__codelineno-3-13" name="__codelineno-3-13"></a><span class="nv">data</span><span class="o">=</span>/home/zyang/pg_workshop/graph_NGS/refs/*.gfa
<a href="#__codelineno-3-14" id="__codelineno-3-14" name="__codelineno-3-14"></a><span class="nv">tem_dir</span><span class="o">=</span>/home/zyang/pg_workshop/graph_NGS/refs/temp_dir
<a href="#__codelineno-3-15" id="__codelineno-3-15" name="__codelineno-3-15"></a>
<a href="#__codelineno-3-16" id="__codelineno-3-16" name="__codelineno-3-16"></a><span class="k">for</span><span class="w"> </span>f<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="nv">$data</span>
<a href="#__codelineno-3-17" id="__codelineno-3-17" name="__codelineno-3-17"></a><span class="k">do</span>
<a href="#__codelineno-3-18" id="__codelineno-3-18" name="__codelineno-3-18"></a>
<a href="#__codelineno-3-19" id="__codelineno-3-19" name="__codelineno-3-19"></a><span class="nv">x</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="nv">$f</span><span class="w"> </span>.gfa<span class="k">)</span>
<a href="#__codelineno-3-20" id="__codelineno-3-20" name="__codelineno-3-20"></a>
<a href="#__codelineno-3-21" id="__codelineno-3-21" name="__codelineno-3-21"></a><span class="c1">#convert the graph into 256 bp chunks, saving as vg format</span>
<a href="#__codelineno-3-22" id="__codelineno-3-22" name="__codelineno-3-22"></a>vg<span class="w"> </span>mod<span class="w"> </span>-X<span class="w"> </span><span class="m">256</span><span class="w"> </span><span class="si">${</span><span class="nv">x</span><span class="si">}</span>.gfa<span class="w"> </span>&gt;<span class="si">${</span><span class="nv">x</span><span class="si">}</span>_256.vg
<a href="#__codelineno-3-23" id="__codelineno-3-23" name="__codelineno-3-23"></a>
<a href="#__codelineno-3-24" id="__codelineno-3-24" name="__codelineno-3-24"></a><span class="c1">#build index of xg and gcsa index</span>
<a href="#__codelineno-3-25" id="__codelineno-3-25" name="__codelineno-3-25"></a>vg<span class="w"> </span>index<span class="w"> </span>-b<span class="w"> </span><span class="nv">$tem_dir</span><span class="w"> </span>-t<span class="w"> </span><span class="m">48</span><span class="w"> </span>-x<span class="w"> </span><span class="si">${</span><span class="nv">x</span><span class="si">}</span>_256.xg<span class="w"> </span>-g<span class="w"> </span><span class="si">${</span><span class="nv">x</span><span class="si">}</span>_256.gcsa<span class="w"> </span>-k<span class="w"> </span><span class="m">16</span><span class="w"> </span><span class="si">${</span><span class="nv">x</span><span class="si">}</span>_256.vg
<a href="#__codelineno-3-26" id="__codelineno-3-26" name="__codelineno-3-26"></a>
<a href="#__codelineno-3-27" id="__codelineno-3-27" name="__codelineno-3-27"></a><span class="c1">#small graph is ok without prunning, complex graph will need to prune first before generating index</span>
<a href="#__codelineno-3-28" id="__codelineno-3-28" name="__codelineno-3-28"></a>
<a href="#__codelineno-3-29" id="__codelineno-3-29" name="__codelineno-3-29"></a><span class="c1">### pruning: use -M if pruning fails</span>
<a href="#__codelineno-3-30" id="__codelineno-3-30" name="__codelineno-3-30"></a><span class="c1">#vg prune -u -m node-mapping.tmp -t 48 -k 24 ${x}_256.vg &gt; ${x}_256_chopped.vg</span>
<a href="#__codelineno-3-31" id="__codelineno-3-31" name="__codelineno-3-31"></a>
<a href="#__codelineno-3-32" id="__codelineno-3-32" name="__codelineno-3-32"></a><span class="c1">#vg index ${x}_256_chopped.vg -x ${x}_256_chopped.xg</span>
<a href="#__codelineno-3-33" id="__codelineno-3-33" name="__codelineno-3-33"></a><span class="c1">### gcsa index</span>
<a href="#__codelineno-3-34" id="__codelineno-3-34" name="__codelineno-3-34"></a><span class="c1">#vg index -b $tem_dir -t 48  -g ${x}_256_chopped.gcsa  ${x}_256_chopped.vg</span>
<a href="#__codelineno-3-35" id="__codelineno-3-35" name="__codelineno-3-35"></a><span class="k">done</span>
</code></pre></div>
<h2 id="vg-map-ngs-to-graph">vg map NGS to graph<a class="headerlink" href="#vg-map-ngs-to-graph" title="Permanent link">¶</a></h2>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-4-1" id="__codelineno-4-1" name="__codelineno-4-1"></a>mkdir<span class="w"> </span>/home/zyang/pg_workshop/graph_NGS/graph_based_mapping
<a href="#__codelineno-4-2" id="__codelineno-4-2" name="__codelineno-4-2"></a><span class="sb">```</span>bash
<a href="#__codelineno-4-3" id="__codelineno-4-3" name="__codelineno-4-3"></a>
<a href="#__codelineno-4-4" id="__codelineno-4-4" name="__codelineno-4-4"></a><span class="sb">```</span>bash
<a href="#__codelineno-4-5" id="__codelineno-4-5" name="__codelineno-4-5"></a><span class="c1">#!/bin.bash</span>
<a href="#__codelineno-4-6" id="__codelineno-4-6" name="__codelineno-4-6"></a>
<a href="#__codelineno-4-7" id="__codelineno-4-7" name="__codelineno-4-7"></a><span class="c1">#SBATCH --account       ga03793</span>
<a href="#__codelineno-4-8" id="__codelineno-4-8" name="__codelineno-4-8"></a><span class="c1">#SBATCH --job-name      vgmap_5e_4Sim</span>
<a href="#__codelineno-4-9" id="__codelineno-4-9" name="__codelineno-4-9"></a><span class="c1">#SBATCH --cpus-per-task 8</span>
<a href="#__codelineno-4-10" id="__codelineno-4-10" name="__codelineno-4-10"></a><span class="c1">#SBATCH --mem           4G</span>
<a href="#__codelineno-4-11" id="__codelineno-4-11" name="__codelineno-4-11"></a><span class="c1">#SBATCH --time          24:00:00</span>
<a href="#__codelineno-4-12" id="__codelineno-4-12" name="__codelineno-4-12"></a>
<a href="#__codelineno-4-13" id="__codelineno-4-13" name="__codelineno-4-13"></a>module<span class="w"> </span>purge
<a href="#__codelineno-4-14" id="__codelineno-4-14" name="__codelineno-4-14"></a>module<span class="w"> </span>load<span class="w"> </span>vg/1.46.0
<a href="#__codelineno-4-15" id="__codelineno-4-15" name="__codelineno-4-15"></a>
<a href="#__codelineno-4-16" id="__codelineno-4-16" name="__codelineno-4-16"></a>
<a href="#__codelineno-4-17" id="__codelineno-4-17" name="__codelineno-4-17"></a><span class="nv">data</span><span class="o">=</span>/home/zyang/pg_workshop/graph_NGS/simu_NGS_data/*R1.fq.gz
<a href="#__codelineno-4-18" id="__codelineno-4-18" name="__codelineno-4-18"></a><span class="nv">input_folder</span><span class="o">=</span>/home/zyang/pg_workshop/graph_NGS/simu_NGS_data
<a href="#__codelineno-4-19" id="__codelineno-4-19" name="__codelineno-4-19"></a><span class="nv">output</span><span class="o">=</span>/home/zyang/pg_workshop/graph_NGS/graph_based_mapping
<a href="#__codelineno-4-20" id="__codelineno-4-20" name="__codelineno-4-20"></a>
<a href="#__codelineno-4-21" id="__codelineno-4-21" name="__codelineno-4-21"></a>
<a href="#__codelineno-4-22" id="__codelineno-4-22" name="__codelineno-4-22"></a><span class="nv">index</span><span class="o">=</span>/home/zyang/pg_workshop/graph_NGS/refs/4Sim_1K96_256.gcsa
<a href="#__codelineno-4-23" id="__codelineno-4-23" name="__codelineno-4-23"></a><span class="nv">basename</span><span class="o">=</span>/home/zyang/pg_workshop/graph_NGS/refs/4Sim_1K96_256
<a href="#__codelineno-4-24" id="__codelineno-4-24" name="__codelineno-4-24"></a>
<a href="#__codelineno-4-25" id="__codelineno-4-25" name="__codelineno-4-25"></a>
<a href="#__codelineno-4-26" id="__codelineno-4-26" name="__codelineno-4-26"></a><span class="k">for</span><span class="w"> </span>f<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="nv">$data</span>
<a href="#__codelineno-4-27" id="__codelineno-4-27" name="__codelineno-4-27"></a><span class="k">do</span>
<a href="#__codelineno-4-28" id="__codelineno-4-28" name="__codelineno-4-28"></a>
<a href="#__codelineno-4-29" id="__codelineno-4-29" name="__codelineno-4-29"></a>
<a href="#__codelineno-4-30" id="__codelineno-4-30" name="__codelineno-4-30"></a><span class="nv">x</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="nv">$f</span><span class="w"> </span>R1.fq.gz<span class="k">)</span>
<a href="#__codelineno-4-31" id="__codelineno-4-31" name="__codelineno-4-31"></a><span class="nb">echo</span><span class="w"> </span><span class="si">${</span><span class="nv">x</span><span class="si">}</span>
<a href="#__codelineno-4-32" id="__codelineno-4-32" name="__codelineno-4-32"></a>
<a href="#__codelineno-4-33" id="__codelineno-4-33" name="__codelineno-4-33"></a><span class="nv">read1</span><span class="o">=</span><span class="si">${</span><span class="nv">x</span><span class="si">}</span>R1.fq.gz
<a href="#__codelineno-4-34" id="__codelineno-4-34" name="__codelineno-4-34"></a><span class="nv">read2</span><span class="o">=</span><span class="k">$(</span><span class="nb">echo</span><span class="w"> </span><span class="nv">$read1</span><span class="p">|</span>sed<span class="w"> </span><span class="s1">'s/R1.fq.gz/R2.fq.gz/'</span><span class="k">)</span>
<a href="#__codelineno-4-35" id="__codelineno-4-35" name="__codelineno-4-35"></a>
<a href="#__codelineno-4-36" id="__codelineno-4-36" name="__codelineno-4-36"></a><span class="nb">echo</span><span class="w"> </span><span class="nv">$read2</span>
<a href="#__codelineno-4-37" id="__codelineno-4-37" name="__codelineno-4-37"></a>
<a href="#__codelineno-4-38" id="__codelineno-4-38" name="__codelineno-4-38"></a><span class="c1">#map paired reads using vg map</span>
<a href="#__codelineno-4-39" id="__codelineno-4-39" name="__codelineno-4-39"></a>vg<span class="w"> </span>map<span class="w"> </span>-t<span class="w"> </span><span class="m">20</span><span class="w">  </span>-d<span class="w"> </span><span class="nv">$basename</span><span class="w"> </span>-g<span class="w"> </span><span class="nv">$index</span><span class="w">  </span>-f<span class="w"> </span><span class="nv">$input_folder</span>/<span class="nv">$read1</span><span class="w"> </span>-f<span class="w"> </span><span class="nv">$input_folder</span>/<span class="nv">$read2</span><span class="w"> </span>-N<span class="w"> </span><span class="nv">$x</span><span class="w">  </span>&gt;<span class="w"> </span><span class="nv">$output</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>vgmap_4Sim.gam
<a href="#__codelineno-4-40" id="__codelineno-4-40" name="__codelineno-4-40"></a><span class="c1">#vg stats to check the mapping statistics</span>
<a href="#__codelineno-4-41" id="__codelineno-4-41" name="__codelineno-4-41"></a>vg<span class="w"> </span>stats<span class="w"> </span>-a<span class="w">  </span><span class="nv">$output</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>vgmap_4Sim.gam<span class="w">  </span>&gt;<span class="nv">$output</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>vgmap_4Sim_stats
<a href="#__codelineno-4-42" id="__codelineno-4-42" name="__codelineno-4-42"></a>
<a href="#__codelineno-4-43" id="__codelineno-4-43" name="__codelineno-4-43"></a><span class="k">done</span>
</code></pre></div>
<h2 id="genotying-known-variants">genotying known variants<a class="headerlink" href="#genotying-known-variants" title="Permanent link">¶</a></h2>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-5-1" id="__codelineno-5-1" name="__codelineno-5-1"></a>mkdir<span class="w"> </span>/home/zyang/pg_workshop/graph_NGS/vgmap_12e_sim4_allR10S3_typing
</code></pre></div>
<p>generate snarls of graph
<div class="highlight"><pre><span></span><code><a href="#__codelineno-6-1" id="__codelineno-6-1" name="__codelineno-6-1"></a><span class="ch">#!/bin.bash</span>
<a href="#__codelineno-6-2" id="__codelineno-6-2" name="__codelineno-6-2"></a>
<a href="#__codelineno-6-3" id="__codelineno-6-3" name="__codelineno-6-3"></a><span class="c1">#SBATCH --account       ga03793</span>
<a href="#__codelineno-6-4" id="__codelineno-6-4" name="__codelineno-6-4"></a><span class="c1">#SBATCH --job-name      vgmap_generate_snarls</span>
<a href="#__codelineno-6-5" id="__codelineno-6-5" name="__codelineno-6-5"></a><span class="c1">#SBATCH --cpus-per-task 8</span>
<a href="#__codelineno-6-6" id="__codelineno-6-6" name="__codelineno-6-6"></a><span class="c1">#SBATCH --mem           4G</span>
<a href="#__codelineno-6-7" id="__codelineno-6-7" name="__codelineno-6-7"></a><span class="c1">#SBATCH --time          1:00:00</span>
<a href="#__codelineno-6-8" id="__codelineno-6-8" name="__codelineno-6-8"></a>
<a href="#__codelineno-6-9" id="__codelineno-6-9" name="__codelineno-6-9"></a>module<span class="w"> </span>purge
<a href="#__codelineno-6-10" id="__codelineno-6-10" name="__codelineno-6-10"></a>module<span class="w"> </span>load<span class="w"> </span>vg/1.46.0
<a href="#__codelineno-6-11" id="__codelineno-6-11" name="__codelineno-6-11"></a>
<a href="#__codelineno-6-12" id="__codelineno-6-12" name="__codelineno-6-12"></a>
<a href="#__codelineno-6-13" id="__codelineno-6-13" name="__codelineno-6-13"></a><span class="nv">input_folder</span><span class="o">=</span>/home/zyang/pg_workshop/graph_NGS/refs
<a href="#__codelineno-6-14" id="__codelineno-6-14" name="__codelineno-6-14"></a><span class="nv">output_folder</span><span class="o">=</span>/home/zyang/pg_workshop/graph_NGS/refs
<a href="#__codelineno-6-15" id="__codelineno-6-15" name="__codelineno-6-15"></a>
<a href="#__codelineno-6-16" id="__codelineno-6-16" name="__codelineno-6-16"></a>vg<span class="w"> </span>snarls<span class="w"> </span><span class="nv">$input_folder</span>/4Sim_1K96_256.xg<span class="w"> </span>&gt;<span class="w"> </span><span class="nv">$output_folder</span>/4Sim_1K96_256.xg.snarls
</code></pre></div></p>
<p>genotyping 
<div class="highlight"><pre><span></span><code><a href="#__codelineno-7-1" id="__codelineno-7-1" name="__codelineno-7-1"></a><span class="ch">#!/bin.bash</span>
<a href="#__codelineno-7-2" id="__codelineno-7-2" name="__codelineno-7-2"></a>
<a href="#__codelineno-7-3" id="__codelineno-7-3" name="__codelineno-7-3"></a><span class="c1">#SBATCH --account       ga03793</span>
<a href="#__codelineno-7-4" id="__codelineno-7-4" name="__codelineno-7-4"></a><span class="c1">#SBATCH --job-name      5e_vgmap_genotying</span>
<a href="#__codelineno-7-5" id="__codelineno-7-5" name="__codelineno-7-5"></a><span class="c1">#SBATCH --cpus-per-task 8</span>
<a href="#__codelineno-7-6" id="__codelineno-7-6" name="__codelineno-7-6"></a><span class="c1">#SBATCH --mem           4G</span>
<a href="#__codelineno-7-7" id="__codelineno-7-7" name="__codelineno-7-7"></a><span class="c1">#SBATCH --time          24:00:00</span>
<a href="#__codelineno-7-8" id="__codelineno-7-8" name="__codelineno-7-8"></a>
<a href="#__codelineno-7-9" id="__codelineno-7-9" name="__codelineno-7-9"></a>module<span class="w"> </span>purge
<a href="#__codelineno-7-10" id="__codelineno-7-10" name="__codelineno-7-10"></a>module<span class="w"> </span>load<span class="w"> </span>vg/1.46.0
<a href="#__codelineno-7-11" id="__codelineno-7-11" name="__codelineno-7-11"></a>
<a href="#__codelineno-7-12" id="__codelineno-7-12" name="__codelineno-7-12"></a><span class="nv">data_gam</span><span class="o">=</span>/home/zyang/pg_workshop/graph_NGS/graph_based_mapping/*.wgsim_er0.005.vgmap_4Sim.gam
<a href="#__codelineno-7-13" id="__codelineno-7-13" name="__codelineno-7-13"></a><span class="nv">input</span><span class="o">=</span>/home/zyang/pg_workshop/graph_NGS/graph_based_mapping
<a href="#__codelineno-7-14" id="__codelineno-7-14" name="__codelineno-7-14"></a><span class="nv">output</span><span class="o">=</span>/home/zyang/pg_workshop/graph_NGS/vgmap_12e_sim4_allR10S3_typing
<a href="#__codelineno-7-15" id="__codelineno-7-15" name="__codelineno-7-15"></a><span class="nv">graph_xg</span><span class="o">=</span>/home/zyang/pg_workshop/graph_NGS/refs/4Sim_1K96_256.xg
<a href="#__codelineno-7-16" id="__codelineno-7-16" name="__codelineno-7-16"></a><span class="nv">snarls_file</span><span class="o">=</span>/home/zyang/pg_workshop/graph_NGS/refs/4Sim_1K96_256.xg.snarls
<a href="#__codelineno-7-17" id="__codelineno-7-17" name="__codelineno-7-17"></a>
<a href="#__codelineno-7-18" id="__codelineno-7-18" name="__codelineno-7-18"></a>
<a href="#__codelineno-7-19" id="__codelineno-7-19" name="__codelineno-7-19"></a>
<a href="#__codelineno-7-20" id="__codelineno-7-20" name="__codelineno-7-20"></a><span class="c1">#compute snarls</span>
<a href="#__codelineno-7-21" id="__codelineno-7-21" name="__codelineno-7-21"></a><span class="c1">#vg snarls $graph_xg &gt;$snarls_file</span>
<a href="#__codelineno-7-22" id="__codelineno-7-22" name="__codelineno-7-22"></a>
<a href="#__codelineno-7-23" id="__codelineno-7-23" name="__codelineno-7-23"></a><span class="k">for</span><span class="w"> </span>f<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="nv">$data_gam</span>
<a href="#__codelineno-7-24" id="__codelineno-7-24" name="__codelineno-7-24"></a><span class="k">do</span>
<a href="#__codelineno-7-25" id="__codelineno-7-25" name="__codelineno-7-25"></a>
<a href="#__codelineno-7-26" id="__codelineno-7-26" name="__codelineno-7-26"></a><span class="nv">x</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="nv">$f</span><span class="w"> </span>.wgsim_er0.005.vgmap_4Sim.gam<span class="k">)</span>
<a href="#__codelineno-7-27" id="__codelineno-7-27" name="__codelineno-7-27"></a><span class="nb">echo</span><span class="w"> </span><span class="si">${</span><span class="nv">x</span><span class="si">}</span>
<a href="#__codelineno-7-28" id="__codelineno-7-28" name="__codelineno-7-28"></a>
<a href="#__codelineno-7-29" id="__codelineno-7-29" name="__codelineno-7-29"></a>
<a href="#__codelineno-7-30" id="__codelineno-7-30" name="__codelineno-7-30"></a><span class="c1">#Calculate the surpport reads ingoring mapping and base quality &lt;5</span>
<a href="#__codelineno-7-31" id="__codelineno-7-31" name="__codelineno-7-31"></a><span class="c1">#vg pack -t 48 -x $graph_xg -g $input/${x}.wgsim_er0.005.vgmap_4Sim.gam -Q 5 -o $output/${x}vgmap_Sim4_256_aln.pack</span>
<a href="#__codelineno-7-32" id="__codelineno-7-32" name="__codelineno-7-32"></a>
<a href="#__codelineno-7-33" id="__codelineno-7-33" name="__codelineno-7-33"></a><span class="c1">#Calculate the surpport reads</span>
<a href="#__codelineno-7-34" id="__codelineno-7-34" name="__codelineno-7-34"></a>vg<span class="w"> </span>pack<span class="w"> </span>-t<span class="w"> </span><span class="m">12</span><span class="w"> </span>-x<span class="w"> </span><span class="nv">$graph_xg</span><span class="w"> </span>-g<span class="w"> </span><span class="nv">$input</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>.wgsim_er0.005.vgmap_4Sim.gam<span class="w"> </span>-o<span class="w"> </span><span class="nv">$output</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>vgmap_sim4_256_aln.pack
<a href="#__codelineno-7-35" id="__codelineno-7-35" name="__codelineno-7-35"></a>
<a href="#__codelineno-7-36" id="__codelineno-7-36" name="__codelineno-7-36"></a><span class="c1">#call variant using the same coordinates and including reference calls (for following compare)</span>
<a href="#__codelineno-7-37" id="__codelineno-7-37" name="__codelineno-7-37"></a>vg<span class="w"> </span>call<span class="w"> </span>-t<span class="w"> </span><span class="m">12</span><span class="w"> </span>-m<span class="w"> </span><span class="m">3</span>,10<span class="w"> </span><span class="nv">$graph_xg</span><span class="w"> </span>-k<span class="w"> </span><span class="nv">$output</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>vgmap_sim4_256_aln.pack<span class="w"> </span>-r<span class="w"> </span><span class="nv">$snarls_file</span><span class="w"> </span>-a<span class="w">  </span>&gt;<span class="nv">$output</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>vgmap_sim4_256_aln.pack_allR10S3.vcf
<a href="#__codelineno-7-38" id="__codelineno-7-38" name="__codelineno-7-38"></a>
<a href="#__codelineno-7-39" id="__codelineno-7-39" name="__codelineno-7-39"></a><span class="k">done</span>
</code></pre></div></p>
<h2 id="noverl-variant-calling-using-graph-reference">noverl variant calling using graph reference<a class="headerlink" href="#noverl-variant-calling-using-graph-reference" title="Permanent link">¶</a></h2>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-8-1" id="__codelineno-8-1" name="__codelineno-8-1"></a>mkdir<span class="w"> </span>/home/zyang/pg_workshop/graph_NGS/vgmap_5e_sim4_allR10S3_novelcalling<span class="w"> </span>
</code></pre></div>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-9-1" id="__codelineno-9-1" name="__codelineno-9-1"></a><span class="ch">#!/bin.bash</span>
<a href="#__codelineno-9-2" id="__codelineno-9-2" name="__codelineno-9-2"></a>
<a href="#__codelineno-9-3" id="__codelineno-9-3" name="__codelineno-9-3"></a><span class="c1">#SBATCH --account       ga03793</span>
<a href="#__codelineno-9-4" id="__codelineno-9-4" name="__codelineno-9-4"></a><span class="c1">#SBATCH --job-name      5e_vgmap_novelvariant_calling</span>
<a href="#__codelineno-9-5" id="__codelineno-9-5" name="__codelineno-9-5"></a><span class="c1">#SBATCH --cpus-per-task 8</span>
<a href="#__codelineno-9-6" id="__codelineno-9-6" name="__codelineno-9-6"></a><span class="c1">#SBATCH --mem           4G</span>
<a href="#__codelineno-9-7" id="__codelineno-9-7" name="__codelineno-9-7"></a><span class="c1">#SBATCH --time          24:00:00</span>
<a href="#__codelineno-9-8" id="__codelineno-9-8" name="__codelineno-9-8"></a>
<a href="#__codelineno-9-9" id="__codelineno-9-9" name="__codelineno-9-9"></a>module<span class="w"> </span>purge
<a href="#__codelineno-9-10" id="__codelineno-9-10" name="__codelineno-9-10"></a>module<span class="w"> </span>load<span class="w"> </span>vg/1.46.0
<a href="#__codelineno-9-11" id="__codelineno-9-11" name="__codelineno-9-11"></a>
<a href="#__codelineno-9-12" id="__codelineno-9-12" name="__codelineno-9-12"></a><span class="nv">data_gam</span><span class="o">=</span>/home/zyang/pg_workshop/graph_NGS/graph_based_mapping/*.wgsim_er0.005.vgmap_4Sim.gam
<a href="#__codelineno-9-13" id="__codelineno-9-13" name="__codelineno-9-13"></a><span class="nv">input</span><span class="o">=</span>/home/zyang/pg_workshop/graph_NGS/graph_based_mapping
<a href="#__codelineno-9-14" id="__codelineno-9-14" name="__codelineno-9-14"></a><span class="nv">output</span><span class="o">=</span>/home/zyang/pg_workshop/graph_NGS/vgmap_5e_sim4_allR10S3_novelcalling
<a href="#__codelineno-9-15" id="__codelineno-9-15" name="__codelineno-9-15"></a><span class="nv">graph_vg</span><span class="o">=</span>/home/zyang/pg_workshop/graph_NGS/refs/4Sim_1K96_256.vg
<a href="#__codelineno-9-16" id="__codelineno-9-16" name="__codelineno-9-16"></a><span class="nv">graph_xg</span><span class="o">=</span>/home/zyang/pg_workshop/graph_NGS/refs/4Sim_1K96_256.xg
<a href="#__codelineno-9-17" id="__codelineno-9-17" name="__codelineno-9-17"></a>
<a href="#__codelineno-9-18" id="__codelineno-9-18" name="__codelineno-9-18"></a>
<a href="#__codelineno-9-19" id="__codelineno-9-19" name="__codelineno-9-19"></a><span class="c1">#compute snarls</span>
<a href="#__codelineno-9-20" id="__codelineno-9-20" name="__codelineno-9-20"></a><span class="c1">#vg snarls $graph_xg &gt;$output/${graph_xg}.snarls</span>
<a href="#__codelineno-9-21" id="__codelineno-9-21" name="__codelineno-9-21"></a>
<a href="#__codelineno-9-22" id="__codelineno-9-22" name="__codelineno-9-22"></a><span class="k">for</span><span class="w"> </span>f<span class="w"> </span><span class="k">in</span><span class="w"> </span><span class="nv">$data_gam</span>
<a href="#__codelineno-9-23" id="__codelineno-9-23" name="__codelineno-9-23"></a><span class="k">do</span>
<a href="#__codelineno-9-24" id="__codelineno-9-24" name="__codelineno-9-24"></a>
<a href="#__codelineno-9-25" id="__codelineno-9-25" name="__codelineno-9-25"></a><span class="nv">x</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="nv">$f</span><span class="w"> </span>.wgsim_er0.005.vgmap_4Sim.gam<span class="k">)</span>
<a href="#__codelineno-9-26" id="__codelineno-9-26" name="__codelineno-9-26"></a><span class="nb">echo</span><span class="w"> </span><span class="si">${</span><span class="nv">x</span><span class="si">}</span>
<a href="#__codelineno-9-27" id="__codelineno-9-27" name="__codelineno-9-27"></a>
<a href="#__codelineno-9-28" id="__codelineno-9-28" name="__codelineno-9-28"></a>
<a href="#__codelineno-9-29" id="__codelineno-9-29" name="__codelineno-9-29"></a><span class="c1">#in order to also consider novel variants from the reads, use the augmented graph and gam (as created in the "Augmentation" example using vg augment -A)</span>
<a href="#__codelineno-9-30" id="__codelineno-9-30" name="__codelineno-9-30"></a><span class="c1">#Augment augment the graph with all variation from the GAM, saving to aug.vg</span>
<a href="#__codelineno-9-31" id="__codelineno-9-31" name="__codelineno-9-31"></a><span class="c1">### augment the graph with all variation from the GAM except</span>
<a href="#__codelineno-9-32" id="__codelineno-9-32" name="__codelineno-9-32"></a><span class="c1">### that implied by soft clips, saving to aug.vg</span>
<a href="#__codelineno-9-33" id="__codelineno-9-33" name="__codelineno-9-33"></a><span class="c1">### *aug-gam contains the same reads as aln.gam but mapped to aug.vg</span>
<a href="#__codelineno-9-34" id="__codelineno-9-34" name="__codelineno-9-34"></a>
<a href="#__codelineno-9-35" id="__codelineno-9-35" name="__codelineno-9-35"></a>vg<span class="w"> </span>augment<span class="w"> </span>-t<span class="w"> </span><span class="m">12</span><span class="w"> </span><span class="nv">$graph_vg</span><span class="w"> </span><span class="nv">$input</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>.wgsim_er0.005.vgmap_4Sim.gam<span class="w"> </span>-A<span class="w"> </span><span class="nv">$output</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>nofilt_aug.gam<span class="w"> </span>&gt;<span class="nv">$output</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>nofilt_aug.vg
<a href="#__codelineno-9-36" id="__codelineno-9-36" name="__codelineno-9-36"></a>
<a href="#__codelineno-9-37" id="__codelineno-9-37" name="__codelineno-9-37"></a><span class="c1">#index the augmented graph</span>
<a href="#__codelineno-9-38" id="__codelineno-9-38" name="__codelineno-9-38"></a>vg<span class="w"> </span>index<span class="w"> </span>-t<span class="w"> </span><span class="m">12</span><span class="w"> </span><span class="nv">$output</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>nofilt_aug.vg<span class="w"> </span>-x<span class="w"> </span><span class="nv">$output</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>nofilt_aug.xg
<a href="#__codelineno-9-39" id="__codelineno-9-39" name="__codelineno-9-39"></a>
<a href="#__codelineno-9-40" id="__codelineno-9-40" name="__codelineno-9-40"></a><span class="c1">## Compute the all read support from the augmented gam</span>
<a href="#__codelineno-9-41" id="__codelineno-9-41" name="__codelineno-9-41"></a>vg<span class="w"> </span>pack<span class="w"> </span>-t<span class="w"> </span><span class="m">12</span><span class="w"> </span>-x<span class="w"> </span><span class="nv">$output</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>nofilt_aug.xg<span class="w"> </span>-g<span class="w"> </span><span class="nv">$output</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>nofilt_aug.gam<span class="w">  </span>-o<span class="w"> </span><span class="nv">$output</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>nofilt_aug_allR.pack
<a href="#__codelineno-9-42" id="__codelineno-9-42" name="__codelineno-9-42"></a>
<a href="#__codelineno-9-43" id="__codelineno-9-43" name="__codelineno-9-43"></a>
<a href="#__codelineno-9-44" id="__codelineno-9-44" name="__codelineno-9-44"></a><span class="c1">#call variant</span>
<a href="#__codelineno-9-45" id="__codelineno-9-45" name="__codelineno-9-45"></a>vg<span class="w"> </span>call<span class="w"> </span>-t<span class="w"> </span><span class="m">12</span><span class="w"> </span>-m<span class="w"> </span><span class="m">3</span>,10<span class="w"> </span><span class="nv">$output</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>nofilt_aug.xg<span class="w"> </span>-k<span class="w"> </span><span class="nv">$output</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>nofilt_aug_allR.pack<span class="w"> </span>&gt;<span class="nv">$output</span>/<span class="si">${</span><span class="nv">x</span><span class="si">}</span>nofilt_aug_allR.pack.vcf
<a href="#__codelineno-9-46" id="__codelineno-9-46" name="__codelineno-9-46"></a>
<a href="#__codelineno-9-47" id="__codelineno-9-47" name="__codelineno-9-47"></a><span class="c1">#call variant snarl using the same coordinate</span>
<a href="#__codelineno-9-48" id="__codelineno-9-48" name="__codelineno-9-48"></a><span class="c1">#vg call -t 48 -m 3,10 $output/${x}nofilt_aug.xg -k $output/${x}nofilt_aug_allR.pack -a &gt;$output/${x}nofilt_aug_allR.pack_snarls.vcf</span>
<a href="#__codelineno-9-49" id="__codelineno-9-49" name="__codelineno-9-49"></a>
<a href="#__codelineno-9-50" id="__codelineno-9-50" name="__codelineno-9-50"></a><span class="k">done</span>
</code></pre></div>
</article>
</div>
</div>
<button class="md-top md-icon" data-md-component="top" hidden="" type="button">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg>
            Back to top
          </button>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top"], "search": "../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
<script src="../assets/javascripts/bundle.fac441b0.min.js"></script>
<script src="../javascripts/mathjax.js"></script>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>