
<!DOCTYPE html>

<html class="no-js" lang="en">
<head>
<meta charset="utf-8"/>
<meta content="width=device-width,initial-scale=1" name="viewport"/>
<meta content="Genomics Aotearoa" name="author"/>
<link href="https://genomicsaotearoa.github.io/Pangenome-Graphs-Workshop/preparing_data_files/" rel="canonical"/>
<link href="../pangenome_graph_construction/" rel="prev"/>
<link href="../simulate_and_compare/" rel="next"/>
<link href="../theme_figures/nesi_ga.png" rel="icon"/>
<meta content="mkdocs-1.4.3, mkdocs-material-9.1.9" name="generator"/>
<title>Preparing required data files with simuG tool - Pangenome Graphs</title>
<link href="../assets/stylesheets/main.85bb2934.min.css" rel="stylesheet"/>
<link href="../assets/stylesheets/palette.a6bdf11c.min.css" rel="stylesheet"/>
<link crossorigin="" href="https://fonts.gstatic.com" rel="preconnect"/>
<link href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&amp;display=fallback" rel="stylesheet"/>
<style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
<link href="../stylesheets/extra.css" rel="stylesheet"/>
<script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
<link href="../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
            html.glightbox-open { overflow: initial; height: 100%; }
            .gslide-title { margin-top: 0px; user-select: text; }
            .gslide-desc { color: #666; user-select: text; }
            .gslide-image img { background: white; }
            
                .gscrollbar-fixer { padding-right: 15px; }
                .gdesc-inner { font-size: 0.75rem; }
                body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
                body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
                body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
                </style><script src="../assets/javascripts/glightbox.min.js"></script></head>
<body data-md-color-accent="indigo" data-md-color-primary="indigo" data-md-color-scheme="default" dir="ltr">
<script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
<input autocomplete="off" class="md-toggle" data-md-toggle="drawer" id="__drawer" type="checkbox"/>
<input autocomplete="off" class="md-toggle" data-md-toggle="search" id="__search" type="checkbox"/>
<label class="md-overlay" for="__drawer"></label>
<div data-md-component="skip">
<a class="md-skip" href="#preparing-required-data-files-with-simug-tool">
          Skip to content
        </a>
</div>
<div data-md-component="announce">
</div>
<header class="md-header md-header--shadow" data-md-component="header">
<nav aria-label="Header" class="md-header__inner md-grid">
<a aria-label="Pangenome Graphs" class="md-header__button md-logo" data-md-component="logo" href=".." title="Pangenome Graphs">
<img alt="logo" src="../theme_figures/nesi_ga.png"/>
</a>
<label class="md-header__button md-icon" for="__drawer">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"></path></svg>
</label>
<div class="md-header__title" data-md-component="header-title">
<div class="md-header__ellipsis">
<div class="md-header__topic">
<span class="md-ellipsis">
            Pangenome Graphs
          </span>
</div>
<div class="md-header__topic" data-md-component="header-topic">
<span class="md-ellipsis">
            
              Preparing required data files with simuG tool
            
          </span>
</div>
</div>
</div>
<form class="md-header__option" data-md-component="palette">
<input aria-label="Switch to light mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_1" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_3" hidden="" title="Switch to light mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"></path></svg>
</label>
<input aria-label="Switch to dark mode" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: light)" data-md-color-primary="indigo" data-md-color-scheme="default" id="__palette_2" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_1" hidden="" title="Switch to dark mode">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
<input aria-label="Switch to system preference" class="md-option" data-md-color-accent="indigo" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-primary="indigo" data-md-color-scheme="slate" id="__palette_3" name="__palette" type="radio"/>
<label class="md-header__button md-icon" for="__palette_2" hidden="" title="Switch to system preference">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"></path></svg>
</label>
</form>
<label class="md-header__button md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
</label>
<div class="md-search" data-md-component="search" role="dialog">
<label class="md-search__overlay" for="__search"></label>
<div class="md-search__inner" role="search">
<form class="md-search__form" name="search">
<input aria-label="Search" autocapitalize="off" autocomplete="off" autocorrect="off" class="md-search__input" data-md-component="search-query" name="query" placeholder="Search" required="" spellcheck="false" type="text"/>
<label class="md-search__icon md-icon" for="__search">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"></path></svg>
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"></path></svg>
</label>
<nav aria-label="Search" class="md-search__options">
<button aria-label="Clear" class="md-search__icon md-icon" tabindex="-1" title="Clear" type="reset">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"></path></svg>
</button>
</nav>
</form>
<div class="md-search__output">
<div class="md-search__scrollwrap" data-md-scrollfix="">
<div class="md-search-result" data-md-component="search-result">
<div class="md-search-result__meta">
            Initializing search
          </div>
<ol class="md-search-result__list" role="presentation"></ol>
</div>
</div>
</div>
</div>
</div>
<div class="md-header__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/Pangenome-Graphs-Workshop" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/Pangenome-Graphs-Workshop
  </div>
</a>
</div>
</nav>
</header>
<div class="md-container" data-md-component="container">
<main class="md-main" data-md-component="main">
<div class="md-main__inner md-grid">
<div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Navigation" class="md-nav md-nav--primary" data-md-level="0">
<label class="md-nav__title" for="__drawer">
<a aria-label="Pangenome Graphs" class="md-nav__button md-logo" data-md-component="logo" href=".." title="Pangenome Graphs">
<img alt="logo" src="../theme_figures/nesi_ga.png"/>
</a>
    Pangenome Graphs
  </label>
<div class="md-nav__source">
<a class="md-source" data-md-component="source" href="https://github.com/GenomicsAotearoa/Pangenome-Graphs-Workshop" title="Go to repository">
<div class="md-source__icon md-icon">
<svg viewbox="0 0 448 512" xmlns="http://www.w3.org/2000/svg"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"></path></svg>
</div>
<div class="md-source__repository">
    GenomicsAotearoa/Pangenome-Graphs-Workshop
  </div>
</a>
</div>
<ul class="md-nav__list" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="..">
        Pangenome Graphs
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../Graph_based_variant_calling_for_NGS/">
        NGS data analysis used graph as reference on Nesi
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../Intro_1/">
        Pangenome Graphs Workshop
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../Intro_2/">
        Intro 2
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../mapping_reads/">
        Mapping Reads using `bwa mem` (Linear Method)
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../pangenome_graph_construction/">
        pangenome graph construction with PGGB on Nesi
      </a>
</li>
<li class="md-nav__item md-nav__item--active">
<input class="md-nav__toggle md-toggle" id="__toc" type="checkbox"/>
<label class="md-nav__link md-nav__link--active" for="__toc">
          Preparing required data files with simuG tool
          <span class="md-nav__icon md-icon"></span>
</label>
<a class="md-nav__link md-nav__link--active" href="./">
        Preparing required data files with simuG tool
      </a>
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#1-setting-up-the-tool">
    1. Setting up the tool
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#2-data-source">
    2. Data source
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#3-simulating-sequences-with-known-genetics-variants-counts-using-simug">
    3. Simulating sequences with known genetics variants counts using simuG
  </a>
<nav aria-label="3. Simulating sequences with known genetics variants counts using simuG" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#i-snp5000">
    i. SNP=5000
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#ii-indel5000-indel-ratio-11">
    ii. INDEL=5000 (IN:DEL ratio 1:1)
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#iii-snp4000-and-indel4000-indel-ratio-14">
    iii. SNP=4000 and INDEL=4000 (IN:DEL ratio 1:4)
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#iv-snp4000-indel4000-indel-ratio-14-and-4-inversions">
    iv. SNP=4000, INDEL=4000 (IN:DEL ratio 1:4) and 4 inversions
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#iv-snp4000-indel4000-indel-ratio-14-and-4-copy-number-variations">
    iv. SNP=4000, INDEL=4000 (IN:DEL ratio 1:4) and 4 copy number variations
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#4-concatenate-all-files-and-make-an-index">
    4. Concatenate all files and make an index
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../simulate_and_compare/">
        Simulate and compare
      </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="../vc_position_based_comparison/">
        How to compare 2 VCF files based on the exact variant position?
      </a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc">
<div class="md-sidebar__scrollwrap">
<div class="md-sidebar__inner">
<nav aria-label="Table of contents" class="md-nav md-nav--secondary">
<label class="md-nav__title" for="__toc">
<span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
<ul class="md-nav__list" data-md-component="toc" data-md-scrollfix="">
<li class="md-nav__item">
<a class="md-nav__link" href="#1-setting-up-the-tool">
    1. Setting up the tool
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#2-data-source">
    2. Data source
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#3-simulating-sequences-with-known-genetics-variants-counts-using-simug">
    3. Simulating sequences with known genetics variants counts using simuG
  </a>
<nav aria-label="3. Simulating sequences with known genetics variants counts using simuG" class="md-nav">
<ul class="md-nav__list">
<li class="md-nav__item">
<a class="md-nav__link" href="#i-snp5000">
    i. SNP=5000
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#ii-indel5000-indel-ratio-11">
    ii. INDEL=5000 (IN:DEL ratio 1:1)
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#iii-snp4000-and-indel4000-indel-ratio-14">
    iii. SNP=4000 and INDEL=4000 (IN:DEL ratio 1:4)
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#iv-snp4000-indel4000-indel-ratio-14-and-4-inversions">
    iv. SNP=4000, INDEL=4000 (IN:DEL ratio 1:4) and 4 inversions
  </a>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#iv-snp4000-indel4000-indel-ratio-14-and-4-copy-number-variations">
    iv. SNP=4000, INDEL=4000 (IN:DEL ratio 1:4) and 4 copy number variations
  </a>
</li>
</ul>
</nav>
</li>
<li class="md-nav__item">
<a class="md-nav__link" href="#4-concatenate-all-files-and-make-an-index">
    4. Concatenate all files and make an index
  </a>
</li>
</ul>
</nav>
</div>
</div>
</div>
<div class="md-content" data-md-component="content">
<article class="md-content__inner md-typeset">
<h1 id="preparing-required-data-files-with-simug-tool">Preparing required data files with simuG tool<a class="headerlink" href="#preparing-required-data-files-with-simug-tool" title="Permanent link">¶</a></h1>
<p><a href="https://github.com/yjx1217/simuG">simuG</a> is a light-weighted tool which can be used to simulate sequences with pre-defined number of random genomic variants against a reference sequence (Please refer the <a href="https://github.com/yjx1217/simuG">Git Page</a> for more details). The following example demostrates generating three new sequence using known SNP and INDEL counts.</p>
<h3 id="1-setting-up-the-tool">1. Setting up the tool<a class="headerlink" href="#1-setting-up-the-tool" title="Permanent link">¶</a></h3>
<p>Please refer the installation instruction in the <a href="https://github.com/yjx1217/simuG">Git Page</a>.
<div class="highlight"><pre><span></span><code><a href="#__codelineno-0-1" id="__codelineno-0-1" name="__codelineno-0-1"></a>git<span class="w"> </span>clone<span class="w"> </span>https://github.com/yjx1217/simuG.git
<a href="#__codelineno-0-2" id="__codelineno-0-2" name="__codelineno-0-2"></a><span class="nb">cd</span><span class="w"> </span>simuG
<a href="#__codelineno-0-3" id="__codelineno-0-3" name="__codelineno-0-3"></a>perl<span class="w"> </span>simuG.pl<span class="w"> </span>-h
<a href="#__codelineno-0-4" id="__codelineno-0-4" name="__codelineno-0-4"></a>perl<span class="w"> </span>vcf2model.pl<span class="w"> </span>-h
</code></pre></div></p>
<h3 id="2-data-source">2. Data source<a class="headerlink" href="#2-data-source" title="Permanent link">¶</a></h3>
<p>You can download the reference sequence (Neisseria gonorrhoeae FA 1090 genome assembly) from the site <a href="https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000191525.1/">Genome assembly ASM19152v1</a> and get the file <em>ncbi_dataset/data/GCF_000191525.1/GCF_000191525.1_ASM19152v1_genomic.fna</em> from he zip file GCF_000191525.1.zip. Rename it as "ref.fa" to refere easily in next steps. </p>
<p>In Unix environment you can use <code>curl</code>. </p>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-1-1" id="__codelineno-1-1" name="__codelineno-1-1"></a>$<span class="w"> </span>curl<span class="w"> </span>-OJX<span class="w"> </span>GET<span class="w"> </span><span class="s2">"https://api.ncbi.nlm.nih.gov/datasets/v2alpha/genome/accession/GCF_000191525.1/download?include_annotation_type=GENOME_FASTA,GENOME_GFF,RNA_FASTA,CDS_FASTA,PROT_FASTA,SEQUENCE_REPORT&amp;filename=GCF_000191525.1.zip"</span><span class="w"> </span>-H<span class="w"> </span><span class="s2">"Accept: application/zip"</span>
<a href="#__codelineno-1-2" id="__codelineno-1-2" name="__codelineno-1-2"></a><span class="w">  </span>%<span class="w"> </span>Total<span class="w">    </span>%<span class="w"> </span>Received<span class="w"> </span>%<span class="w"> </span>Xferd<span class="w">  </span>Average<span class="w"> </span>Speed<span class="w">   </span>Time<span class="w">    </span>Time<span class="w">     </span>Time<span class="w">  </span>Current
<a href="#__codelineno-1-3" id="__codelineno-1-3" name="__codelineno-1-3"></a><span class="w">                                 </span>Dload<span class="w">  </span>Upload<span class="w">   </span>Total<span class="w">   </span>Spent<span class="w">    </span>Left<span class="w">  </span>Speed
<a href="#__codelineno-1-4" id="__codelineno-1-4" name="__codelineno-1-4"></a><span class="m">100</span><span class="w"> </span>1807k<span class="w">    </span><span class="m">0</span><span class="w"> </span>1807k<span class="w">    </span><span class="m">0</span><span class="w">     </span><span class="m">0</span><span class="w">   </span>294k<span class="w">      </span><span class="m">0</span><span class="w"> </span>--:--:--<span class="w">  </span><span class="m">0</span>:00:06<span class="w"> </span>--:--:--<span class="w">  </span>421k
<a href="#__codelineno-1-5" id="__codelineno-1-5" name="__codelineno-1-5"></a>
<a href="#__codelineno-1-6" id="__codelineno-1-6" name="__codelineno-1-6"></a>$<span class="w"> </span>ls
<a href="#__codelineno-1-7" id="__codelineno-1-7" name="__codelineno-1-7"></a>GCF_000191525.1.zip
<a href="#__codelineno-1-8" id="__codelineno-1-8" name="__codelineno-1-8"></a>
<a href="#__codelineno-1-9" id="__codelineno-1-9" name="__codelineno-1-9"></a>$<span class="w"> </span>unzip<span class="w"> </span>GCF_000191525.1.zip<span class="w"> </span>
<a href="#__codelineno-1-10" id="__codelineno-1-10" name="__codelineno-1-10"></a>Archive:<span class="w">  </span>GCF_000191525.1.zip
<a href="#__codelineno-1-11" id="__codelineno-1-11" name="__codelineno-1-11"></a><span class="w">  </span>inflating:<span class="w"> </span>README.md<span class="w">               </span>
<a href="#__codelineno-1-12" id="__codelineno-1-12" name="__codelineno-1-12"></a><span class="w">  </span>inflating:<span class="w"> </span>ncbi_dataset/data/assembly_data_report.jsonl<span class="w">  </span>
<a href="#__codelineno-1-13" id="__codelineno-1-13" name="__codelineno-1-13"></a><span class="w">  </span>inflating:<span class="w"> </span>ncbi_dataset/data/GCF_000191525.1/GCF_000191525.1_ASM19152v1_genomic.fna<span class="w">  </span>
<a href="#__codelineno-1-14" id="__codelineno-1-14" name="__codelineno-1-14"></a><span class="w">  </span>inflating:<span class="w"> </span>ncbi_dataset/data/GCF_000191525.1/genomic.gff<span class="w">  </span>
<a href="#__codelineno-1-15" id="__codelineno-1-15" name="__codelineno-1-15"></a><span class="w">  </span>inflating:<span class="w"> </span>ncbi_dataset/data/GCF_000191525.1/cds_from_genomic.fna<span class="w">  </span>
<a href="#__codelineno-1-16" id="__codelineno-1-16" name="__codelineno-1-16"></a><span class="w">  </span>inflating:<span class="w"> </span>ncbi_dataset/data/GCF_000191525.1/protein.faa<span class="w">  </span>
<a href="#__codelineno-1-17" id="__codelineno-1-17" name="__codelineno-1-17"></a><span class="w">  </span>inflating:<span class="w"> </span>ncbi_dataset/data/GCF_000191525.1/sequence_report.jsonl<span class="w">  </span>
<a href="#__codelineno-1-18" id="__codelineno-1-18" name="__codelineno-1-18"></a><span class="w">  </span>inflating:<span class="w"> </span>ncbi_dataset/data/dataset_catalog.json<span class="w">  </span>
<a href="#__codelineno-1-19" id="__codelineno-1-19" name="__codelineno-1-19"></a>
<a href="#__codelineno-1-20" id="__codelineno-1-20" name="__codelineno-1-20"></a>$<span class="w"> </span>cp<span class="w"> </span>ncbi_dataset/data/GCF_000191525.1/GCF_000191525.1_ASM19152v1_genomic.fna<span class="w"> </span>./
<a href="#__codelineno-1-21" id="__codelineno-1-21" name="__codelineno-1-21"></a>
<a href="#__codelineno-1-22" id="__codelineno-1-22" name="__codelineno-1-22"></a>$<span class="w"> </span>head<span class="w"> </span>GCF_000191525.1_ASM19152v1_genomic.fna<span class="w"> </span>
<a href="#__codelineno-1-23" id="__codelineno-1-23" name="__codelineno-1-23"></a>&gt;NC_017518.1<span class="w"> </span>Neisseria<span class="w"> </span>meningitidis<span class="w"> </span>NZ-05/33,<span class="w"> </span><span class="nb">complete</span><span class="w"> </span>sequence
<a href="#__codelineno-1-24" id="__codelineno-1-24" name="__codelineno-1-24"></a>TTCGGCTTAAACCTTATCCATATCCAAACGCATAACCGTAACCCATTCACCGTTATGGAAATGTCGCCCGACAACCGCCC
<a href="#__codelineno-1-25" id="__codelineno-1-25" name="__codelineno-1-25"></a>AGCCGAATGATTCATAAAATATTTGCACATCAGGCGTATAAAGATACAAGAACTTTATCCCCAGCGAACGCGCTGCGCCT
<a href="#__codelineno-1-26" id="__codelineno-1-26" name="__codelineno-1-26"></a>ATGCAGTGGGCGACCAGCCTCCTGCCAATGCCTTTTCCGCGATATTCAGGTAAAACAAAGACATCACCCAACCAATATTC
<a href="#__codelineno-1-27" id="__codelineno-1-27" name="__codelineno-1-27"></a>ATACTGTGGAAAACTTTCCATATCATGCCGCTTGACCGTAGCCGAACCCAACAGGGTTCCGGAATCATCCACAGCCGCAA
<a href="#__codelineno-1-28" id="__codelineno-1-28" name="__codelineno-1-28"></a>AAGCCAGCGGCAGTTCGTCATCCTTCAAACACCTGCCGTAATAGGCATGAATCTTATCCACAGAAGACCACGGTTCAAAT
<a href="#__codelineno-1-29" id="__codelineno-1-29" name="__codelineno-1-29"></a>CCGTGCCACTCCTCAAACAACGCCTGAACCAACCTGCCGATATGCCCGGCTTTCAGCCGTGTAATGAAAACAGTATTGTC
<a href="#__codelineno-1-30" id="__codelineno-1-30" name="__codelineno-1-30"></a>CACAAAGAGGGAATTCATCGGTCAATTCCCCGACGCTTTCGTTCCCCCTGCGCCGTAAACCGCATTCCAAGCATAGTCCA
<a href="#__codelineno-1-31" id="__codelineno-1-31" name="__codelineno-1-31"></a>AACGCACTCCGATTTGCCTCAGCTCTTCAGCCTGCCGGGCTTTTTGCGCCATTGCTGCAGGAATTTCCGCTTCCAAACGG
<a href="#__codelineno-1-32" id="__codelineno-1-32" name="__codelineno-1-32"></a>GCGATGTCTGCCTGAGCCGTCTGCAAACGCCGGCGCGCATCTTCCAAATCCGACTGCATCCCGATGATTTTTCCGTCCAG
</code></pre></div>
<h3 id="3-simulating-sequences-with-known-genetics-variants-counts-using-simug">3. Simulating sequences with known genetics variants counts using simuG<a class="headerlink" href="#3-simulating-sequences-with-known-genetics-variants-counts-using-simug" title="Permanent link">¶</a></h3>
<h4 id="i-snp5000">i. SNP=5000<a class="headerlink" href="#i-snp5000" title="Permanent link">¶</a></h4>
<p><div class="highlight"><pre><span></span><code><a href="#__codelineno-2-1" id="__codelineno-2-1" name="__codelineno-2-1"></a>$ perl simuG.pl -refseq GCF_000191525.1_ASM19152v1_genomic.fna -snp_count 5000 -prefix Simulation_SNP_5000 -seed 112345678900
<a href="#__codelineno-2-2" id="__codelineno-2-2" name="__codelineno-2-2"></a>
<a href="#__codelineno-2-3" id="__codelineno-2-3" name="__codelineno-2-3"></a>$ ls -1sh Simulation_SNP_5000*
<a href="#__codelineno-2-4" id="__codelineno-2-4" name="__codelineno-2-4"></a>512K Simulation_SNP_5000.refseq2simseq.map.txt
<a href="#__codelineno-2-5" id="__codelineno-2-5" name="__codelineno-2-5"></a>768K Simulation_SNP_5000.refseq2simseq.SNP.vcf
<a href="#__codelineno-2-6" id="__codelineno-2-6" name="__codelineno-2-6"></a>2.3M Simulation_SNP_5000.simseq.genome.fa
<a href="#__codelineno-2-7" id="__codelineno-2-7" name="__codelineno-2-7"></a>
<a href="#__codelineno-2-8" id="__codelineno-2-8" name="__codelineno-2-8"></a>$ less Simulation_SNP_5000.simseq.genome.fa 
<a href="#__codelineno-2-9" id="__codelineno-2-9" name="__codelineno-2-9"></a>&gt;NC_017518.1
<a href="#__codelineno-2-10" id="__codelineno-2-10" name="__codelineno-2-10"></a>TTCGGCTTAAACCTTATCCATATCCAAACGCATAACCGTAACCCATTCACCGTTATGGAAATGTCGCCCGACAACCGCCCAGCCGAATGATTCATAAAATATTTGCACATCAGGCGTATAAAGATACAAGAACTTTATCCCCAGCGAACGCGCTGCGCCTATGCAGTGGGCGACCAGCCTCCTGCCAATGCCTTTTCCGCGATATTCAGGTAAAACAAAGACATCAC
</code></pre></div>
After the simulation the Chromosome number is same. Let's rename it be more clear in the next steps. We can use <code>sed -i</code> for this. 
<div class="highlight"><pre><span></span><code><a href="#__codelineno-3-1" id="__codelineno-3-1" name="__codelineno-3-1"></a>$ sed -i '/&gt;/ s/\(.*\)/\1_SNP_5000/' Simulation_SNP_5000.simseq.genome.fa
<a href="#__codelineno-3-2" id="__codelineno-3-2" name="__codelineno-3-2"></a>
<a href="#__codelineno-3-3" id="__codelineno-3-3" name="__codelineno-3-3"></a>$ less Simulation_SNP_5000.simseq.genome.fa
<a href="#__codelineno-3-4" id="__codelineno-3-4" name="__codelineno-3-4"></a>&gt;NC_017518.1_SNP_5000
<a href="#__codelineno-3-5" id="__codelineno-3-5" name="__codelineno-3-5"></a>TTCGGCTTAAACCTTATCCATATCCAAACGCATAACCGTAACCCATTCACCGTTATGGAAATGTCGCCCGACAACCGCCCAGCCGAATGATTCATAAAATATTTGCACATCAGGCGTATAAAGATACAAGAACTTTATCCCCAGCGAACGCGCTGCGCCTATGCAGTGGGCGACCAGCCTCCTGCCAATGCCTTTTCCGCGATATTCAGGTAAAACAAAGACATCAC
</code></pre></div></p>
<h4 id="ii-indel5000-indel-ratio-11">ii. INDEL=5000 (IN:DEL ratio 1:1)<a class="headerlink" href="#ii-indel5000-indel-ratio-11" title="Permanent link">¶</a></h4>
<p><div class="highlight"><pre><span></span><code><a href="#__codelineno-4-1" id="__codelineno-4-1" name="__codelineno-4-1"></a>$<span class="w"> </span>perl<span class="w"> </span>simuG.pl<span class="w"> </span>-refseq<span class="w"> </span>GCF_000191525.1_ASM19152v1_genomic.fna<span class="w"> </span>-indel_count<span class="w"> </span><span class="m">5000</span><span class="w"> </span>-prefix<span class="w"> </span>Simulation_INDEL_5000<span class="w"> </span>-seed<span class="w"> </span><span class="m">212345678900</span>
<a href="#__codelineno-4-2" id="__codelineno-4-2" name="__codelineno-4-2"></a>
<a href="#__codelineno-4-3" id="__codelineno-4-3" name="__codelineno-4-3"></a>$<span class="w"> </span>ls<span class="w"> </span>-1tr<span class="w"> </span>Simulation_INDEL_5000*
<a href="#__codelineno-4-4" id="__codelineno-4-4" name="__codelineno-4-4"></a>Simulation_INDEL_5000.refseq2simseq.INDEL.vcf
<a href="#__codelineno-4-5" id="__codelineno-4-5" name="__codelineno-4-5"></a>Simulation_INDEL_5000.refseq2simseq.map.txt
<a href="#__codelineno-4-6" id="__codelineno-4-6" name="__codelineno-4-6"></a>Simulation_INDEL_5000.simseq.genome.fa
<a href="#__codelineno-4-7" id="__codelineno-4-7" name="__codelineno-4-7"></a>
<a href="#__codelineno-4-8" id="__codelineno-4-8" name="__codelineno-4-8"></a>$<span class="w"> </span>sed<span class="w"> </span>-i<span class="w"> </span><span class="s1">'/&gt;/ s/\(.*\)/\1_INDEL_5000/'</span><span class="w"> </span>Simulation_INDEL_5000.simseq.genome.fa
<a href="#__codelineno-4-9" id="__codelineno-4-9" name="__codelineno-4-9"></a>
<a href="#__codelineno-4-10" id="__codelineno-4-10" name="__codelineno-4-10"></a>$<span class="w"> </span>less<span class="w"> </span>Simulation_INDEL_5000.simseq.genome.fa
<a href="#__codelineno-4-11" id="__codelineno-4-11" name="__codelineno-4-11"></a>&gt;NC_017518.1_INDEL_5000
<a href="#__codelineno-4-12" id="__codelineno-4-12" name="__codelineno-4-12"></a>TTCGGCTTAAACCTTATCCATATCCAAACGCATAACCGTAACCCATTCACCGTTATGGAAATGTCGCCCGACAACCGCCCAGCCGAATGATTCATAAAATATTTGCACATCAGGCGTATAAAGATACAAGAACTTTATCCCCAGCGAACGCGCTGCGCCTATGCAGTGGGCGACCAGCCTCCTGCCAATGCCTTTTCCGCGATATTCAGGTAAAACAAAGACATCACC
</code></pre></div>
<em>note: Remeber to use different seeds for each simulation</em></p>
<h4 id="iii-snp4000-and-indel4000-indel-ratio-14">iii. SNP=4000 and INDEL=4000 (IN:DEL ratio 1:4)<a class="headerlink" href="#iii-snp4000-and-indel4000-indel-ratio-14" title="Permanent link">¶</a></h4>
<div class="highlight"><pre><span></span><code><a href="#__codelineno-5-1" id="__codelineno-5-1" name="__codelineno-5-1"></a>$<span class="w"> </span>perl<span class="w"> </span>simuG.pl<span class="w"> </span>-refseq<span class="w"> </span>GCF_000191525.1_ASM19152v1_genomic.fna<span class="w"> </span>-snp_count<span class="w"> </span><span class="m">4000</span><span class="w"> </span>-indel_count<span class="w"> </span><span class="m">4000</span><span class="w"> </span>-ins_del_ratio<span class="w"> </span><span class="m">0</span>.25<span class="w"> </span>-prefix<span class="w"> </span>Simulation_SNP_4000_INDEL_4000<span class="w"> </span>-seed<span class="w"> </span><span class="m">312345678900</span>
<a href="#__codelineno-5-2" id="__codelineno-5-2" name="__codelineno-5-2"></a>
<a href="#__codelineno-5-3" id="__codelineno-5-3" name="__codelineno-5-3"></a>$<span class="w"> </span>ls<span class="w"> </span>-1tr<span class="w"> </span>Simulation_SNP_4000_INDEL_4000*
<a href="#__codelineno-5-4" id="__codelineno-5-4" name="__codelineno-5-4"></a>Simulation_SNP_4000_INDEL_4000.refseq2simseq.SNP.vcf
<a href="#__codelineno-5-5" id="__codelineno-5-5" name="__codelineno-5-5"></a>Simulation_SNP_4000_INDEL_4000.refseq2simseq.INDEL.vcf
<a href="#__codelineno-5-6" id="__codelineno-5-6" name="__codelineno-5-6"></a>Simulation_SNP_4000_INDEL_4000.refseq2simseq.map.txt
<a href="#__codelineno-5-7" id="__codelineno-5-7" name="__codelineno-5-7"></a>Simulation_SNP_4000_INDEL_4000.simseq.genome.fa
<a href="#__codelineno-5-8" id="__codelineno-5-8" name="__codelineno-5-8"></a>
<a href="#__codelineno-5-9" id="__codelineno-5-9" name="__codelineno-5-9"></a>$<span class="w"> </span>sed<span class="w"> </span>-i<span class="w"> </span><span class="s1">'/&gt;/ s/\(.*\)/\1_SNP_4000_INDEL_4000/'</span><span class="w"> </span>Simulation_SNP_4000_INDEL_4000.simseq.genome.fa
<a href="#__codelineno-5-10" id="__codelineno-5-10" name="__codelineno-5-10"></a>
<a href="#__codelineno-5-11" id="__codelineno-5-11" name="__codelineno-5-11"></a>$<span class="w"> </span>less<span class="w"> </span>Simulation_SNP_4000_INDEL_4000.simseq.genome.fa
<a href="#__codelineno-5-12" id="__codelineno-5-12" name="__codelineno-5-12"></a>&gt;NC_017518.1_SNP_4000_INDEL_4000
<a href="#__codelineno-5-13" id="__codelineno-5-13" name="__codelineno-5-13"></a>TTCGGCTTAAACCTTATCCATATCCAAACGCATAACCGTAACCCATTCACCGTTATGGAAATGTCGCCCGACAACCGCCCAGCCGAATGATTCATAAAATATTTGCACATCAGGCGTATAAAGATACAAGAACTTTATCCCCAGCGAACGCGCTGCGCCTATGCAGTGGGCGACCAGCCTCCTGCCAATGCCTTGTCCGCGATATTCAGGTAAAACAAAGACATCACCCAAC
</code></pre></div>
<h4 id="iv-snp4000-indel4000-indel-ratio-14-and-4-inversions">iv. SNP=4000, INDEL=4000 (IN:DEL ratio 1:4) and 4 inversions<a class="headerlink" href="#iv-snp4000-indel4000-indel-ratio-14-and-4-inversions" title="Permanent link">¶</a></h4>
<p>To have a more complex sample we'll apply 4 inversions to the above simulated sequence. 
<div class="highlight"><pre><span></span><code><a href="#__codelineno-6-1" id="__codelineno-6-1" name="__codelineno-6-1"></a>$<span class="w"> </span>perl<span class="w"> </span>simuG.pl<span class="w"> </span>-refseq<span class="w"> </span>Simulation_SNP_4000_INDEL_4000.simseq.genome.fa<span class="w"> </span>-inversion_count<span class="w"> </span><span class="m">4</span><span class="w"> </span>-inversion_min_size<span class="w"> </span><span class="m">50000</span><span class="w"> </span>-prefix<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_INV_4<span class="w"> </span>-seed<span class="w"> </span><span class="m">412345678900</span>
<a href="#__codelineno-6-2" id="__codelineno-6-2" name="__codelineno-6-2"></a>
<a href="#__codelineno-6-3" id="__codelineno-6-3" name="__codelineno-6-3"></a>$<span class="w"> </span>ls<span class="w"> </span>-1tr<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_INV_4*
<a href="#__codelineno-6-4" id="__codelineno-6-4" name="__codelineno-6-4"></a>Simulation_SNP_4000_INDEL_4000_INV_4.refseq2simseq.inversion.vcf
<a href="#__codelineno-6-5" id="__codelineno-6-5" name="__codelineno-6-5"></a>Simulation_SNP_4000_INDEL_4000_INV_4.refseq2simseq.map.txt
<a href="#__codelineno-6-6" id="__codelineno-6-6" name="__codelineno-6-6"></a>Simulation_SNP_4000_INDEL_4000_INV_4.simseq.genome.fa
<a href="#__codelineno-6-7" id="__codelineno-6-7" name="__codelineno-6-7"></a>
<a href="#__codelineno-6-8" id="__codelineno-6-8" name="__codelineno-6-8"></a>$<span class="w"> </span>sed<span class="w"> </span>-i<span class="w"> </span><span class="s1">'/&gt;/ s/\(.*\)/\1_INV_4/'</span><span class="w"> </span>Simulation_SNP_4000_INDEL_4000_INV_4.simseq.genome.fa
<a href="#__codelineno-6-9" id="__codelineno-6-9" name="__codelineno-6-9"></a>
<a href="#__codelineno-6-10" id="__codelineno-6-10" name="__codelineno-6-10"></a>$<span class="w"> </span>less<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_INV_4.simseq.genome.fa
<a href="#__codelineno-6-11" id="__codelineno-6-11" name="__codelineno-6-11"></a>&gt;NC_017518.1_SNP_4000_INDEL_4000_INV_4
<a href="#__codelineno-6-12" id="__codelineno-6-12" name="__codelineno-6-12"></a>TTCGGCTTAAACCTTATCCATATCCAAACGCATAACCGTAACCCATTCACCGTTATGGAAATGTCGCCCGACAACCGCCCAGCCGAATGATTCATAAAATATTTGCACATCAGGCGTATAAAGATACAAGAACTTTATCCCCAGCGAACGCGCTGCGCCTATGCAGTGGGCGACCAGCCTCCTGCCAATGCCTTGTCCGCGATATTCAGGTAAAACAAAGACATCAC
</code></pre></div></p>
<h4 id="iv-snp4000-indel4000-indel-ratio-14-and-4-copy-number-variations">iv. SNP=4000, INDEL=4000 (IN:DEL ratio 1:4) and 4 copy number variations<a class="headerlink" href="#iv-snp4000-indel4000-indel-ratio-14-and-4-copy-number-variations" title="Permanent link">¶</a></h4>
<p>Let's apply 4 copy number variations also and make another simulated sequence. 
<div class="highlight"><pre><span></span><code><a href="#__codelineno-7-1" id="__codelineno-7-1" name="__codelineno-7-1"></a>$<span class="w"> </span>perl<span class="w"> </span>simuG.pl<span class="w"> </span>-refseq<span class="w"> </span>Simulation_SNP_4000_INDEL_4000.simseq.genome.fa<span class="w"> </span>-cnv_count<span class="w"> </span><span class="m">4</span><span class="w"> </span>-cnv_min_size<span class="w"> </span><span class="m">20000</span><span class="w"> </span>-prefix<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_CNV_4<span class="w"> </span>-seed<span class="w"> </span><span class="m">512345678900</span>
<a href="#__codelineno-7-2" id="__codelineno-7-2" name="__codelineno-7-2"></a>
<a href="#__codelineno-7-3" id="__codelineno-7-3" name="__codelineno-7-3"></a>$<span class="w"> </span>ls<span class="w"> </span>-1tr<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_CNV_4*
<a href="#__codelineno-7-4" id="__codelineno-7-4" name="__codelineno-7-4"></a>Simulation_SNP_4000_INDEL_4000_CNV_4.refseq2simseq.CNV.vcf
<a href="#__codelineno-7-5" id="__codelineno-7-5" name="__codelineno-7-5"></a>Simulation_SNP_4000_INDEL_4000_CNV_4.refseq2simseq.map.txt
<a href="#__codelineno-7-6" id="__codelineno-7-6" name="__codelineno-7-6"></a>Simulation_SNP_4000_INDEL_4000_CNV_4.simseq.genome.fa
<a href="#__codelineno-7-7" id="__codelineno-7-7" name="__codelineno-7-7"></a>
<a href="#__codelineno-7-8" id="__codelineno-7-8" name="__codelineno-7-8"></a>$<span class="w"> </span>sed<span class="w"> </span>-i<span class="w"> </span><span class="s1">'/&gt;/ s/\(.*\)/\1_CNV_4/'</span><span class="w"> </span>Simulation_SNP_4000_INDEL_4000_CNV_4.simseq.genome.fa
<a href="#__codelineno-7-9" id="__codelineno-7-9" name="__codelineno-7-9"></a>
<a href="#__codelineno-7-10" id="__codelineno-7-10" name="__codelineno-7-10"></a>$<span class="w"> </span>less<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_CNV_4.simseq.genome.fa
<a href="#__codelineno-7-11" id="__codelineno-7-11" name="__codelineno-7-11"></a>&gt;NC_017518.1_SNP_4000_INDEL_4000_CNV_4
<a href="#__codelineno-7-12" id="__codelineno-7-12" name="__codelineno-7-12"></a>TTCGGCTTAAACCTTATCCATATCCAAACGCATAACCGTAACCCATTCACCGTTATGGAAATGTCGCCCGACAACCGCCCAGCCGAATGATTCATAAAATATTTGCACATCAGGCGTATAAAGATACAAGAACTTTATCCCCAGCGAACGCGCTGCGCCTATGCAGTGGGCGACCAGCCTCCTGCCAATGCCTTGTCCGCGATATTCAGGTAAAACAAAGACATCAC
</code></pre></div></p>
<h3 id="4-concatenate-all-files-and-make-an-index">4. Concatenate all files and make an index<a class="headerlink" href="#4-concatenate-all-files-and-make-an-index" title="Permanent link">¶</a></h3>
<p>We'll concatenate all these files into one and make an index too
<div class="highlight"><pre><span></span><code><a href="#__codelineno-8-1" id="__codelineno-8-1" name="__codelineno-8-1"></a>$<span class="w"> </span>cat<span class="w"> </span>GCF_000191525.1_ASM19152v1_genomic.fna<span class="w"> </span>Simulation_SNP_5000.simseq.genome.fa<span class="w"> </span>Simulation_INDEL_5000.simseq.genome.fa<span class="w"> </span>Simulation_SNP_4000_INDEL_4000.simseq.genome.fa<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_INV_4.simseq.genome.fa<span class="w"> </span>Simulation_SNP_4000_INDEL_4000_CNV_4.simseq.genome.fa<span class="w"> </span>&gt;<span class="w"> </span>ASM19152v1_pgsim.fa
<a href="#__codelineno-8-2" id="__codelineno-8-2" name="__codelineno-8-2"></a>$<span class="w"> </span>module<span class="w"> </span>load<span class="w"> </span>SAMtools
<a href="#__codelineno-8-3" id="__codelineno-8-3" name="__codelineno-8-3"></a>$<span class="w"> </span>samtools<span class="w"> </span>faidx<span class="w"> </span>ASM19152v1_pgsim.fa<span class="w"> </span>
<a href="#__codelineno-8-4" id="__codelineno-8-4" name="__codelineno-8-4"></a>$<span class="w"> </span>cat<span class="w"> </span>ASM19152v1_pgsim.fa.fai<span class="w"> </span>
<a href="#__codelineno-8-5" id="__codelineno-8-5" name="__codelineno-8-5"></a>NC_017518.1<span class="w">     </span><span class="m">2248966</span><span class="w"> </span><span class="m">64</span><span class="w">      </span><span class="m">80</span><span class="w">      </span><span class="m">81</span>
<a href="#__codelineno-8-6" id="__codelineno-8-6" name="__codelineno-8-6"></a>NC_017518.1_SNP_5000<span class="w">    </span><span class="m">2248966</span><span class="w"> </span><span class="m">2277165</span><span class="w"> </span><span class="m">2248966</span><span class="w"> </span><span class="m">2248967</span>
<a href="#__codelineno-8-7" id="__codelineno-8-7" name="__codelineno-8-7"></a>NC_017518.1_INDEL_5000<span class="w">  </span><span class="m">2249048</span><span class="w"> </span><span class="m">4526156</span><span class="w"> </span><span class="m">2249048</span><span class="w"> </span><span class="m">2249049</span>
<a href="#__codelineno-8-8" id="__codelineno-8-8" name="__codelineno-8-8"></a>NC_017518.1_SNP_4000_INDEL_4000<span class="w"> </span><span class="m">2242147</span><span class="w"> </span><span class="m">6775238</span><span class="w"> </span><span class="m">2242147</span><span class="w"> </span><span class="m">2242148</span>
<a href="#__codelineno-8-9" id="__codelineno-8-9" name="__codelineno-8-9"></a>NC_017518.1_SNP_4000_INDEL_4000_INV_4<span class="w">   </span><span class="m">2242147</span><span class="w"> </span><span class="m">9017425</span><span class="w"> </span><span class="m">2242147</span><span class="w"> </span><span class="m">2242148</span>
<a href="#__codelineno-8-10" id="__codelineno-8-10" name="__codelineno-8-10"></a>NC_017518.1_SNP_4000_INDEL_4000_CNV_4<span class="w">   </span><span class="m">2415498</span><span class="w"> </span><span class="m">11259612</span><span class="w">        </span><span class="m">2415498</span><span class="w"> </span><span class="m">2415499</span>
</code></pre></div></p>
</article>
</div>
</div>
<button class="md-top md-icon" data-md-component="top" hidden="" type="button">
<svg viewbox="0 0 24 24" xmlns="http://www.w3.org/2000/svg"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"></path></svg>
            Back to top
          </button>
</main>
<footer class="md-footer">
<div class="md-footer-meta md-typeset">
<div class="md-footer-meta__inner md-grid">
<div class="md-copyright">
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" rel="noopener" target="_blank">
      Material for MkDocs
    </a>
</div>
</div>
</div>
</footer>
</div>
<div class="md-dialog" data-md-component="dialog">
<div class="md-dialog__inner md-typeset"></div>
</div>
<script id="__config" type="application/json">{"base": "..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top"], "search": "../assets/javascripts/workers/search.208ed371.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
<script src="../assets/javascripts/bundle.fac441b0.min.js"></script>
<script src="../javascripts/mathjax.js"></script>
<script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
<script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
<script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>