
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
        <meta name="author" content="Genomics Aotearoa">
      
      
        <link rel="canonical" href="https://genomicsaotearoa.github.io/Pangenome-Graphs-Workshop/Supplementary/supplementary_3_Short_reads/">
      
      
        <link rel="prev" href="../supplementary_2/">
      
      
      <link rel="icon" href="../../theme_figures/nesi_ga.png">
      <meta name="generator" content="mkdocs-1.4.3, mkdocs-material-9.1.19">
    
    
      
        <title>Short reads - Pangenome Graphs</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.eebd395e.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.ecc896b0.min.css">
      
      

    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Mukta:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Mukta";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../stylesheets/extra.css">
    
    <script>__md_scope=new URL("../..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
   <link href="../../assets/stylesheets/glightbox.min.css" rel="stylesheet"/><style>
        html.glightbox-open { overflow: initial; height: 100%; }
        .gslide-title { margin-top: 0px; user-select: text; }
        .gslide-desc { color: #666; user-select: text; }
        .gslide-image img { background: white; }
        
            .gscrollbar-fixer { padding-right: 15px; }
            .gdesc-inner { font-size: 0.75rem; }
            body[data-md-color-scheme="slate"] .gdesc-inner { background: var(--md-default-bg-color);}
            body[data-md-color-scheme="slate"] .gslide-title { color: var(--md-default-fg-color);}
            body[data-md-color-scheme="slate"] .gslide-desc { color: var(--md-default-fg-color);}
            </style> <script src="../../assets/javascripts/glightbox.min.js"></script></head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="custom" data-md-color-accent="indigo">
  
    
    
      <script>var palette=__md_get("__palette");if(palette&&"object"==typeof palette.color)for(var key of Object.keys(palette.color))document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#short-reads" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
<!-- Custom Banner -->
<div class="banner">
    <p style="color:rgb(250, 246, 246); display:inline;font-size: 160%;">This repository is in the early stages of development (Beta Version)</p>
</div>

<!-- Original Header -->


  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="Pangenome Graphs" class="md-header__button md-logo" aria-label="Pangenome Graphs" data-md-component="logo">
      
  <img src="../../theme_figures/nesi_ga.png" alt="logo">

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Pangenome Graphs
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Short reads
            
          </span>
        </div>
      </div>
    </div>
    
      
        <form class="md-header__option" data-md-component="palette">
          
            
            
            
            <input class="md-option" data-md-color-media="(prefers-color-scheme)" data-md-color-scheme="default" data-md-color-primary="custom" data-md-color-accent="indigo"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_1">
            
              <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_3" hidden>
                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="m14.3 16-.7-2h-3.2l-.7 2H7.8L11 7h2l3.2 9h-1.9M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69m-9.15 3.96h2.3L12 9l-1.15 3.65Z"/></svg>
              </label>
            
          
            
            
            
            <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="custom" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_2">
            
              <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_1" hidden>
                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a4 4 0 0 0-4 4 4 4 0 0 0 4 4 4 4 0 0 0 4-4 4 4 0 0 0-4-4m0 10a6 6 0 0 1-6-6 6 6 0 0 1 6-6 6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
              </label>
            
          
            
            
            
            <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="custom" data-md-color-accent="indigo"  aria-label="Switch to system preference"  type="radio" name="__palette" id="__palette_3">
            
              <label class="md-header__button md-icon" title="Switch to system preference" for="__palette_2" hidden>
                <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12c0-2.42-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6a6 6 0 0 1 6 6 6 6 0 0 1-6 6m8-9.31V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69Z"/></svg>
              </label>
            
          
        </form>
      
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/GenomicsAotearoa/Pangenome-Graphs-Workshop" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/Pangenome-Graphs-Workshop
  </div>
</a>
      </div>
    
  </nav>
  
</header>

    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="Pangenome Graphs" class="md-nav__button md-logo" aria-label="Pangenome Graphs" data-md-component="logo">
      
  <img src="../../theme_figures/nesi_ga.png" alt="logo">

    </a>
    Pangenome Graphs
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/GenomicsAotearoa/Pangenome-Graphs-Workshop" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.4.0 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2023 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GenomicsAotearoa/Pangenome-Graphs-Workshop
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../1_Introduction_to_pangenome_graphs/" class="md-nav__link">
        1. Introduction to pangenome graphs
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../2_Tools_and_setup/" class="md-nav__link">
        2. Tools & setup
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../3_Introduction_to_pggb/" class="md-nav__link">
        3. Introduction to PGGB
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../4_Running_pggb/" class="md-nav__link">
        4. Running PGGB
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../5_Evaluating_pggb_output/" class="md-nav__link">
        5. Evaluating output
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../6_Variant_calling/" class="md-nav__link">
        6. Variant calling
      </a>
    </li>
  

    
      
      
      

  
  
    
  
  
    
      
    
    <li class="md-nav__item md-nav__item--active md-nav__item--section md-nav__item--nested">
      
      
      
      
      <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_8" checked>
      
      
      
        <label class="md-nav__link" for="__nav_8" id="__nav_8_label" tabindex="0">
          Supplementary
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_8_label" aria-expanded="true">
        <label class="md-nav__title" for="__nav_8">
          <span class="md-nav__icon md-icon"></span>
          Supplementary
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../supplementary_1/" class="md-nav__link">
        NeSI HPC Auth.Factors Setup & Jupyter Login
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../supplementary_2/" class="md-nav__link">
        Jupyter Virtual Desktop
      </a>
    </li>
  

            
          
            
              
  
  
    
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          Short reads
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        Short reads
      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#vg-mapping-preliminaries" class="md-nav__link">
    vg mapping preliminaries
  </a>
  
    <nav class="md-nav" aria-label="vg mapping preliminaries">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#learning-objectives" class="md-nav__link">
    Learning objectives
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#build-index-for-graph" class="md-nav__link">
    Build index for graph
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#map-ngs-reads-to-graph" class="md-nav__link">
    Map NGS reads to graph
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#genotying-known-variants" class="md-nav__link">
    Genotying known variants
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#novel-variant-calling-using-graph-reference" class="md-nav__link">
    Novel variant calling using graph reference
  </a>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#vg-mapping-preliminaries" class="md-nav__link">
    vg mapping preliminaries
  </a>
  
    <nav class="md-nav" aria-label="vg mapping preliminaries">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#learning-objectives" class="md-nav__link">
    Learning objectives
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#build-index-for-graph" class="md-nav__link">
    Build index for graph
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#map-ngs-reads-to-graph" class="md-nav__link">
    Map NGS reads to graph
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#genotying-known-variants" class="md-nav__link">
    Genotying known variants
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#novel-variant-calling-using-graph-reference" class="md-nav__link">
    Novel variant calling using graph reference
  </a>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  

  
  


<h1 id="short-reads">Short reads<a class="headerlink" href="#short-reads" title="Permanent link">&para;</a></h1>
<p>NGS data analysis used graph as a reference </p>
<h2 id="vg-mapping-preliminaries">vg mapping preliminaries<a class="headerlink" href="#vg-mapping-preliminaries" title="Permanent link">&para;</a></h2>
<p>Although vg contains a number of tools for working with pangenome graphs, it is best-known for read mapping. This is ultimately what many of its users are interested in <code>vg</code> for. In fact, vg contains three mature short read mapping tools:</p>
<div class="admonition info">
<ul>
<li><code>vg map</code>: the original, highly accurate mapping algorithm</li>
<li><code>vg giraffe</code>: the much faster and still accurate haplotype-based mapping algorithm</li>
<li><code>vg mpmap</code>: the splice-aware RNA-seq mapping algorithm</li>
</ul>
</div>
<p>more details of vg can be found <a href="https://github.com/vgteam/vg">https://github.com/vgteam/vg</a></p>
<p>we use vg map in this workshop </p>
<h3 id="learning-objectives">Learning objectives<a class="headerlink" href="#learning-objectives" title="Permanent link">&para;</a></h3>
<div class="admonition quote">
<ul>
<li>Map NGS data to graph using vg map</li>
<li>Variant calling for NGS data against genome graph </li>
</ul>
</div>
<h2 id="build-index-for-graph">Build index for graph<a class="headerlink" href="#build-index-for-graph" title="Permanent link">&para;</a></h2>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-0-1" name="__codelineno-0-1" href="#__codelineno-0-1"></a>mkdir<span class="w"> </span>graph_NGS
<a id="__codelineno-0-2" name="__codelineno-0-2" href="#__codelineno-0-2"></a>
<a id="__codelineno-0-3" name="__codelineno-0-3" href="#__codelineno-0-3"></a><span class="c1">#copy graph to the graph reference (.gfa file) to work direvtory graph_NGS </span>
<a id="__codelineno-0-4" name="__codelineno-0-4" href="#__codelineno-0-4"></a>cp<span class="w"> </span>/home/<span class="nv">$your_home_dir</span>/pg_workshop/5NM*.gfa<span class="w"> </span>./home/<span class="nv">$your_home_dir</span>/pg_workshop/graph_NGS/5NM.gfa
<a id="__codelineno-0-5" name="__codelineno-0-5" href="#__codelineno-0-5"></a>
<a id="__codelineno-0-6" name="__codelineno-0-6" href="#__codelineno-0-6"></a><span class="nb">cd</span><span class="w"> </span>/home/<span class="nv">$your_home_dir</span>/pg_workshop/graph_NGS
</code></pre></div>
</div>
<p>Load the necessary modules for an example run.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-1-1" name="__codelineno-1-1" href="#__codelineno-1-1"></a>module<span class="w"> </span>purge
<a id="__codelineno-1-2" name="__codelineno-1-2" href="#__codelineno-1-2"></a>module<span class="w"> </span>load<span class="w"> </span>vg/1.46.0
</code></pre></div>
</div>
<p>Build the index.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-2-1" name="__codelineno-2-1" href="#__codelineno-2-1"></a>mkdir<span class="w"> </span>-p<span class="w"> </span><span class="si">${</span><span class="nv">temp_dir</span><span class="si">}</span>
<a id="__codelineno-2-2" name="__codelineno-2-2" href="#__codelineno-2-2"></a>
<a id="__codelineno-2-3" name="__codelineno-2-3" href="#__codelineno-2-3"></a><span class="c1"># Convert graph into 256 bp chunks, saving as vg format</span>
<a id="__codelineno-2-4" name="__codelineno-2-4" href="#__codelineno-2-4"></a>vg<span class="w"> </span>mod<span class="w"> </span>-X<span class="w"> </span><span class="m">256</span><span class="w"> </span>5NM.gfa<span class="w"> </span>&gt;<span class="w"> </span>5NM_256.vg
<a id="__codelineno-2-5" name="__codelineno-2-5" href="#__codelineno-2-5"></a>
<a id="__codelineno-2-6" name="__codelineno-2-6" href="#__codelineno-2-6"></a><span class="c1">#small graph is ok without prunning, complex graph will need to prune first before generating index</span>
<a id="__codelineno-2-7" name="__codelineno-2-7" href="#__codelineno-2-7"></a><span class="c1">#Build xg and gcsa index</span>
<a id="__codelineno-2-8" name="__codelineno-2-8" href="#__codelineno-2-8"></a>vg<span class="w"> </span>index<span class="w"> </span>-b<span class="w"> </span><span class="si">${</span><span class="nv">temp_dir</span><span class="si">}</span><span class="w"> </span>-t<span class="w"> </span><span class="m">4</span><span class="w"> </span>-x<span class="w"> </span>5NM_256.xg<span class="w"> </span>-g<span class="w"> </span>5NM_256.gcsa<span class="w"> </span>-k<span class="w"> </span><span class="m">16</span><span class="w"> </span>5NM_256.vg
<a id="__codelineno-2-9" name="__codelineno-2-9" href="#__codelineno-2-9"></a><span class="c1">### you may have run out of temporary disk space at temp_dir</span>
<a id="__codelineno-2-10" name="__codelineno-2-10" href="#__codelineno-2-10"></a>
<a id="__codelineno-2-11" name="__codelineno-2-11" href="#__codelineno-2-11"></a><span class="c1">### pruning: use -M if pruning fails</span>
<a id="__codelineno-2-12" name="__codelineno-2-12" href="#__codelineno-2-12"></a>vg<span class="w"> </span>prune<span class="w"> </span>-u<span class="w"> </span>-m<span class="w"> </span>node-mapping.tmp<span class="w"> </span>-t<span class="w"> </span><span class="m">4</span><span class="w"> </span>-k<span class="w"> </span><span class="m">24</span><span class="w"> </span>5NM_256.vg<span class="w"> </span>&gt;<span class="w"> </span>5NM_256_chopped.vg
<a id="__codelineno-2-13" name="__codelineno-2-13" href="#__codelineno-2-13"></a>
<a id="__codelineno-2-14" name="__codelineno-2-14" href="#__codelineno-2-14"></a>vg<span class="w"> </span>index<span class="w"> </span>5NM_256_chopped.vg<span class="w"> </span>-x<span class="w"> </span>5NM_256_chopped.xg
<a id="__codelineno-2-15" name="__codelineno-2-15" href="#__codelineno-2-15"></a><span class="c1">### gcsa index, it takes .......</span>
<a id="__codelineno-2-16" name="__codelineno-2-16" href="#__codelineno-2-16"></a>
<a id="__codelineno-2-17" name="__codelineno-2-17" href="#__codelineno-2-17"></a>vg<span class="w"> </span>index<span class="w"> </span>-b<span class="w"> </span>temp_dir<span class="w"> </span>-g<span class="w"> </span>5NM_256_chopped.gcsa<span class="w"> </span>-x<span class="w"> </span>5NM_256_chopped.xg<span class="w"> </span>-g<span class="w"> </span>5NM_256_chopped.gcsa<span class="w"> </span>-k<span class="w"> </span><span class="m">16</span><span class="w"> </span>5NM_256_chopped.vg
</code></pre></div>
</div>
<details class="terminal">
<summary>code</summary>
<div class="highlight"><pre><span></span><code><a id="__codelineno-3-1" name="__codelineno-3-1" href="#__codelineno-3-1"></a><span class="c1">#run a slurm job for this, it takes ~10 mins based on the following setting </span>
<a id="__codelineno-3-2" name="__codelineno-3-2" href="#__codelineno-3-2"></a>
<a id="__codelineno-3-3" name="__codelineno-3-3" href="#__codelineno-3-3"></a>
<a id="__codelineno-3-4" name="__codelineno-3-4" href="#__codelineno-3-4"></a><span class="c1">#!/bin/bash</span>
<a id="__codelineno-3-5" name="__codelineno-3-5" href="#__codelineno-3-5"></a>
<a id="__codelineno-3-6" name="__codelineno-3-6" href="#__codelineno-3-6"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a id="__codelineno-3-7" name="__codelineno-3-7" href="#__codelineno-3-7"></a><span class="c1">#SBATCH --job-name      build_index_for_5NMGraph</span>
<a id="__codelineno-3-8" name="__codelineno-3-8" href="#__codelineno-3-8"></a><span class="c1">#SBATCH --cpus-per-task 8</span>
<a id="__codelineno-3-9" name="__codelineno-3-9" href="#__codelineno-3-9"></a><span class="c1">#SBATCH --mem           16G</span>
<a id="__codelineno-3-10" name="__codelineno-3-10" href="#__codelineno-3-10"></a><span class="c1">#SBATCH --time          1:00:00</span>
<a id="__codelineno-3-11" name="__codelineno-3-11" href="#__codelineno-3-11"></a><span class="c1">#SBATCH --error         %x_%j.err</span>
<a id="__codelineno-3-12" name="__codelineno-3-12" href="#__codelineno-3-12"></a><span class="c1">#SBATCH --output        %x_%j.out</span>
<a id="__codelineno-3-13" name="__codelineno-3-13" href="#__codelineno-3-13"></a>
<a id="__codelineno-3-14" name="__codelineno-3-14" href="#__codelineno-3-14"></a><span class="c1"># Modules</span>
<a id="__codelineno-3-15" name="__codelineno-3-15" href="#__codelineno-3-15"></a>module<span class="w"> </span>purge
<a id="__codelineno-3-16" name="__codelineno-3-16" href="#__codelineno-3-16"></a>module<span class="w"> </span>load<span class="w"> </span>vg/1.46.0
<a id="__codelineno-3-17" name="__codelineno-3-17" href="#__codelineno-3-17"></a>
<a id="__codelineno-3-18" name="__codelineno-3-18" href="#__codelineno-3-18"></a><span class="c1"># Variables</span>
<a id="__codelineno-3-19" name="__codelineno-3-19" href="#__codelineno-3-19"></a><span class="c1">#cd /home/zyang/pg_workshop/graph_NGS</span>
<a id="__codelineno-3-20" name="__codelineno-3-20" href="#__codelineno-3-20"></a><span class="nv">data</span><span class="o">=</span>5NM.gfa
<a id="__codelineno-3-21" name="__codelineno-3-21" href="#__codelineno-3-21"></a>mkdir<span class="w"> </span>-p<span class="w"> </span>temp_dir
<a id="__codelineno-3-22" name="__codelineno-3-22" href="#__codelineno-3-22"></a>
<a id="__codelineno-3-23" name="__codelineno-3-23" href="#__codelineno-3-23"></a><span class="c1"># Convert graph into 256 bp chunks, saving as vg format</span>
<a id="__codelineno-3-24" name="__codelineno-3-24" href="#__codelineno-3-24"></a>vg<span class="w"> </span>mod<span class="w"> </span>-X<span class="w"> </span><span class="m">256</span><span class="w"> </span>5NM.gfa<span class="w"> </span>&gt;<span class="w"> </span>5NM_256.vg
<a id="__codelineno-3-25" name="__codelineno-3-25" href="#__codelineno-3-25"></a>
<a id="__codelineno-3-26" name="__codelineno-3-26" href="#__codelineno-3-26"></a><span class="c1">#small graph is ok without prunning</span>
<a id="__codelineno-3-27" name="__codelineno-3-27" href="#__codelineno-3-27"></a><span class="c1"># Build xg and gcsa index</span>
<a id="__codelineno-3-28" name="__codelineno-3-28" href="#__codelineno-3-28"></a><span class="c1">#vg index -b ${temp_dir} -t $cpus-per-task -x 5NM_256.xg -g 5NM_256.gcsa -k 16 5NM_256.vg</span>
<a id="__codelineno-3-29" name="__codelineno-3-29" href="#__codelineno-3-29"></a>
<a id="__codelineno-3-30" name="__codelineno-3-30" href="#__codelineno-3-30"></a><span class="c1">#complex graph will need to prune first before generating index</span>
<a id="__codelineno-3-31" name="__codelineno-3-31" href="#__codelineno-3-31"></a><span class="c1">### pruning: use -M if pruning fails</span>
<a id="__codelineno-3-32" name="__codelineno-3-32" href="#__codelineno-3-32"></a>vg<span class="w"> </span>prune<span class="w"> </span>-u<span class="w"> </span>-m<span class="w"> </span>node-mapping.tmp<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span><span class="w"> </span>-k<span class="w"> </span><span class="m">24</span><span class="w"> </span>5NM_256.vg<span class="w"> </span>&gt;<span class="w"> </span>5NM_256_chopped.vg
<a id="__codelineno-3-33" name="__codelineno-3-33" href="#__codelineno-3-33"></a>
<a id="__codelineno-3-34" name="__codelineno-3-34" href="#__codelineno-3-34"></a>vg<span class="w"> </span>index<span class="w"> </span>5NM_256_chopped.vg<span class="w"> </span>-x<span class="w"> </span>5NM_256_chopped.xg
<a id="__codelineno-3-35" name="__codelineno-3-35" href="#__codelineno-3-35"></a><span class="c1">### gcsa index</span>
<a id="__codelineno-3-36" name="__codelineno-3-36" href="#__codelineno-3-36"></a>vg<span class="w"> </span>index<span class="w"> </span>-b<span class="w"> </span>temp_dir<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span><span class="w"> </span>-x<span class="w"> </span>5NM_256_chopped.xg<span class="w"> </span>-g<span class="w"> </span>5NM_256_chopped.gcsa<span class="w"> </span>-k<span class="w"> </span><span class="m">16</span><span class="w"> </span>5NM_256_chopped.vg
</code></pre></div>
</details>
<h2 id="map-ngs-reads-to-graph">Map NGS reads to graph<a class="headerlink" href="#map-ngs-reads-to-graph" title="Permanent link">&para;</a></h2>
<p>Map reads back to graph reference</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-4-1" name="__codelineno-4-1" href="#__codelineno-4-1"></a><span class="c1"># Modules</span>
<a id="__codelineno-4-2" name="__codelineno-4-2" href="#__codelineno-4-2"></a>module<span class="w"> </span>purge
<a id="__codelineno-4-3" name="__codelineno-4-3" href="#__codelineno-4-3"></a>module<span class="w"> </span>load<span class="w"> </span>vg/1.46.0
<a id="__codelineno-4-4" name="__codelineno-4-4" href="#__codelineno-4-4"></a>
<a id="__codelineno-4-5" name="__codelineno-4-5" href="#__codelineno-4-5"></a>
<a id="__codelineno-4-6" name="__codelineno-4-6" href="#__codelineno-4-6"></a><span class="c1"># Map reads</span>
<a id="__codelineno-4-7" name="__codelineno-4-7" href="#__codelineno-4-7"></a><span class="c1">#SAMN13450731 is the NCBI record for NMI138 </span>
<a id="__codelineno-4-8" name="__codelineno-4-8" href="#__codelineno-4-8"></a>vg<span class="w"> </span>map<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span><span class="w"> </span>-d<span class="w"> </span>5NM_256_chopped<span class="w"> </span>-f<span class="w"> </span>NMI138_S5_R1_P.fastq.gz<span class="w"> </span>-f<span class="w"> </span>NMI138_S5_R1_P.fastq.gz<span class="w"> </span>-N<span class="w"> </span>NMI138<span class="w">  </span>&gt;<span class="w"> </span>NM138.vgmap_5NM.gam
<a id="__codelineno-4-9" name="__codelineno-4-9" href="#__codelineno-4-9"></a>
<a id="__codelineno-4-10" name="__codelineno-4-10" href="#__codelineno-4-10"></a><span class="c1"># Output mapping statistics</span>
<a id="__codelineno-4-11" name="__codelineno-4-11" href="#__codelineno-4-11"></a>vg<span class="w"> </span>stats<span class="w"> </span>-a<span class="w"> </span>NM138.vgmap_5NM.gam<span class="w"> </span>&gt;<span class="w"> </span>/NM138.vgmap_4Sim_stats<span class="w"> </span>
</code></pre></div>
</div>
<p>Map reads back to graph reference as a slurm job.</p>
<details class="terminal">
<summary>code</summary>
<div class="highlight"><pre><span></span><code><a id="__codelineno-5-1" name="__codelineno-5-1" href="#__codelineno-5-1"></a><span class="ch">#!/bin/bash -e</span>
<a id="__codelineno-5-2" name="__codelineno-5-2" href="#__codelineno-5-2"></a>
<a id="__codelineno-5-3" name="__codelineno-5-3" href="#__codelineno-5-3"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a id="__codelineno-5-4" name="__codelineno-5-4" href="#__codelineno-5-4"></a><span class="c1">#SBATCH --job-name      vgmap_5e_5NM</span>
<a id="__codelineno-5-5" name="__codelineno-5-5" href="#__codelineno-5-5"></a><span class="c1">#SBATCH --cpus-per-task 24</span>
<a id="__codelineno-5-6" name="__codelineno-5-6" href="#__codelineno-5-6"></a><span class="c1">#SBATCH --mem           4G</span>
<a id="__codelineno-5-7" name="__codelineno-5-7" href="#__codelineno-5-7"></a><span class="c1">#SBATCH --time          01:00:00</span>
<a id="__codelineno-5-8" name="__codelineno-5-8" href="#__codelineno-5-8"></a><span class="c1">#SBATCH --error         %x_%j.err</span>
<a id="__codelineno-5-9" name="__codelineno-5-9" href="#__codelineno-5-9"></a><span class="c1">#SBATCH --output        %x_%j.out</span>
<a id="__codelineno-5-10" name="__codelineno-5-10" href="#__codelineno-5-10"></a>
<a id="__codelineno-5-11" name="__codelineno-5-11" href="#__codelineno-5-11"></a><span class="c1"># Modules</span>
<a id="__codelineno-5-12" name="__codelineno-5-12" href="#__codelineno-5-12"></a>module<span class="w"> </span>purge
<a id="__codelineno-5-13" name="__codelineno-5-13" href="#__codelineno-5-13"></a>module<span class="w"> </span>load<span class="w"> </span>vg/1.46.0
<a id="__codelineno-5-14" name="__codelineno-5-14" href="#__codelineno-5-14"></a>
<a id="__codelineno-5-15" name="__codelineno-5-15" href="#__codelineno-5-15"></a><span class="c1"># Variables</span>
<a id="__codelineno-5-16" name="__codelineno-5-16" href="#__codelineno-5-16"></a><span class="nv">wkdir</span><span class="o">=</span>~/pg_test/graph_NGS
<a id="__codelineno-5-17" name="__codelineno-5-17" href="#__codelineno-5-17"></a><span class="nv">index</span><span class="o">=</span><span class="si">${</span><span class="nv">wkdir</span><span class="si">}</span>/5NM_256_chopped.gcsa
<a id="__codelineno-5-18" name="__codelineno-5-18" href="#__codelineno-5-18"></a><span class="nv">index_prefix</span><span class="o">=</span><span class="si">${</span><span class="nv">index</span><span class="p">%%.gcsa</span>
<a id="__codelineno-5-19" name="__codelineno-5-19" href="#__codelineno-5-19"></a>
<a id="__codelineno-5-20" name="__codelineno-5-20" href="#__codelineno-5-20"></a><span class="p"># Map reads</span>
<a id="__codelineno-5-21" name="__codelineno-5-21" href="#__codelineno-5-21"></a><span class="p">vg map -t </span><span class="nv">$SLURM_CPUS_PER_TASK</span><span class="p"> -d </span><span class="si">${</span><span class="nv">index_prefix</span><span class="si">}</span><span class="p"> -f NMI138_S5_R1_P.fastq.gz -f NMI138_S5_R2_P.fastq.gz -N NMI138 &gt; NM138.vgmap_5NM.gam</span>
<a id="__codelineno-5-22" name="__codelineno-5-22" href="#__codelineno-5-22"></a>
<a id="__codelineno-5-23" name="__codelineno-5-23" href="#__codelineno-5-23"></a><span class="p"># Output mapping statistics</span>
<a id="__codelineno-5-24" name="__codelineno-5-24" href="#__codelineno-5-24"></a><span class="p">vg stats -a NM138.vgmap_5NM.gam &gt; NM138.vgmap_5NM_stats</span>
</code></pre></div>
</details>
<!-- 3 min per read pair -->

<h2 id="genotying-known-variants">Genotying known variants<a class="headerlink" href="#genotying-known-variants" title="Permanent link">&para;</a></h2>
<p>Generate snarls of graph.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-6-1" name="__codelineno-6-1" href="#__codelineno-6-1"></a>module<span class="w"> </span>purge
<a id="__codelineno-6-2" name="__codelineno-6-2" href="#__codelineno-6-2"></a>module<span class="w"> </span>load<span class="w"> </span>vg/1.46.0
<a id="__codelineno-6-3" name="__codelineno-6-3" href="#__codelineno-6-3"></a>
<a id="__codelineno-6-4" name="__codelineno-6-4" href="#__codelineno-6-4"></a>vg<span class="w"> </span>snarls<span class="w"> </span>-t<span class="w"> </span><span class="m">2</span><span class="w"> </span>5NM_256.xg<span class="w"> </span>&gt;<span class="w"> </span>5NM_256.xg.snarls
</code></pre></div>
</div>
<p>Perform genotyping.</p>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-7-1" name="__codelineno-7-1" href="#__codelineno-7-1"></a><span class="c1"># Calculate support reads</span>
<a id="__codelineno-7-2" name="__codelineno-7-2" href="#__codelineno-7-2"></a>vg<span class="w"> </span>pack<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span><span class="w"> </span>-x<span class="w"> </span>5NM_256.xg<span class="w"> </span>-g<span class="w"> </span>NM138.vgmap_5NM.gam<span class="w"> </span>-o<span class="w"> </span>NM138_vgmap_5NM_256.pack
<a id="__codelineno-7-3" name="__codelineno-7-3" href="#__codelineno-7-3"></a>
<a id="__codelineno-7-4" name="__codelineno-7-4" href="#__codelineno-7-4"></a><span class="c1"># Call variants using the same coordinates and include reference calls for comparison</span>
<a id="__codelineno-7-5" name="__codelineno-7-5" href="#__codelineno-7-5"></a>vg<span class="w"> </span>call<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span><span class="w"> </span>-m<span class="w"> </span><span class="m">3</span>,10<span class="w"> </span>5NM_256.xg<span class="w"> </span>-k<span class="w"> </span>NM138_vgmap_5NM_256.pack<span class="w"> </span>-r<span class="w"> </span>5NM_256.xg.snarls<span class="w"> </span>-a<span class="w"> </span>&gt;<span class="w"> </span>NM138.vgmap_5NM_256.pack_allR10S3.vcf
</code></pre></div>
</div>
<details class="terminal">
<summary>code</summary>
<div class="highlight"><pre><span></span><code><a id="__codelineno-8-1" name="__codelineno-8-1" href="#__codelineno-8-1"></a><span class="ch">#!/bin/bash -e</span>
<a id="__codelineno-8-2" name="__codelineno-8-2" href="#__codelineno-8-2"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a id="__codelineno-8-3" name="__codelineno-8-3" href="#__codelineno-8-3"></a><span class="c1">#SBATCH --job-name      5NM_vgmap_genotying</span>
<a id="__codelineno-8-4" name="__codelineno-8-4" href="#__codelineno-8-4"></a><span class="c1">#SBATCH --cpus-per-task 24</span>
<a id="__codelineno-8-5" name="__codelineno-8-5" href="#__codelineno-8-5"></a><span class="c1">#SBATCH --mem           4G</span>
<a id="__codelineno-8-6" name="__codelineno-8-6" href="#__codelineno-8-6"></a><span class="c1">#SBATCH --time          01:00:00</span>
<a id="__codelineno-8-7" name="__codelineno-8-7" href="#__codelineno-8-7"></a><span class="c1">#SBATCH --error         %x_%j.err</span>
<a id="__codelineno-8-8" name="__codelineno-8-8" href="#__codelineno-8-8"></a><span class="c1">#SBATCH --output        %x_%j.out</span>
<a id="__codelineno-8-9" name="__codelineno-8-9" href="#__codelineno-8-9"></a>
<a id="__codelineno-8-10" name="__codelineno-8-10" href="#__codelineno-8-10"></a><span class="c1"># Modules</span>
<a id="__codelineno-8-11" name="__codelineno-8-11" href="#__codelineno-8-11"></a>module<span class="w"> </span>purge<span class="w"> </span>
<a id="__codelineno-8-12" name="__codelineno-8-12" href="#__codelineno-8-12"></a>module<span class="w"> </span>load<span class="w"> </span>vg/1.46.0
<a id="__codelineno-8-13" name="__codelineno-8-13" href="#__codelineno-8-13"></a>
<a id="__codelineno-8-14" name="__codelineno-8-14" href="#__codelineno-8-14"></a>vg<span class="w"> </span>index<span class="w"> </span>5NM_256.vg<span class="w"> </span>-x<span class="w"> </span>5NM_256.xg
<a id="__codelineno-8-15" name="__codelineno-8-15" href="#__codelineno-8-15"></a>
<a id="__codelineno-8-16" name="__codelineno-8-16" href="#__codelineno-8-16"></a>vg<span class="w"> </span>snarls<span class="w"> </span>-t<span class="w"> </span><span class="m">2</span><span class="w"> </span>5NM_256.xg<span class="w"> </span>&gt;<span class="w"> </span>5NM_256.xg.snarls
<a id="__codelineno-8-17" name="__codelineno-8-17" href="#__codelineno-8-17"></a>
<a id="__codelineno-8-18" name="__codelineno-8-18" href="#__codelineno-8-18"></a><span class="c1"># Calculate support reads</span>
<a id="__codelineno-8-19" name="__codelineno-8-19" href="#__codelineno-8-19"></a>vg<span class="w"> </span>pack<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span><span class="w"> </span>-x<span class="w"> </span>5NM_256.xg<span class="w"> </span>-g<span class="w"> </span>NM138.vgmap_5NM.gam<span class="w"> </span>-o<span class="w"> </span>NM138_vgmap_5NM_256.pack
<a id="__codelineno-8-20" name="__codelineno-8-20" href="#__codelineno-8-20"></a>
<a id="__codelineno-8-21" name="__codelineno-8-21" href="#__codelineno-8-21"></a><span class="c1"># Call variants using the same coordinates and include reference calls for comparison</span>
<a id="__codelineno-8-22" name="__codelineno-8-22" href="#__codelineno-8-22"></a>vg<span class="w"> </span>call<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span><span class="w"> </span>-m<span class="w"> </span><span class="m">3</span>,10<span class="w"> </span>5NM_256.xg<span class="w"> </span>-k<span class="w"> </span>NM138_vgmap_5NM_256.pack<span class="w"> </span>-r<span class="w"> </span>5NM_256.xg.snarls<span class="w"> </span>-a<span class="w"> </span>&gt;<span class="w"> </span>NM138.vgmap_5NM_256.pack_allR10S3.vcf
</code></pre></div>
</details>
<h2 id="novel-variant-calling-using-graph-reference">Novel variant calling using graph reference<a class="headerlink" href="#novel-variant-calling-using-graph-reference" title="Permanent link">&para;</a></h2>
<div class="admonition terminal">
<p class="admonition-title">code</p>
<div class="highlight"><pre><span></span><code><a id="__codelineno-9-1" name="__codelineno-9-1" href="#__codelineno-9-1"></a><span class="w"> </span><span class="c1"># In order to also consider novel variants from the reads, use the augmented graph and gam </span>
<a id="__codelineno-9-2" name="__codelineno-9-2" href="#__codelineno-9-2"></a><span class="c1"># (as created in the &quot;Augmentation&quot; example using vg augment -A).</span>
<a id="__codelineno-9-3" name="__codelineno-9-3" href="#__codelineno-9-3"></a><span class="c1"># Augment the graph with all variation from the GAM, saving to aug.vg</span>
<a id="__codelineno-9-4" name="__codelineno-9-4" href="#__codelineno-9-4"></a><span class="c1">### Augment the graph with all variation form the GAM except </span>
<a id="__codelineno-9-5" name="__codelineno-9-5" href="#__codelineno-9-5"></a><span class="c1">### that implied by soft clips, saving to aug.vg.</span>
<a id="__codelineno-9-6" name="__codelineno-9-6" href="#__codelineno-9-6"></a><span class="c1">### *aug-gam contains the same reads as aln.gam but mapped to aug.vg</span>
<a id="__codelineno-9-7" name="__codelineno-9-7" href="#__codelineno-9-7"></a>
<a id="__codelineno-9-8" name="__codelineno-9-8" href="#__codelineno-9-8"></a><span class="c1"># Augment graph</span>
<a id="__codelineno-9-9" name="__codelineno-9-9" href="#__codelineno-9-9"></a>vg<span class="w"> </span>augment<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span><span class="w">  </span>5NM_256_chopped.vg<span class="w"> </span>NM138.vgmap_5NM.gam<span class="w"> </span>-A<span class="w"> </span>NM138.nofilt_aug.gam<span class="w"> </span>&gt;<span class="w"> </span>NM138.nofilt_aug.vg
<a id="__codelineno-9-10" name="__codelineno-9-10" href="#__codelineno-9-10"></a>
<a id="__codelineno-9-11" name="__codelineno-9-11" href="#__codelineno-9-11"></a><span class="c1"># Index the augmented graph</span>
<a id="__codelineno-9-12" name="__codelineno-9-12" href="#__codelineno-9-12"></a>vg<span class="w"> </span>index<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span><span class="w">  </span>NM138.nofilt_aug.vg<span class="w">  </span>-x<span class="w"> </span>NM138.nofilt_aug.xg
<a id="__codelineno-9-13" name="__codelineno-9-13" href="#__codelineno-9-13"></a>
<a id="__codelineno-9-14" name="__codelineno-9-14" href="#__codelineno-9-14"></a><span class="c1"># Compute the all read support from the augmented GAM</span>
<a id="__codelineno-9-15" name="__codelineno-9-15" href="#__codelineno-9-15"></a>vg<span class="w"> </span>pack<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span><span class="w"> </span>-x<span class="w"> </span>NM138.nofilt_aug.xg<span class="w"> </span>-g<span class="w"> </span>NM138.nofilt_aug.gam<span class="w"> </span>-o<span class="w"> </span>NM138.nofilt_aug_allR.pack
<a id="__codelineno-9-16" name="__codelineno-9-16" href="#__codelineno-9-16"></a>
<a id="__codelineno-9-17" name="__codelineno-9-17" href="#__codelineno-9-17"></a><span class="c1"># Call variants.</span>
<a id="__codelineno-9-18" name="__codelineno-9-18" href="#__codelineno-9-18"></a><span class="c1">#we need to trouble shooting about this, why the vcf file is empty </span>
<a id="__codelineno-9-19" name="__codelineno-9-19" href="#__codelineno-9-19"></a>vg<span class="w"> </span>call<span class="w"> </span>-t<span class="w"> </span><span class="m">8</span><span class="w"> </span>-m<span class="w"> </span><span class="m">3</span>,10<span class="w"> </span>NM138.nofilt_aug.xg<span class="w"> </span>-k<span class="w"> </span>NM138.nofilt_aug_allR.pack<span class="w"> </span>&gt;<span class="w"> </span>NM138.nofilt_aug_allR.pack.vcf
</code></pre></div>
</div>
<details class="terminal-2">
<summary>Slurm script</summary>
<div class="highlight"><pre><span></span><code><a id="__codelineno-10-1" name="__codelineno-10-1" href="#__codelineno-10-1"></a><span class="ch">#!/bin/bash -e</span>
<a id="__codelineno-10-2" name="__codelineno-10-2" href="#__codelineno-10-2"></a>
<a id="__codelineno-10-3" name="__codelineno-10-3" href="#__codelineno-10-3"></a><span class="c1">#SBATCH --account       nesi02659</span>
<a id="__codelineno-10-4" name="__codelineno-10-4" href="#__codelineno-10-4"></a><span class="c1">#SBATCH --job-name      5.call_novel_variant</span>
<a id="__codelineno-10-5" name="__codelineno-10-5" href="#__codelineno-10-5"></a><span class="c1">#SBATCH --cpus-per-task 24</span>
<a id="__codelineno-10-6" name="__codelineno-10-6" href="#__codelineno-10-6"></a><span class="c1">#SBATCH --mem           4G</span>
<a id="__codelineno-10-7" name="__codelineno-10-7" href="#__codelineno-10-7"></a><span class="c1">#SBATCH --time          01:00:00</span>
<a id="__codelineno-10-8" name="__codelineno-10-8" href="#__codelineno-10-8"></a><span class="c1">#SBATCH --error         %x_%j.err</span>
<a id="__codelineno-10-9" name="__codelineno-10-9" href="#__codelineno-10-9"></a><span class="c1">#SBATCH --output        %x_%j.out</span>
<a id="__codelineno-10-10" name="__codelineno-10-10" href="#__codelineno-10-10"></a><span class="c1">#SBATCH --array         0-5</span>
<a id="__codelineno-10-11" name="__codelineno-10-11" href="#__codelineno-10-11"></a>
<a id="__codelineno-10-12" name="__codelineno-10-12" href="#__codelineno-10-12"></a><span class="c1"># Modules</span>
<a id="__codelineno-10-13" name="__codelineno-10-13" href="#__codelineno-10-13"></a>module<span class="w"> </span>purge<span class="w"> </span>
<a id="__codelineno-10-14" name="__codelineno-10-14" href="#__codelineno-10-14"></a>module<span class="w"> </span>load<span class="w"> </span>vg/1.46.0
<a id="__codelineno-10-15" name="__codelineno-10-15" href="#__codelineno-10-15"></a>
<a id="__codelineno-10-16" name="__codelineno-10-16" href="#__codelineno-10-16"></a><span class="c1"># Variables</span>
<a id="__codelineno-10-17" name="__codelineno-10-17" href="#__codelineno-10-17"></a><span class="nv">wkdir</span><span class="o">=</span>~/pg_workshop/graph_NGS
<a id="__codelineno-10-18" name="__codelineno-10-18" href="#__codelineno-10-18"></a><span class="nv">gam_dir</span><span class="o">=</span><span class="si">${</span><span class="nv">wkdir</span><span class="si">}</span>/graph_based_mapping
<a id="__codelineno-10-19" name="__codelineno-10-19" href="#__codelineno-10-19"></a><span class="nv">out_dir</span><span class="o">=</span><span class="si">${</span><span class="nv">wkdir</span><span class="si">}</span>/vgmap_5e_sim4_allR10S3_novelcalling
<a id="__codelineno-10-20" name="__codelineno-10-20" href="#__codelineno-10-20"></a>
<a id="__codelineno-10-21" name="__codelineno-10-21" href="#__codelineno-10-21"></a>mkdir<span class="w"> </span>-p<span class="w"> </span><span class="si">${</span><span class="nv">out_dir</span><span class="si">}</span>
<a id="__codelineno-10-22" name="__codelineno-10-22" href="#__codelineno-10-22"></a>
<a id="__codelineno-10-23" name="__codelineno-10-23" href="#__codelineno-10-23"></a><span class="nv">vg</span><span class="o">=</span><span class="si">${</span><span class="nv">wkdir</span><span class="si">}</span>/refs/4Sim_1K96_256.vg
<a id="__codelineno-10-24" name="__codelineno-10-24" href="#__codelineno-10-24"></a><span class="nv">xg</span><span class="o">=</span><span class="si">${</span><span class="nv">wkdir</span><span class="si">}</span>/refs/4Sim_1K96_256.xg
<a id="__codelineno-10-25" name="__codelineno-10-25" href="#__codelineno-10-25"></a>
<a id="__codelineno-10-26" name="__codelineno-10-26" href="#__codelineno-10-26"></a><span class="c1"># Array</span>
<a id="__codelineno-10-27" name="__codelineno-10-27" href="#__codelineno-10-27"></a><span class="nv">file_array</span><span class="o">=(</span><span class="si">${</span><span class="nv">gam_dir</span><span class="si">}</span>/*.gam<span class="o">)</span>
<a id="__codelineno-10-28" name="__codelineno-10-28" href="#__codelineno-10-28"></a><span class="nv">file</span><span class="o">=</span><span class="si">${</span><span class="nv">file_array</span><span class="p">[</span><span class="nv">$SLURM_ARRAY_TASK_ID</span><span class="p">]</span><span class="si">}</span>
<a id="__codelineno-10-29" name="__codelineno-10-29" href="#__codelineno-10-29"></a><span class="nv">prefix</span><span class="o">=</span><span class="k">$(</span>basename<span class="w"> </span><span class="si">${</span><span class="nv">file</span><span class="si">}</span><span class="w"> </span>.wgsim_er0.005.vgmap_4Sim.gam<span class="k">)</span>
<a id="__codelineno-10-30" name="__codelineno-10-30" href="#__codelineno-10-30"></a>
<a id="__codelineno-10-31" name="__codelineno-10-31" href="#__codelineno-10-31"></a><span class="c1"># In order to also consider novel variants from the reads, use the augmented graph and gam </span>
<a id="__codelineno-10-32" name="__codelineno-10-32" href="#__codelineno-10-32"></a><span class="c1"># (as created in the &quot;Augmentation&quot; example using vg augment -A).</span>
<a id="__codelineno-10-33" name="__codelineno-10-33" href="#__codelineno-10-33"></a><span class="c1"># Augment the graph with all variation from the GAM, saving to aug.vg</span>
<a id="__codelineno-10-34" name="__codelineno-10-34" href="#__codelineno-10-34"></a><span class="c1">### Augment the graph with all variation form the GAM except </span>
<a id="__codelineno-10-35" name="__codelineno-10-35" href="#__codelineno-10-35"></a><span class="c1">### that implied by soft clips, saving to aug.vg.</span>
<a id="__codelineno-10-36" name="__codelineno-10-36" href="#__codelineno-10-36"></a><span class="c1">### *aug-gam contains the same reads as aln.gam but mapped to aug.vg</span>
<a id="__codelineno-10-37" name="__codelineno-10-37" href="#__codelineno-10-37"></a>
<a id="__codelineno-10-38" name="__codelineno-10-38" href="#__codelineno-10-38"></a><span class="c1"># Augment graph</span>
<a id="__codelineno-10-39" name="__codelineno-10-39" href="#__codelineno-10-39"></a>vg<span class="w"> </span>augment<span class="w"> </span>-t<span class="w"> </span><span class="nv">$SLURM_CPUS_PER_TASK</span><span class="w"> </span><span class="si">${</span><span class="nv">vg</span><span class="si">}</span><span class="w"> </span><span class="si">${</span><span class="nv">file</span><span class="si">}</span><span class="w"> </span>-A<span class="w"> </span><span class="si">${</span><span class="nv">out_dir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.nofilt_aug.gam<span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">out_dir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.nofilt_aug.vg
<a id="__codelineno-10-40" name="__codelineno-10-40" href="#__codelineno-10-40"></a>
<a id="__codelineno-10-41" name="__codelineno-10-41" href="#__codelineno-10-41"></a><span class="c1"># Index the augmented graph</span>
<a id="__codelineno-10-42" name="__codelineno-10-42" href="#__codelineno-10-42"></a>vg<span class="w"> </span>index<span class="w"> </span>-t<span class="w"> </span><span class="nv">$SLURM_CPUS_PER_TASK</span><span class="w"> </span><span class="si">${</span><span class="nv">out_dir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.nofilt_aug.vg<span class="w"> </span>-x<span class="w"> </span><span class="si">${</span><span class="nv">out_dir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.nofilt_aug.xg
<a id="__codelineno-10-43" name="__codelineno-10-43" href="#__codelineno-10-43"></a>
<a id="__codelineno-10-44" name="__codelineno-10-44" href="#__codelineno-10-44"></a><span class="c1"># Compute the all read support from the augmented GAM</span>
<a id="__codelineno-10-45" name="__codelineno-10-45" href="#__codelineno-10-45"></a>vg<span class="w"> </span>pack<span class="w"> </span>-t<span class="w"> </span><span class="nv">$SLURM_CPUS_PER_TASK</span><span class="w"> </span>-x<span class="w"> </span><span class="si">${</span><span class="nv">out_dir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.nofilt_aug.xg<span class="w"> </span>-g<span class="w"> </span><span class="si">${</span><span class="nv">out_dir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.nofilt_aug.gam<span class="w"> </span>-o<span class="w"> </span><span class="si">${</span><span class="nv">out_dir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.nofilt_aug_allR.pack
<a id="__codelineno-10-46" name="__codelineno-10-46" href="#__codelineno-10-46"></a>
<a id="__codelineno-10-47" name="__codelineno-10-47" href="#__codelineno-10-47"></a><span class="c1"># Call variants</span>
<a id="__codelineno-10-48" name="__codelineno-10-48" href="#__codelineno-10-48"></a>vg<span class="w"> </span>call<span class="w"> </span>-t<span class="w"> </span><span class="nv">$SLURM_CPUS_PER_TASK</span><span class="w"> </span>-m<span class="w"> </span><span class="m">3</span>,10<span class="w"> </span><span class="si">${</span><span class="nv">out_dir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.nofilt_aug.xg<span class="w"> </span>-k<span class="w"> </span><span class="si">${</span><span class="nv">out_dir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.nofilt_aug_allR.pack<span class="w"> </span>&gt;<span class="w"> </span><span class="si">${</span><span class="nv">out_dir</span><span class="si">}</span>/<span class="si">${</span><span class="nv">prefix</span><span class="si">}</span>.nofilt_aug_allR.pack.vcf
</code></pre></div>
</details>





                
              </article>
            </div>
          
          
        </div>
        
          <button type="button" class="md-top md-icon" data-md-component="top" hidden>
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M13 20h-2V8l-5.5 5.5-1.42-1.42L12 4.16l7.92 7.92-1.42 1.42L13 8v12Z"/></svg>
            Back to top
          </button>
        
      </main>
      
        <footer class="md-footer">
  
    
      
      <nav class="md-footer__inner md-grid" aria-label="Footer" >
        
          
          <a href="../supplementary_2/" class="md-footer__link md-footer__link--prev" aria-label="Previous: Jupyter Virtual Desktop" rel="prev">
            <div class="md-footer__button md-icon">
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
            </div>
            <div class="md-footer__title">
              <span class="md-footer__direction">
                Previous
              </span>
              <div class="md-ellipsis">
                Jupyter Virtual Desktop
              </div>
            </div>
          </a>
        
        
      </nav>
    
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Pangenome Graphs workshop material is licensed under a <a rel="noopener" target="_blank" href="https://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html">GNU General Public License v2.0</a>
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    <script id="__config" type="application/json">{"base": "../..", "features": ["navigation.tracking", "navigation.sections", "navigation.expand", "navigation.top", "navigation.footer", "content.code.copy", "content.code.annotate"], "search": "../../assets/javascripts/workers/search.74e28a9f.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../../assets/javascripts/bundle.220ee61c.min.js"></script>
      
        
          <script src="../../javascripts/mathjax.js"></script>
        
      
        
          <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
        
      
        
          <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
        
      
    
  <script>document$.subscribe(() => {const lightbox = GLightbox({"touchNavigation": true, "loop": false, "zoomable": true, "draggable": true, "openEffect": "zoom", "closeEffect": "zoom", "slideEffect": "slide"});})</script></body>
</html>